<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Importing data • gpatterns</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">gpatterns</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Home</a>
</li>
<li>
  <a href="../articles/import.html">
    <span class="fa fa-truck"></span>
     
    Import
  </a>
</li>
<li>
  <a href="../articles/articles/analysis.html">
    <span class="fa fa-bar-chart"></span>
     
    Analyse
  </a>
</li>
<li>
  <a href="../articles/articles/advanced.html">
    <span class="fa fa-stethoscope"></span>
     
    Advanced
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-th-list"></span>
     
    Reference
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://bitbucket.org/tanaylab/gpatterns">
    <span class="fa fa-bitbucket-square"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>Importing data</h1>
                        <h4 class="author">Aviezer Lifshitz</h4>
            
            <h4 class="date">2017-06-18</h4>
          </div>

    
    
<div class="contents">
<p>Data can be imported to gpatterns from raw reads (.fastq files), mapped reads (.bam files) or from <code>tidy_cpgs</code> files were generated by the gpatterns package.</p>
<div id="data-umi-rrbs-of-breast-and-lung-tumors" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#data-umi-rrbs-of-breast-and-lung-tumors" class="anchor"> </a></body></html>Data: UMI-RRBS of Breast and Lung tumors</h2>
<p>To explore gpatterns’ import functions, we’ll start with a dataset of UMI-RRBS of breast and lung tumors. The dataset contains 16 breast tumors samples from 4 patients, with multiple sections from each sample.</p>
<ul>
<li><em>table</em></li>
</ul>
</div>
<div id="importing-from-raw-reads--fastq" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#importing-from-raw-reads--fastq" class="anchor"> </a></body></html>Importing from raw reads (.fastq)</h2>
<div id="bissli2" class="section level3">
<h3 class="hasAnchor">
<html><body><a href="#bissli2" class="anchor"> </a></body></html>bissli2</h3>
<p><code>bissli2</code> is a wrapper around bowtie2 that maps bisulfite converted reads and extracts methylation calls. - more about bissli params and usage</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/gpatterns.bissli2.html">gpatterns.bissli2</a></span>(<span class="dt">r1_fastq =</span> <span class="st">'Breast_2460_1.3_R1/fastq/Breast_2460_1.3_R1.fastq'</span>, <span class="dt">r2_fastq =</span> <span class="st">'Breast_2460_1.3_R1/fastq/Breast_2460_1.3_R2.fastq'</span>, <span class="dt">out_bam =</span> <span class="st">'Breast_2460_1.3_R1/bam/Breast_2460_1.3.bam'</span>, <span class="dt">genome_seq =</span> <span class="st">"/net/mraid14/export/data/db/tgdb/hg19/seq"</span>, <span class="dt">genome_type=</span><span class="st">"ct_ga"</span>, <span class="dt">bissli2_idx =</span> <span class="st">"/net/mraid14/export/data/tools/bissli2/hg19/hg19"</span>)</code></pre></div>
</div>
</div>
<div id="importing-from-mapped-reads--bam" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#importing-from-mapped-reads--bam" class="anchor"> </a></body></html>Importing from mapped reads (.bam)</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/gpatterns.import_from_bam.html">gpatterns.import_from_bam</a></span>(<span class="st">'bam/Breast_2460_1.3.bam'</span>, <span class="dt">workdir=</span><span class="st">'Breast_2460_1.3_R1'</span>, <span class="dt">trim=</span><span class="dv">1</span>, <span class="dt">cgs_mask_file=</span><span class="st">'/net/mraid14/export/tgdata/users/aviezerl/proj/gpatterns_nugget/msp1_stickey_ends'</span>, <span class="dt">frag_intervs=</span><span class="st">'intervs.msp1.fid'</span>, <span class="dt">rm_off_target=</span>T, <span class="dt">maxdist=</span><span class="dv">4</span>, <span class="dt">use_sge=</span>F, <span class="dt">parallel=</span><span class="ot">TRUE</span>, <span class="dt">paired_end=</span><span class="ot">TRUE</span>)</code></pre></div>
</div>
<div id="importing-from-tidy_cpgs-files" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#importing-from-tidy_cpgs-files" class="anchor"> </a></body></html>Importing from <code>tidy_cpgs</code> files</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/gpatterns.import_from_tidy_cpgs.html">gpatterns.import_from_tidy_cpgs</a></span>(<span class="dt">tidy_cpgs=</span><span class="st">"Breast_2460_1.3_R1/tidy_cpgs"</span>, <span class="dt">track=</span><span class="st">"gpatterns_nugget.Breast_2460_1.3_R1"</span>, <span class="dt">description=</span><span class="st">""</span>, <span class="dt">groot=</span><span class="st">"/home/aviezerl/hg19"</span>, <span class="dt">max_span=</span><span class="dv">600</span>, <span class="dt">parallel=</span><span class="ot">TRUE</span>)</code></pre></div>
</div>
<div id="generating-methylation-patterns" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#generating-methylation-patterns" class="anchor"> </a></body></html>Generating methylation patterns</h2>
<p>In order to generate methylation patterns we first need to define the CpG scope, or ‘pattern space’, i.e. the set of adjacent CpGs we will be looking at. In theory, we could choose a different set of CpGs for each sample separately, in fact, gpatterns does this routinely by generating .patX tracks for each sample with patterns of length X for each CpG.</p>
<p>However, this approach becomes problematic if we want to compare the pattern distribution of samples directly. Therefore, the best practice is to define the ‘pattern space’ based on all the samples together. gpatterns provides two functions to calculate such space:</p>
<ul>
<li>
<code>gpatterns.tracks_to_pat_space</code>: generates patterns space by dividing the genome to non-overlapping chunks and taking the most covered stretches of <code>pat_l</code> adjacent CpGs.</li>
<li>
<code>gpatterns.intervs_to_pat_space</code>: generates pattern space for specific intervals (e.g. MSP1 fragments, capture regions, enhancers, CpG islands etc.) by taking the most covered stretches of <code>pat_l</code> CpGs than cover each interval, and only then calls <code>gpatterms.tracks_to_pat_space</code> to calculate for the rest of the genome. The function can be limited to only adjacent CpGs (<code>adjacent=TRUE</code>) or take the most covered CpGs even if there are other CpGs in the middle.</li>
</ul>
<p>Since our example is from RRBS data we would use <code>gpatterns.intervs_to_pat_space</code> with the msp1 fragments:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pat_space &lt;-<span class="st"> </span><span class="kw"><a href="../reference/gpatterns.intervs_to_pat_space.html">gpatterns.intervs_to_pat_space</a></span>(tracks, <span class="dt">intervals=</span><span class="st">'intervs.msp1.fid'</span>, <span class="dt">adjacent=</span><span class="ot">TRUE</span>, <span class="dt">pat_len=</span><span class="dv">5</span>, <span class="dt">parallel=</span><span class="ot">TRUE</span>)</code></pre></div>
<p>And now we can create the patterns attributes for each of our tracks:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">purrr::<span class="kw">walk</span>(tracks, ~<span class="st"> </span><span class="kw"><a href="../reference/gpatterns.create_patterns_track.html">gpatterns.create_patterns_track</a></span>(<span class="dt">track =</span> .x, <span class="dt">description =</span> <span class="kw">paste0</span>(<span class="st">'patterns of '</span>, .x), <span class="dt">pat_space=</span>pat_space, <span class="dt">max_missing=</span><span class="dv">0</span>))</code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#data-umi-rrbs-of-breast-and-lung-tumors">Data: UMI-RRBS of Breast and Lung tumors</a></li>
      <li><a href="#importing-from-raw-reads-.fastq">Importing from raw reads (.fastq)</a></li>
      <li><a href="#importing-from-mapped-reads-.bam">Importing from mapped reads (.bam)</a></li>
      <li><a href="#importing-from-tidy_cpgs-files">Importing from <code>tidy_cpgs</code> files</a></li>
      <li><a href="#generating-methylation-patterns">Generating methylation patterns</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Aviezer Lifshitz, Elad Chomsky, Amos Tanay.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
