<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Repsc vignette • Repsc</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Repsc vignette">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Repsc</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="../articles/Repsc.html">
    <span class="fa fa-play-circle"></span>
     
    Get started
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-th-list"></span>
     
    Reference
  </a>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-news"></span>
     
    News
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://bitbucket.org/tanaylab/repsc/src/default/">
    <span class="fa fa-bitbucket-square"></span>
     
    Bitbucket
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Repsc vignette</h1>
                        <h4 class="author">David Brocks</h4>
            
            <h4 class="date">2019-06-27</h4>
      
      
      <div class="hidden name"><code>Repsc.Rmd</code></div>

    </div>

    
    
<div id="prerequisites" class="section level1">
<h1 class="hasAnchor">
<a href="#prerequisites" class="anchor"></a>Prerequisites</h1>
<div id="genome-assembly-optional" class="section level2">
<h2 class="hasAnchor">
<a href="#genome-assembly-optional" class="anchor"></a>Genome assembly (optional)</h2>
<p>Repsc can compute family-wise multiple sequence alignments of all repeat sequences in the genome to improve read mapping onto a consensus model. To do so, we require the genome assembly of our organism of choice stored as a <em>BSgenome</em> object. You can retrieve the full list of supported genomes by typing <code><a href="https://www.rdocumentation.org/packages/BSgenome/topics/available.genomes">BSgenome::available.genomes()</a></code> or create a custom <em>BSgenome</em> object following the <a href="https://www.bioconductor.org/packages//2.7/bioc/vignettes/BSgenome/inst/doc/BSgenomeForge.pdf">instructions</a>.</p>
<p>Since we are working with expression data from human cancer cell lines and mouse embryos, we will first install the UCSC hg38 and mm10 assemblies.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">BiocManager<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/BiocManager/topics/install">install</a></span>(<span class="st">"BSgenome.Hsapiens.UCSC.hg38"</span>)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">BiocManager<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/BiocManager/topics/install">install</a></span>(<span class="st">"BSgenome.Mmusculus.UCSC.mm10"</span>)</a></code></pre></div>
<p><strong>Important</strong> <strong>note</strong>: Do not use <strong>repeat-masked</strong> <em>BSgenome</em> objects (contain ‘masked’ suffix, e.g. <em>BSgenome.Hsapiens.UCSC.hg38.masked</em>)!</p>
</div>
<div id="mafft-multiple-alignment-program-optional" class="section level2">
<h2 class="hasAnchor">
<a href="#mafft-multiple-alignment-program-optional" class="anchor"></a>MAFFT multiple alignment program (optional)</h2>
<p>To compute family-wise multiple sequence alignments (used to improve mapping of read/UMI signal along consensus TE models), Repsc utilizes the <a href="https://mafft.cbrc.jp/alignment/software/">MAFFT multiple alignment program</a>. To use this feature, make sure mafft is in your command PATH.</p>
</div>
<div id="repeat-annotation-required" class="section level2">
<h2 class="hasAnchor">
<a href="#repeat-annotation-required" class="anchor"></a>Repeat annotation (required)</h2>
<p>A convient solution is to download transposon coordinates from the <a href="http://www.repeatmasker.org/genomicDatasets/RMGenomicDatasets.html">Repeatmasker homepage</a> or the <a href="http://www.dfam.org/releases/Dfam_3.0/annotations/">DFAM database</a> for your genome and assembly of choice. You can import the Repeatmasker <em>fa.out.gz</em> or DFAM <em>dfam.hits.gz</em> files using the Repsc <code>importRMSK</code> and <code>importDFAM</code> functions, respectively. Another option is to provide custom annotation as long as it provides the basic information about chromosome, start, end, strand, repname (family identifier), and id_unique (unique locus identifier). In this tutorial, we will show you how to import such information using the provided example datasets.</p>
</div>
<div id="gene-model-annotation-required" class="section level2">
<h2 class="hasAnchor">
<a href="#gene-model-annotation-required" class="anchor"></a>Gene model annotation (required)</h2>
<p>In addition to TE expression counts, Repsc also quantifies genic expression levels using common gene interval and annotation formats (e.g. gtf). In this tutorial, we will use the <a href="https://www.gencodegenes.org/">Gencode</a> comprehensive gene annotation on the human reference chromosomes only. Other annotation ressources are currently untested and should be used with caution.</p>
</div>
<div id="readumi-genomic-coordinates-required" class="section level2">
<h2 class="hasAnchor">
<a href="#readumi-genomic-coordinates-required" class="anchor"></a>Read/UMI genomic coordinates (required)</h2>
<p>Repsc requires read alignment coordinates stored in <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">BAM format</a> as input, which are routinely generated during most common scRNA-seq workflows, including <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger">10x’ Cellranger pipeline</a>. BAM inputs should be duplicate removed, position sorted, and indexed. Chunking BAM inputs (e.g. by chromosome) can accelerate the import into your R environment using the <code>importBAM</code> function. <strong>Important:</strong> Repsc assumes the cell barcode is either stored as CB tag or BAM input files are seperated per cell. Other formats are currently not supported (e.g. cell barcode in read name).</p>
</div>
</div>
<div id="workflow-human-10x-scrna-seq-dataset-5" class="section level1">
<h1 class="hasAnchor">
<a href="#workflow-human-10x-scrna-seq-dataset-5" class="anchor"></a>Workflow human 10x scRNA-seq dataset (5’)</h1>
<p>In this tutorial, we are going to utilize 5’ scRNA-seq data on epigenetically de-repressed cancer cell lines to quantify transposable element (TE) expression levels at single-cell and locus resolution. Following the workflow, you’ll learn the specifics of Repsc to adapt it to your single-cell dataset.</p>
<div id="getting-started" class="section level2">
<h2 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h2>
<p>We start the workflow by loading Repsc and the human hg38 <em>BSgenome</em> object into our R environment:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/load_all">load_all</a></span>(<span class="st">'/net/mraid14/export/tgdata/users/davidbr/src/Repsc/'</span>)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"></a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="co"># adjust to your genome of interest (e.g. BSgenome.Mmusculus.UCSC.mm10)</span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(BSgenome.Hsapiens.UCSC.hg38)</a></code></pre></div>
</div>
<div id="create-scset" class="section level2">
<h2 class="hasAnchor">
<a href="#create-scset" class="anchor"></a>Create scSet</h2>
<p>We then import our gene and TE annotation files as <a href="https://bioconductor.org/packages/release/bioc/vignettes/GenomicRanges/inst/doc/GenomicRangesIntroduction.html">GRanges objects</a> followed by Repsc-specific curation and formatting using the <code>curateGenes</code> and <code>curateTEs</code> functions.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co"># path to Gencode gtf file (provided)</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">gene_path &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="dt">package =</span> <span class="st">'Repsc'</span>, </a>
<a class="sourceLine" id="cb3-3" data-line-number="3">                        <span class="st">'extdata'</span>, </a>
<a class="sourceLine" id="cb3-4" data-line-number="4">                        <span class="st">'gencode.v29.annotation.gtf.gz'</span>)</a>
<a class="sourceLine" id="cb3-5" data-line-number="5"></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="co"># path to RepeatMasker hg38 repeat annotation (provided)</span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7">rmsk_path &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="dt">package =</span> <span class="st">'Repsc'</span>, </a>
<a class="sourceLine" id="cb3-8" data-line-number="8">                         <span class="st">'extdata'</span>, </a>
<a class="sourceLine" id="cb3-9" data-line-number="9">                         <span class="st">'hg38.fa.out.gz'</span>)</a>
<a class="sourceLine" id="cb3-10" data-line-number="10">                         </a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co"># creating the scSet</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12">sc &lt;-<span class="st"> </span><span class="kw">createScSet</span>(<span class="dt">genome   =</span> Hsapiens,</a>
<a class="sourceLine" id="cb3-13" data-line-number="13">                  <span class="dt">protocol =</span> <span class="st">'fiveprime'</span>,</a>
<a class="sourceLine" id="cb3-14" data-line-number="14">                  <span class="dt">tes      =</span> rmsk_path,</a>
<a class="sourceLine" id="cb3-15" data-line-number="15">                  <span class="dt">genes    =</span> gene_path)</a></code></pre></div>
</div>
<div id="compute-multiple-sequence-alignments" class="section level2">
<h2 class="hasAnchor">
<a href="#compute-multiple-sequence-alignments" class="anchor"></a>Compute multiple sequence alignments</h2>
<p>Repsc computes the read/UMI coverage along genes and the consensus model of TE families. This can be useful to sanity check 5’/3’ enrichment (depending on the protocol) and to identify putative TE consensus TSSs (5’ protocols), polyA-sites (3’ protocols), and to distinguish true de-repression from spurious background signal (e.g. intronic TE read mis-assignment, broad-scale genomic background transcription, etc.). As a rough estimate, we can utilize the consensus mapping information from Repeatmasker or DFAM output files for that purpose. This will usually provide reasonable results for highly conserved families. To increase accuracy, Repsc can also compute family-wise multiple sequence alignments to improve mapping of individual loci onto a de novo alignment. When time and computational ressources are no limitation, we recommend this step by running:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="kw">compAlignments</span>(sc, </a>
<a class="sourceLine" id="cb4-3" data-line-number="3">               <span class="dt">max_gap   =</span> <span class="fl">0.95</span>,       <span class="co"># max gap percentage in the alignment to consider</span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4">               <span class="dt">n_jobs    =</span> <span class="dv">500</span>,        <span class="co"># number of parallel jobs (needs HPC access)</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5">               <span class="dt">outdir    =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/tempfile">tempdir</a></span>(),      </a>
<a class="sourceLine" id="cb4-6" data-line-number="6">               <span class="dt">overwrite =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb4-7" data-line-number="7"></a>
<a class="sourceLine" id="cb4-8" data-line-number="8">## Compute count matrix</a>
<a class="sourceLine" id="cb4-9" data-line-number="9"></a>
<a class="sourceLine" id="cb4-10" data-line-number="10">After we have imported and curated the data, we can procede to generate the actual read<span class="op">/</span>UMI count matrix using the <span class="st">`</span><span class="dt">compCounts</span><span class="st">`</span> function.</a>
<a class="sourceLine" id="cb4-11" data-line-number="11"> </a></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="co"># path to bam files containing mapped reads                         </span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2">bam_paths &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list.files">dir</a></span>(<span class="st">'/net/mraid14/export/data/users/davidbr/proj/epitherapy/data/h1299/10x/dacsb/aligned/chunked_bams/'</span>,</a>
<a class="sourceLine" id="cb5-3" data-line-number="3">                 <span class="dt">pattern =</span> <span class="st">'_deduplicated_chr[0-9, X, Y, MT]+.bam'</span>, </a>
<a class="sourceLine" id="cb5-4" data-line-number="4">                 <span class="dt">full.names =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb5-5" data-line-number="5">                 </a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="co"># create a data.frame specifying import parameters                 </span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7">bam_df    &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="dt">paths   =</span> bam_paths,</a>
<a class="sourceLine" id="cb5-8" data-line-number="8">                        <span class="dt">paired  =</span> <span class="ot">TRUE</span>,       <span class="co"># use FALSE for single-end libraries</span></a>
<a class="sourceLine" id="cb5-9" data-line-number="9">                        <span class="dt">mate    =</span> <span class="st">'first'</span>,    <span class="co"># only imports the first mate of properly aligned read pairs, set to NA when using single-end libraries</span></a>
<a class="sourceLine" id="cb5-10" data-line-number="10">                        <span class="dt">barcode =</span> <span class="st">'CB'</span>,       <span class="co"># 10x barcode included in BAM flag</span></a>
<a class="sourceLine" id="cb5-11" data-line-number="11">                        <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb5-12" data-line-number="12">sc &lt;-<span class="st"> </span><span class="kw">addCounts</span>(sc,</a>
<a class="sourceLine" id="cb5-13" data-line-number="13">                <span class="dt">reads    =</span> bam_df,</a>
<a class="sourceLine" id="cb5-14" data-line-number="14">                <span class="dt">bin_size =</span> <span class="dv">25</span>,</a>
<a class="sourceLine" id="cb5-15" data-line-number="15">                <span class="dt">msa_dir  =</span> <span class="ot">NULL</span>)</a></code></pre></div>
</div>
<div id="call-cells" class="section level2">
<h2 class="hasAnchor">
<a href="#call-cells" class="anchor"></a>Call cells</h2>
<p>To distinguish real cells from empty droplets, we utilize the <code>emptyDrops</code> function from the <a href="https://rdrr.io/github/MarioniLab/DropletUtils/">DropletUtils</a> package[1].</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">sc &lt;-<span class="st"> </span><span class="kw">callCells</span>(sc, </a>
<a class="sourceLine" id="cb6-2" data-line-number="2">                <span class="dt">fdr =</span> <span class="fl">0.01</span>)</a></code></pre></div>
</div>
<div id="call-tsss" class="section level2">
<h2 class="hasAnchor">
<a href="#call-tsss" class="anchor"></a>Call TSSs</h2>
</div>
</div>
<div id="workflow-mouse-smart-seq2-scrna-seq-dataset" class="section level1">
<h1 class="hasAnchor">
<a href="#workflow-mouse-smart-seq2-scrna-seq-dataset" class="anchor"></a>Workflow mouse Smart-seq2 scRNA-seq dataset</h1>
<p>In this part of the tutorial, we will use Smart-seq2 data generated on early mouse embryos[1] to illustrate the specifics of full-coverage scRNA-seq data.</p>
<div id="getting-started-1" class="section level2">
<h2 class="hasAnchor">
<a href="#getting-started-1" class="anchor"></a>Getting started</h2>
<p>We start the workflow by loading Repsc and the mouse mm10 <em>BSgenome</em> object into our R environment:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/load_all">load_all</a></span>(<span class="st">'/net/mraid14/export/tgdata/users/davidbr/src/Repsc/'</span>)</a>
<a class="sourceLine" id="cb7-2" data-line-number="2"></a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="co"># adjust to your genome of interest (e.g. BSgenome.Hsapiens.UCSC.hg38)</span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(BSgenome.Mmusculus.UCSC.mm10)</a></code></pre></div>
</div>
<div id="import-genome-annotation" class="section level2">
<h2 class="hasAnchor">
<a href="#import-genome-annotation" class="anchor"></a>Import genome annotation</h2>
<p>As before, we first import our gene and TE annotation files as <a href="https://bioconductor.org/packages/release/bioc/vignettes/GenomicRanges/inst/doc/GenomicRangesIntroduction.html">GRanges objects</a> followed by Repsc-specific curation and formatting using the <code>curateGenes</code> and <code>curateTEs</code> functions.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="co"># path to Gencode gtf file (provided)</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2">gene_path &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="dt">package =</span> <span class="st">'Repsc'</span>, </a>
<a class="sourceLine" id="cb8-3" data-line-number="3">                        <span class="st">'extdata'</span>, </a>
<a class="sourceLine" id="cb8-4" data-line-number="4">                        <span class="st">'gencode.vM21.annotation.gtf.gz'</span>)</a>
<a class="sourceLine" id="cb8-5" data-line-number="5"><span class="co"># path to RepeatMasker hg38 repeat annotation (provided)</span></a>
<a class="sourceLine" id="cb8-6" data-line-number="6">rmsk_path &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="dt">package =</span> <span class="st">'Repsc'</span>, </a>
<a class="sourceLine" id="cb8-7" data-line-number="7">                         <span class="st">'extdata'</span>, </a>
<a class="sourceLine" id="cb8-8" data-line-number="8">                         <span class="st">'mm10.fa.out.gz'</span>)</a>
<a class="sourceLine" id="cb8-9" data-line-number="9">                            </a>
<a class="sourceLine" id="cb8-10" data-line-number="10"><span class="co"># import</span></a>
<a class="sourceLine" id="cb8-11" data-line-number="11">genes &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/rtracklayer/topics/export">import</a></span>(gene_path)</a>
<a class="sourceLine" id="cb8-12" data-line-number="12">tes   &lt;-<span class="st"> </span><span class="kw">importRMSK</span>(rmsk_path)         <span class="co"># by default only imports DNA, SINE, LINE, and LTR class repeats on the main reference chromosomes                           </span></a>
<a class="sourceLine" id="cb8-13" data-line-number="13"></a>
<a class="sourceLine" id="cb8-14" data-line-number="14"><span class="co"># curation</span></a>
<a class="sourceLine" id="cb8-15" data-line-number="15">genes_cur &lt;-<span class="st"> </span><span class="kw">curateGenes</span>(genes)        <span class="co"># only retains exons, resolves naming ambiguity</span></a>
<a class="sourceLine" id="cb8-16" data-line-number="16">tes_cur   &lt;-<span class="st"> </span><span class="kw">curateTEs</span>(tes,            <span class="co"># tries to resolve nested TE intervals and adds annotation</span></a>
<a class="sourceLine" id="cb8-17" data-line-number="17">                       genes_cur)      <span class="co"># to filter exonic TEs</span></a></code></pre></div>
</div>
<div id="import-reads-and-compute-count-matrix-in-parallel" class="section level2">
<h2 class="hasAnchor">
<a href="#import-reads-and-compute-count-matrix-in-parallel" class="anchor"></a>Import reads and compute count matrix in parallel</h2>
<p>Given the size of the example Smart-seq2 dataset and memory limitations of most systems, we will compute single-cell gene/TE counts for subsets of the BAM files and combine the processed results afterwards. For enhanced speed performance, we will employ the parallel backend of the <a href="https://github.com/HenrikBengtsson/future">future package</a>.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="co"># path to bam files containing mapped reads                         </span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2">bam_paths &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list.files">dir</a></span>(<span class="st">'/net/mraid14/export/tgdata/users/evghenic/data/cheng2019/aligned/'</span>, </a>
<a class="sourceLine" id="cb9-3" data-line-number="3">                 <span class="dt">pattern =</span> <span class="st">'duprm.bam'</span>, </a>
<a class="sourceLine" id="cb9-4" data-line-number="4">                 <span class="dt">full.names =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb9-5" data-line-number="5">bam_paths</a>
<a class="sourceLine" id="cb9-6" data-line-number="6"></a>
<a class="sourceLine" id="cb9-7" data-line-number="7"><span class="co"># vector of cell barcodes </span></a>
<a class="sourceLine" id="cb9-8" data-line-number="8">barcodes &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/substr">substring</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/basename">basename</a></span>(bam_paths), <span class="dv">1</span>, <span class="dv">10</span>)</a>
<a class="sourceLine" id="cb9-9" data-line-number="9"></a>
<a class="sourceLine" id="cb9-10" data-line-number="10"><span class="co"># commands for parallel execution</span></a>
<a class="sourceLine" id="cb9-11" data-line-number="11">cmds   &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>()</a>
<a class="sourceLine" id="cb9-12" data-line-number="12">chunks &lt;-<span class="st"> </span><span class="kw">chunk</span>(bam_paths, <span class="dt">chunk_size =</span> <span class="dv">10</span>)</a>
<a class="sourceLine" id="cb9-13" data-line-number="13"></a>
<a class="sourceLine" id="cb9-14" data-line-number="14"><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(chunks))</a>
<a class="sourceLine" id="cb9-15" data-line-number="15">{</a>
<a class="sourceLine" id="cb9-16" data-line-number="16">  bam_chunks     &lt;-<span class="st"> </span><span class="kw">glue</span>(<span class="st">"c({glue_collapse(single_quote(bam_paths[chunks[[i]]]), sep = ',')})"</span>)</a>
<a class="sourceLine" id="cb9-17" data-line-number="17">  barcode_chunks &lt;-<span class="st"> </span><span class="kw">glue</span>(<span class="st">"c({glue_collapse(single_quote(barcodes[chunks[[i]]]), sep = ',')})"</span>)</a>
<a class="sourceLine" id="cb9-18" data-line-number="18"> </a>
<a class="sourceLine" id="cb9-19" data-line-number="19">  cmds[[i]] &lt;-<span class="st"> </span><span class="kw">glue</span>(<span class="st">"Repsc:::compCounts(reads = Repsc:::importBAM({bam_chunks}, paired = FALSE, mate = NA, anchor = 'fiveprime', barcode = {barcode_chunks}),</span></a>
<a class="sourceLine" id="cb9-20" data-line-number="20"><span class="st">                     tes_cur,</span></a>
<a class="sourceLine" id="cb9-21" data-line-number="21"><span class="st">                     genes_cur,</span></a>
<a class="sourceLine" id="cb9-22" data-line-number="22"><span class="st">                     bin_size = 20,</span></a>
<a class="sourceLine" id="cb9-23" data-line-number="23"><span class="st">                     protocol = 'full')"</span>)</a>
<a class="sourceLine" id="cb9-24" data-line-number="24">}</a>
<a class="sourceLine" id="cb9-25" data-line-number="25"></a>
<a class="sourceLine" id="cb9-26" data-line-number="26"><span class="co"># distribute commands</span></a>
<a class="sourceLine" id="cb9-27" data-line-number="27"></a>
<a class="sourceLine" id="cb9-28" data-line-number="28"><span class="co"># create new empty env and fill with relevant</span></a>
<a class="sourceLine" id="cb9-29" data-line-number="29">empty           &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/environment">new.env</a></span>(<span class="dt">parent =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/environment">emptyenv</a></span>())</a>
<a class="sourceLine" id="cb9-30" data-line-number="30">empty<span class="op">$</span>tes_cur   &lt;-<span class="st"> </span>tes_cur</a>
<a class="sourceLine" id="cb9-31" data-line-number="31">empty<span class="op">$</span>genes_cur &lt;-<span class="st"> </span>genes_cur</a>
<a class="sourceLine" id="cb9-32" data-line-number="32">  </a>
<a class="sourceLine" id="cb9-33" data-line-number="33"><span class="co"># distribute, compute and combine</span></a>
<a class="sourceLine" id="cb9-34" data-line-number="34">counts &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb9-35" data-line-number="35"><span class="st">  </span><span class="kw">gcluster.run3</span>(<span class="dt">command_list =</span> cmds,  </a>
<a class="sourceLine" id="cb9-36" data-line-number="36">                <span class="dt">packages =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"data.table"</span>, <span class="st">"plyranges"</span>, <span class="st">"base"</span>, <span class="st">"stats"</span>), </a>
<a class="sourceLine" id="cb9-37" data-line-number="37">                <span class="dt">envir =</span> empty, </a>
<a class="sourceLine" id="cb9-38" data-line-number="38">                <span class="dt">collapse_results =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb9-39" data-line-number="39">                <span class="dt">io_saturation =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
</div>
<div id="normalize-counts" class="section level2">
<h2 class="hasAnchor">
<a href="#normalize-counts" class="anchor"></a>Normalize counts</h2>
<p>The full transcript coverage of Smart-seq2 comes with a cost: Unlike most scRNA-seq protocols it lacks UMIs to quantify true molecule counts per cell. As an approximation, Repsc samples <code>N</code> molecules per cell with empirical weights learned from the data to simulate integer molecule counts.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">counts_norm &lt;-<span class="st"> </span><span class="kw">normCounts</span>(counts, </a>
<a class="sourceLine" id="cb10-2" data-line-number="2">                          <span class="dt">N =</span> <span class="dv">20000</span>) <span class="co"># number of molecules to sample per cell</span></a>
<a class="sourceLine" id="cb10-3" data-line-number="3"></a>
<a class="sourceLine" id="cb10-4" data-line-number="4"><span class="co"># Workflow mouse 10x scRNA-seq dataset (3')</span></a>
<a class="sourceLine" id="cb10-5" data-line-number="5"></a>
<a class="sourceLine" id="cb10-6" data-line-number="6">In this part of the tutorial, we will use <span class="dv">3</span><span class="st">' scRNA-seq data generated on mouse embryos[1] to illustrate the specifics of 3'</span> scRNA<span class="op">-</span>seq data.</a>
<a class="sourceLine" id="cb10-7" data-line-number="7"></a>
<a class="sourceLine" id="cb10-8" data-line-number="8">## Getting started</a>
<a class="sourceLine" id="cb10-9" data-line-number="9"></a>
<a class="sourceLine" id="cb10-10" data-line-number="10">We start the workflow by loading Repsc and the mouse mm10 _BSgenome_ object into our R environment<span class="op">:</span></a></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/load_all">load_all</a></span>(<span class="st">'/net/mraid14/export/tgdata/users/davidbr/src/Repsc/'</span>)</a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(BSgenome.Mmusculus.UCSC.mm10)</a></code></pre></div>
</div>
<div id="import-genome-annotation-1" class="section level2">
<h2 class="hasAnchor">
<a href="#import-genome-annotation-1" class="anchor"></a>Import genome annotation</h2>
<p>As before, we first import our gene and TE annotation files as <a href="https://bioconductor.org/packages/release/bioc/vignettes/GenomicRanges/inst/doc/GenomicRangesIntroduction.html">GRanges objects</a> followed by Repsc-specific curation and formatting using the <code>curateGenes</code> and <code>curateTEs</code> functions.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="co"># path to Gencode gtf file (provided)</span></a>
<a class="sourceLine" id="cb12-2" data-line-number="2">gene_path &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="dt">package =</span> <span class="st">'Repsc'</span>, </a>
<a class="sourceLine" id="cb12-3" data-line-number="3">                        <span class="st">'extdata'</span>, </a>
<a class="sourceLine" id="cb12-4" data-line-number="4">                        <span class="st">'gencode.vM21.annotation.gtf.gz'</span>)</a>
<a class="sourceLine" id="cb12-5" data-line-number="5"><span class="co"># path to RepeatMasker hg38 repeat annotation (provided)</span></a>
<a class="sourceLine" id="cb12-6" data-line-number="6">rmsk_path &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="dt">package =</span> <span class="st">'Repsc'</span>, </a>
<a class="sourceLine" id="cb12-7" data-line-number="7">                         <span class="st">'extdata'</span>, </a>
<a class="sourceLine" id="cb12-8" data-line-number="8">                         <span class="st">'mm10.fa.out.gz'</span>)</a>
<a class="sourceLine" id="cb12-9" data-line-number="9">                            </a>
<a class="sourceLine" id="cb12-10" data-line-number="10"><span class="co"># import</span></a>
<a class="sourceLine" id="cb12-11" data-line-number="11">genes &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/rtracklayer/topics/export">import</a></span>(gene_path)</a>
<a class="sourceLine" id="cb12-12" data-line-number="12">tes   &lt;-<span class="st"> </span><span class="kw">importRMSK</span>(rmsk_path)         <span class="co"># by default only imports DNA, SINE, LINE, and LTR class repeats on the main reference chromosomes                           </span></a>
<a class="sourceLine" id="cb12-13" data-line-number="13"></a>
<a class="sourceLine" id="cb12-14" data-line-number="14"><span class="co"># curation</span></a>
<a class="sourceLine" id="cb12-15" data-line-number="15">genes_cur &lt;-<span class="st"> </span><span class="kw">curateGenes</span>(genes)        <span class="co"># only retains exons, resolves naming ambiguity</span></a>
<a class="sourceLine" id="cb12-16" data-line-number="16">tes_cur   &lt;-<span class="st"> </span><span class="kw">curateTEs</span>(tes,            <span class="co"># tries to resolve nested TE intervals and adds annotation</span></a>
<a class="sourceLine" id="cb12-17" data-line-number="17">                       genes_cur)      <span class="co"># to filter exonic TEs</span></a></code></pre></div>
</div>
<div id="import-reads-and-compute-count-matrix-in-parallel-1" class="section level2">
<h2 class="hasAnchor">
<a href="#import-reads-and-compute-count-matrix-in-parallel-1" class="anchor"></a>Import reads and compute count matrix in parallel</h2>
<p>Given the size of the example Smart-seq2 dataset and memory limitations of most systems, we will compute single-cell gene/TE counts for subsets of the BAM files and combine the processed results afterwards. For enhanced speed performance, we will employ the parallel backend of the <a href="https://github.com/HenrikBengtsson/future">future package</a>.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="co"># path to bam files containing mapped reads                         </span></a>
<a class="sourceLine" id="cb13-2" data-line-number="2">bam_paths &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list.files">dir</a></span>(<span class="st">'/net/mraid14/export/tgdata/users/davidbr/data/goettgens_2019/aligned/'</span>, </a>
<a class="sourceLine" id="cb13-3" data-line-number="3">                 <span class="dt">pattern =</span> <span class="st">'bam.bam$'</span>,</a>
<a class="sourceLine" id="cb13-4" data-line-number="4">                 <span class="dt">full.names =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb13-5" data-line-number="5">bam_paths</a>
<a class="sourceLine" id="cb13-6" data-line-number="6"></a>
<a class="sourceLine" id="cb13-7" data-line-number="7"><span class="co"># commands for parallel execution</span></a>
<a class="sourceLine" id="cb13-8" data-line-number="8">cmds   &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>()</a>
<a class="sourceLine" id="cb13-9" data-line-number="9">chunks &lt;-<span class="st"> </span><span class="kw">chunk</span>(bam_paths, <span class="dt">chunk_size =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb13-10" data-line-number="10"></a>
<a class="sourceLine" id="cb13-11" data-line-number="11"><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(chunks))</a>
<a class="sourceLine" id="cb13-12" data-line-number="12">{</a>
<a class="sourceLine" id="cb13-13" data-line-number="13">  bam_chunks     &lt;-<span class="st"> </span><span class="kw">glue</span>(<span class="st">"c({glue_collapse(single_quote(bam_paths[chunks[[i]]]), sep = ',')})"</span>)</a>
<a class="sourceLine" id="cb13-14" data-line-number="14"> </a>
<a class="sourceLine" id="cb13-15" data-line-number="15">  cmds[[i]] &lt;-<span class="st"> </span><span class="kw">glue</span>(<span class="st">"Repsc:::compCounts(reads = Repsc:::importBAM({bam_chunks}, paired = FALSE, mate = NA, anchor = 'fiveprime'),</span></a>
<a class="sourceLine" id="cb13-16" data-line-number="16"><span class="st">                     tes_cur,</span></a>
<a class="sourceLine" id="cb13-17" data-line-number="17"><span class="st">                     genes_cur,</span></a>
<a class="sourceLine" id="cb13-18" data-line-number="18"><span class="st">                     bin_size = 20,</span></a>
<a class="sourceLine" id="cb13-19" data-line-number="19"><span class="st">                     protocol = 'full')"</span>)</a>
<a class="sourceLine" id="cb13-20" data-line-number="20">}</a>
<a class="sourceLine" id="cb13-21" data-line-number="21"></a>
<a class="sourceLine" id="cb13-22" data-line-number="22"><span class="co"># distribute commands</span></a>
<a class="sourceLine" id="cb13-23" data-line-number="23"></a>
<a class="sourceLine" id="cb13-24" data-line-number="24"><span class="co"># create new empty env and fill with relevant</span></a>
<a class="sourceLine" id="cb13-25" data-line-number="25">empty           &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/environment">new.env</a></span>(<span class="dt">parent =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/environment">emptyenv</a></span>())</a>
<a class="sourceLine" id="cb13-26" data-line-number="26">empty<span class="op">$</span>tes_cur   &lt;-<span class="st"> </span>tes_cur</a>
<a class="sourceLine" id="cb13-27" data-line-number="27">empty<span class="op">$</span>genes_cur &lt;-<span class="st"> </span>genes_cur</a>
<a class="sourceLine" id="cb13-28" data-line-number="28">  </a>
<a class="sourceLine" id="cb13-29" data-line-number="29"><span class="co"># distribute, compute and combine</span></a>
<a class="sourceLine" id="cb13-30" data-line-number="30">res_gclust &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb13-31" data-line-number="31"><span class="st">  </span><span class="kw">gcluster.run3</span>(<span class="dt">command_list =</span> cmds,  </a>
<a class="sourceLine" id="cb13-32" data-line-number="32">                <span class="dt">packages =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"data.table"</span>, <span class="st">"plyranges"</span>, <span class="st">"base"</span>, <span class="st">"stats"</span>), </a>
<a class="sourceLine" id="cb13-33" data-line-number="33">                <span class="dt">envir =</span> empty, </a>
<a class="sourceLine" id="cb13-34" data-line-number="34">                <span class="dt">collapse_results =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb13-35" data-line-number="35">                <span class="dt">io_saturation =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
</div>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<p>[1]</p>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/sessionInfo">sessionInfo</a></span>()</a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="co">#&gt; R version 3.5.3 (2019-03-11)</span></a>
<a class="sourceLine" id="cb14-3" data-line-number="3"><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></a>
<a class="sourceLine" id="cb14-4" data-line-number="4"><span class="co">#&gt; Running under: CentOS Linux 7 (Core)</span></a>
<a class="sourceLine" id="cb14-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb14-6" data-line-number="6"><span class="co">#&gt; Matrix products: default</span></a>
<a class="sourceLine" id="cb14-7" data-line-number="7"><span class="co">#&gt; BLAS/LAPACK: /net/mraid14/export/data/users/eladch/tools/CO7/mkl/2018.3/compilers_and_libraries_2018.3.222/linux/mkl/lib/intel64_lin/libmkl_gf_lp64.so</span></a>
<a class="sourceLine" id="cb14-8" data-line-number="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb14-9" data-line-number="9"><span class="co">#&gt; locale:</span></a>
<a class="sourceLine" id="cb14-10" data-line-number="10"><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></a>
<a class="sourceLine" id="cb14-11" data-line-number="11"><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></a>
<a class="sourceLine" id="cb14-12" data-line-number="12"><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></a>
<a class="sourceLine" id="cb14-13" data-line-number="13"><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></a>
<a class="sourceLine" id="cb14-14" data-line-number="14"><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></a>
<a class="sourceLine" id="cb14-15" data-line-number="15"><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></a>
<a class="sourceLine" id="cb14-16" data-line-number="16"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb14-17" data-line-number="17"><span class="co">#&gt; attached base packages:</span></a>
<a class="sourceLine" id="cb14-18" data-line-number="18"><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></a>
<a class="sourceLine" id="cb14-19" data-line-number="19"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb14-20" data-line-number="20"><span class="co">#&gt; loaded via a namespace (and not attached):</span></a>
<a class="sourceLine" id="cb14-21" data-line-number="21"><span class="co">#&gt;  [1] Rcpp_1.0.1       roxygen2_6.1.1   rprojroot_1.3-2  crayon_1.3.4    </span></a>
<a class="sourceLine" id="cb14-22" data-line-number="22"><span class="co">#&gt;  [5] assertthat_0.2.1 digest_0.6.19    commonmark_1.7   MASS_7.3-51.1   </span></a>
<a class="sourceLine" id="cb14-23" data-line-number="23"><span class="co">#&gt;  [9] R6_2.4.0         backports_1.1.4  magrittr_1.5     evaluate_0.13   </span></a>
<a class="sourceLine" id="cb14-24" data-line-number="24"><span class="co">#&gt; [13] stringi_1.4.3    rlang_0.3.2      fs_1.2.7         xml2_1.2.0      </span></a>
<a class="sourceLine" id="cb14-25" data-line-number="25"><span class="co">#&gt; [17] rmarkdown_1.12   pkgdown_1.3.0    desc_1.2.0       tools_3.5.3     </span></a>
<a class="sourceLine" id="cb14-26" data-line-number="26"><span class="co">#&gt; [21] stringr_1.4.0    yaml_2.2.0       xfun_0.5         compiler_3.5.3  </span></a>
<a class="sourceLine" id="cb14-27" data-line-number="27"><span class="co">#&gt; [25] memoise_1.1.0    htmltools_0.3.6  knitr_1.22</span></a></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#prerequisites">Prerequisites</a><ul class="nav nav-pills nav-stacked">
<li><a href="#genome-assembly-optional">Genome assembly (optional)</a></li>
      <li><a href="#mafft-multiple-alignment-program-optional">MAFFT multiple alignment program (optional)</a></li>
      <li><a href="#repeat-annotation-required">Repeat annotation (required)</a></li>
      <li><a href="#gene-model-annotation-required">Gene model annotation (required)</a></li>
      <li><a href="#readumi-genomic-coordinates-required">Read/UMI genomic coordinates (required)</a></li>
      </ul>
</li>
      <li>
<a href="#workflow-human-10x-scrna-seq-dataset-5">Workflow human 10x scRNA-seq dataset (5’)</a><ul class="nav nav-pills nav-stacked">
<li><a href="#getting-started">Getting started</a></li>
      <li><a href="#create-scset">Create scSet</a></li>
      <li><a href="#compute-multiple-sequence-alignments">Compute multiple sequence alignments</a></li>
      <li><a href="#call-cells">Call cells</a></li>
      <li><a href="#call-tsss">Call TSSs</a></li>
      </ul>
</li>
      <li>
<a href="#workflow-mouse-smart-seq2-scrna-seq-dataset">Workflow mouse Smart-seq2 scRNA-seq dataset</a><ul class="nav nav-pills nav-stacked">
<li><a href="#getting-started-1">Getting started</a></li>
      <li><a href="#import-genome-annotation">Import genome annotation</a></li>
      <li><a href="#import-reads-and-compute-count-matrix-in-parallel">Import reads and compute count matrix in parallel</a></li>
      <li><a href="#normalize-counts">Normalize counts</a></li>
      <li><a href="#import-genome-annotation-1">Import genome annotation</a></li>
      <li><a href="#import-reads-and-compute-count-matrix-in-parallel-1">Import reads and compute count matrix in parallel</a></li>
      </ul>
</li>
      <li><a href="#references">References</a></li>
      <li><a href="#session-information">Session information</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by <a href="https://scholar.google.de/citations?user=ossdDj4AAAAJ&amp;hl=en&amp;oi=ao">David Brocks</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
