<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>plots hic normalized map with annotations — shaman_plot_map_score_with_annotations • shaman</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">


<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>
  
  
<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">shaman</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">Home</a>
</li>
<li>
  <a href="../articles/shaman-package.html">
    <span class="fa fa-road"></span>
     
    Usage
  </a>
</li>
<li>
  <a href="../articles/import.html">
    <span class="fa fa-filter"></span>
     
    Import
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-th-list"></span>
     
    Reference
  </a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://bitbucket.org/tanaylab/shaman">
    <span class="fa fa-bitbucket-square"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

      <div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>plots hic normalized map with annotations</h1>
    </div>

    
    <p><code>shaman_plot_map_score_with_annotations</code></p>
    

    <pre class="usage"><span class='fu'>shaman_plot_map_score_with_annotations</span>(<span class='no'>genome</span>, <span class='no'>points_score</span>, <span class='no'>interval_range</span>,
  <span class='kw'>point_size</span> <span class='kw'>=</span> <span class='fl'>0.1</span>, <span class='kw'>misha_tracks</span> <span class='kw'>=</span> <span class='fu'>list</span>(),
  <span class='kw'>mt_colors</span> <span class='kw'>=</span> <span class='fu'>getOption</span>(<span class='st'>"shaman.track_colors"</span>), <span class='kw'>mt_ylims</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>annotations</span> <span class='kw'>=</span> <span class='fu'>list</span>(), <span class='kw'>a_colors</span> <span class='kw'>=</span> <span class='fu'>getOption</span>(<span class='st'>"shaman.annotation_colors"</span>),
  <span class='kw'>add_genes</span> <span class='kw'>=</span> <span class='no'>T</span>, <span class='kw'>add_ideogram</span> <span class='kw'>=</span> <span class='no'>T</span>, <span class='kw'>add_axis</span> <span class='kw'>=</span> <span class='no'>T</span>,
  <span class='kw'>gene_stacking</span> <span class='kw'>=</span> <span class='st'>"squish"</span>, <span class='kw'>gene_size</span> <span class='kw'>=</span> <span class='fl'>0.7</span>, <span class='kw'>track_size</span> <span class='kw'>=</span> <span class='fl'>0.8</span>,
  <span class='kw'>annotation_size</span> <span class='kw'>=</span> <span class='fl'>0.7</span>, <span class='kw'>fig_fn</span> <span class='kw'>=</span> <span class='st'>""</span>, <span class='kw'>fig_width</span> <span class='kw'>=</span> <span class='fl'>900</span>, <span class='kw'>fig_height</span> <span class='kw'>=</span> <span class='fl'>5</span>/<span class='fl'>6</span>
  * <span class='fl'>900</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a> Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>genome</th>
      <td><p>Name of reference genome (e.g. "hg19", "mm10")</p></td>
    </tr>
    <tr>
      <th>points_score</th>
      <td><p>A dataframe containing the points (start1, start2) and their normalized score.</p></td>
    </tr>
    <tr>
      <th>interval_range</th>
      <td><p>1D interval (chrom, start, end) specifying the region to plot</p></td>
    </tr>
    <tr>
      <th>misha_tracks</th>
      <td><p>List of 1D track expressions (virtual tracks also supported) which can be extracted from.</p></td>
    </tr>
    <tr>
      <th>mt_colors</th>
      <td><p>Array of colors, one for each misha_track, which will be used to plot each 1d track.</p></td>
    </tr>
    <tr>
      <th>annotations</th>
      <td><p>List of gintervals highlighting annotated regions.</p></td>
    </tr>
    <tr>
      <th>a_colors</th>
      <td><p>Array of colors, one for each annotation set.</p></td>
    </tr>
    <tr>
      <th>add_genes</th>
      <td><p>Boolean flag indicating whether to show gene annotations.</p></td>
    </tr>
    <tr>
      <th>add_ideogram</th>
      <td><p>Boolean flag indicating whether to add an chromosomal ideogram.</p></td>
    </tr>
    <tr>
      <th>add_axis</th>
      <td><p>Boolean flag indicating whether to show chromosmal axis.</p></td>
    </tr>
    <tr>
      <th>gene_stacking</th>
      <td><p>Describes the viewing option of the genes track. Can be either "squish", or "dense".</p></td>
    </tr>
    <tr>
      <th>gene_size</th>
      <td><p>Size of gene annotation view.</p></td>
    </tr>
    <tr>
      <th>track_size</th>
      <td><p>Size of track view.</p></td>
    </tr>
    <tr>
      <th>fig_fn</th>
      <td><p>Name of png file to output to. Empty string will cause the figure to be plotted to the current device.</p></td>
    </tr>
    <tr>
      <th>fig_width</th>
      <td><p>Width in pixels of output png figure.</p></td>
    </tr>
    <tr>
      <th>fig_height</th>
      <td><p>Height in pixels of output png figure.</p></td>
    </tr>
    <tr>
      <th>my_ylims</th>
      <td><p>Y-axis limits for each misha track. If not provided, generated automatically based on the data
to be displayed in the region.</p></td>
    </tr>
    <tr>
      <th>annot_size</th>
      <td><p>Size of annotation view.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Plots observerved hic contact matrix color-coded by normalized scores, aligned with
linear data such as gene annotations, rna, chromatin, transcription factor binding, etc').</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='fu'>gsetroot</span>(<span class='fu'><a href='shaman_get_test_track_db.html'>shaman_get_test_track_db</a></span>())
<span class='no'>points</span> <span class='kw'>&lt;-</span> <span class='fu'>gextract</span>(<span class='st'>"hic_score"</span>, <span class='fu'>gintervals.2d</span>(<span class='fl'>2</span>, <span class='fl'>175e06</span>, <span class='fl'>178e06</span>, <span class='fl'>2</span>, <span class='fl'>175e06</span>, <span class='fl'>178e06</span>), <span class='kw'>colnames</span><span class='kw'>=</span><span class='st'>"score"</span>)
<span class='fu'>shaman_plot_map_score_with_annotations</span>(<span class='st'>"hg19"</span>, <span class='no'>points</span>, <span class='fu'>gintervals</span>(<span class='fl'>2</span>, <span class='fl'>175e06</span>, <span class='fl'>178e06</span>))</div><div class='img'><img src='shaman_plot_map_score_with_annotations-1.png' alt='' width='700' height='432.632880098887' /></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#details">Details</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Amos Tanay, Netta Mendelson Cohen.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  </body>
</html>
