<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Reputils vignette â€¢ Reputils</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Reputils vignette">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Reputils</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="../articles/Prerequisites.html">
    <span class="fa fa-play-circle"></span>
     
    Prerequisites
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Mapping_QC.html">Read alignment QC</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-th-list"></span>
     
    Reference
  </a>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-news"></span>
     
    News
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://bitbucket.org/tanaylab/Reputils/src/default/">
    <span class="fa fa-bitbucket-square"></span>
     
    Bitbucket
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Reputils vignette</h1>
                        <h4 class="author">David Brocks</h4>
            
            <h4 class="date">2019-07-03</h4>
      
      
      <div class="hidden name"><code>Mapping_QC.Rmd</code></div>

    </div>

    
    
<div id="benchmark-read-alignments-to-transposable-elements" class="section level1">
<h1 class="hasAnchor">
<a href="#benchmark-read-alignments-to-transposable-elements" class="anchor"></a>Benchmark read alignments to transposable elements</h1>
<div id="getting-started" class="section level2">
<h2 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h2>
<p>We start the workflow by loading Repsc and the human hg38 <em>BSgenome</em> object into our R environment:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/load_all">load_all</a></span>(<span class="st">'/net/mraid14/export/tgdata/users/davidbr/src/Reputils/'</span>)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/load_all">load_all</a></span>(<span class="st">'/net/mraid14/export/tgdata/users/davidbr/src/Repsc/'</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"></a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(BSgenome.Hsapiens.UCSC.hg38)</a></code></pre></div>
</div>
<div id="introduce-sequence-errors" class="section level2">
<h2 class="hasAnchor">
<a href="#introduce-sequence-errors" class="anchor"></a>Introduce sequence errors</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">bams    &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">grep</a></span>(<span class="st">'deduplicated'</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list.files">dir</a></span>(<span class="st">'~/davidbr/proj/epitherapy/data/h1299/10x/dacsb/aligned/chunked_bams/'</span>, <span class="dt">pattern =</span> <span class="st">'.bam$'</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>), <span class="dt">value =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">tes     &lt;-<span class="st"> </span><span class="kw">importRMSK</span>(<span class="st">'~/tgdata/src/Repsc/inst/extdata/hg38.fa.out.gz'</span>, <span class="dt">curate =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="kw">mutateReads</span>(<span class="dt">BSgenome   =</span> Hsapiens,</a>
<a class="sourceLine" id="cb2-5" data-line-number="5">            <span class="dt">reads      =</span> bams,</a>
<a class="sourceLine" id="cb2-6" data-line-number="6">            <span class="dt">paired     =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb2-7" data-line-number="7">            <span class="dt">tes        =</span> tes,</a>
<a class="sourceLine" id="cb2-8" data-line-number="8">            <span class="dt">n_reads    =</span> <span class="fl">1e6</span>,</a>
<a class="sourceLine" id="cb2-9" data-line-number="9">            <span class="dt">outdir     =</span> <span class="st">'~/davidbr/proj/epitherapy/data/h1299/10x/dacsb/fastqs/'</span>,</a>
<a class="sourceLine" id="cb2-10" data-line-number="10">            <span class="dt">error_rate =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>, <span class="fl">0.25</span>, <span class="fl">0.5</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">4</span>, <span class="dv">8</span>))</a></code></pre></div>
</div>
<div id="star-alignment" class="section level2">
<h2 class="hasAnchor">
<a href="#star-alignment" class="anchor"></a>STAR alignment</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">r1 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list.files">dir</a></span>(<span class="st">'~/davidbr/proj/epitherapy/data/h1299/10x/dacsb/fastqs/'</span>, <span class="dt">pattern =</span> <span class="st">'perc_error_R1'</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">r2 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list.files">dir</a></span>(<span class="st">'~/davidbr/proj/epitherapy/data/h1299/10x/dacsb/fastqs/'</span>, <span class="dt">pattern =</span> <span class="st">'perc_error_R2'</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/getwd">setwd</a></span>(<span class="st">'~/davidbr/proj/epitherapy/data/h1299/10x/dacsb/aligned/'</span>)</a>
<a class="sourceLine" id="cb3-5" data-line-number="5">cmds &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>()</a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(r1))</a>
<a class="sourceLine" id="cb3-7" data-line-number="7">{</a>
<a class="sourceLine" id="cb3-8" data-line-number="8">  cmds[[i]] &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="st">    </span><span class="kw">glue</span>(<span class="st">"system('/net/mraid14/export/data/tools/star/STAR-2.5.1b/source/STAR --runThreadN 10 --genomeDir /net/mraid14/export/data/users/davidbr/tools/cellranger/references/refdata-cellranger-GRCh38-1.2.0/star/ --outSAMtype BAM SortedByCoordinate --outFilterMultimapScoreRange 2 --outFileNamePrefix {gsub('.fastq', '.', basename(r1[i]))} --outSAMunmapped Within --readFilesIn {r1[i]} {r2[i]}')"</span>)</a>
<a class="sourceLine" id="cb3-10" data-line-number="10">}</a>
<a class="sourceLine" id="cb3-11" data-line-number="11"></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="co"># create new empty env and fill with relevant</span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13">empty &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/environment">new.env</a></span>(<span class="dt">parent =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/environment">emptyenv</a></span>())</a>
<a class="sourceLine" id="cb3-14" data-line-number="14"></a>
<a class="sourceLine" id="cb3-15" data-line-number="15"><span class="co"># distribute, compute and combine</span></a>
<a class="sourceLine" id="cb3-16" data-line-number="16">res &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="st">  </span><span class="kw">gcluster.run3</span>(<span class="dt">command_list =</span> cmds,  </a>
<a class="sourceLine" id="cb3-18" data-line-number="18">                  <span class="dt">max.jobs =</span> <span class="dv">50</span>, </a>
<a class="sourceLine" id="cb3-19" data-line-number="19">                  <span class="dt">envir =</span> empty, </a>
<a class="sourceLine" id="cb3-20" data-line-number="20">                  <span class="dt">io_saturation =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
</div>
<div id="import-reads" class="section level2">
<h2 class="hasAnchor">
<a href="#import-reads" class="anchor"></a>Import reads</h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">bams  &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">grep</a></span>(<span class="st">'error'</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list.files">dir</a></span>(<span class="st">'~/davidbr/proj/epitherapy/data/h1299/10x/dacsb/aligned/'</span>, <span class="dt">pattern =</span> <span class="st">'bam$'</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>), <span class="dt">value =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2"></a>
<a class="sourceLine" id="cb4-3" data-line-number="3">reads &lt;-<span class="st"> </span><span class="kw">importBAM</span>(bams,</a>
<a class="sourceLine" id="cb4-4" data-line-number="4">                  <span class="dt">paired  =</span> <span class="ot">TRUE</span>, </a>
<a class="sourceLine" id="cb4-5" data-line-number="5">                  <span class="dt">mate    =</span> <span class="st">'first'</span>,</a>
<a class="sourceLine" id="cb4-6" data-line-number="6">                  <span class="dt">anchor  =</span> <span class="st">'fiveprime'</span>,</a>
<a class="sourceLine" id="cb4-7" data-line-number="7">                  <span class="dt">multi   =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb4-8" data-line-number="8">                  <span class="dt">what    =</span> <span class="st">'qname'</span>,</a>
<a class="sourceLine" id="cb4-9" data-line-number="9">                  <span class="dt">barcode =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'0.25'</span>, <span class="st">'0.5'</span>, <span class="st">'0'</span>, <span class="st">'1'</span>, <span class="st">'2'</span>, <span class="st">'4'</span>, <span class="st">'8'</span>))</a>
<a class="sourceLine" id="cb4-10" data-line-number="10"></a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="co"># add overlapping TE locus</span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12">reads_anno &lt;-<span class="st"> </span>plyranges<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/plyranges/topics/overlap-joins">join_overlap_left_directed</a></span>(reads, tes)  </a></code></pre></div>
</div>
<div id="benchmarking" class="section level2">
<h2 class="hasAnchor">
<a href="#benchmarking" class="anchor"></a>Benchmarking</h2>
<div id="percent-correctly-mapped" class="section level3">
<h3 class="hasAnchor">
<a href="#percent-correctly-mapped" class="anchor"></a>Percent correctly mapped</h3>
<p>reads_dt &lt;- reads_anno %&gt;% as.data.table()</p>
<p>p_n_reads &lt;- reads_dt %&gt;% count(barcode) %&gt;% ggplot(aes(barcode, n)) + geom_bar(stat = â€˜identityâ€™)</p>
<p>fam_counts &lt;- reads_dt[which(barcode == 0), ] %&gt;% count(name)</p>
<p>bla = reads_dt[, same := abs(start - start[barcode == 0]) &lt; 10, by = â€˜qnameâ€™ ][, .(perc_cor_map = sum(same, na.rm = TRUE) / length(same) * 100, repclass = repclass[barcode == 0]), by = c(â€˜barcodeâ€™, â€˜nameâ€™)] bla %&gt;% filter(name %in% (fam_counts %&gt;% filter(n &gt; 100) %&gt;% pull(name))) %&gt;% ggplot(aes(x = as.numeric(barcode), y = perc_cor_map, group = name, col = repclass, alpha = 0.25)) + geom_point() + geom_line() + facet_wrap(~repclass) + theme(legend.position = â€˜noneâ€™) + scale_color_brewer(palette = â€˜Set1â€™) ```</p>
</div>
<div id="umi-counts" class="section level3">
<h3 class="hasAnchor">
<a href="#umi-counts" class="anchor"></a>UMI counts</h3>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">ltr12c &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="st">  </span>reads_anno <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="st">    </span>as_tibble <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="st">    </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/filter">filter</a></span>(name <span class="op">==</span><span class="st"> 'LTR12C'</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="st">    </span><span class="kw">group_by</span>(id_unique, barcode) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="st">    </span><span class="kw">summarise</span>(<span class="dt">counts =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sum">sum</a></span>(NH_weight)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="st">    </span>ungroup <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="st">    </span><span class="kw">spread</span>(barcode, counts, <span class="dt">fill =</span> <span class="dv">0</span>)             </a>
<a class="sourceLine" id="cb5-9" data-line-number="9">                </a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="co"># plot</span></a>
<a class="sourceLine" id="cb5-11" data-line-number="11">p1 &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb5-12" data-line-number="12"><span class="st">  </span>ltr12c <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-13" data-line-number="13"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="st">`</span><span class="dt">0</span><span class="st">`</span>, <span class="dt">y =</span> <span class="st">`</span><span class="dt">0.25</span><span class="st">`</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-14" data-line-number="14"><span class="st">      </span>ggrastr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/ggrastr/topics/geom_point_rast">geom_point_rast</a></span>(<span class="dt">size =</span> <span class="dv">2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-15" data-line-number="15"><span class="st">      </span><span class="kw">scale_y_log10</span>(<span class="dt">limits =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>, <span class="fl">1e4</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-16" data-line-number="16"><span class="st">      </span><span class="kw">scale_x_log10</span>(<span class="dt">limits =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>, <span class="fl">1e4</span>))  </a>
<a class="sourceLine" id="cb5-17" data-line-number="17"></a>
<a class="sourceLine" id="cb5-18" data-line-number="18"><span class="co"># plot</span></a>
<a class="sourceLine" id="cb5-19" data-line-number="19">p2 &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb5-20" data-line-number="20"><span class="st">  </span>ltr12c <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-21" data-line-number="21"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="st">`</span><span class="dt">0</span><span class="st">`</span>, <span class="dt">y =</span> <span class="st">`</span><span class="dt">0.5</span><span class="st">`</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-22" data-line-number="22"><span class="st">      </span>ggrastr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/ggrastr/topics/geom_point_rast">geom_point_rast</a></span>(<span class="dt">size =</span> <span class="dv">2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-23" data-line-number="23"><span class="st">      </span><span class="kw">scale_y_log10</span>(<span class="dt">limits =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>, <span class="fl">1e4</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-24" data-line-number="24"><span class="st">      </span><span class="kw">scale_x_log10</span>(<span class="dt">limits =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>, <span class="fl">1e4</span>))  </a>
<a class="sourceLine" id="cb5-25" data-line-number="25">      </a>
<a class="sourceLine" id="cb5-26" data-line-number="26">p3 &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb5-27" data-line-number="27"><span class="st">  </span>ltr12c <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-28" data-line-number="28"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="st">`</span><span class="dt">0</span><span class="st">`</span>, <span class="dt">y =</span> <span class="st">`</span><span class="dt">1</span><span class="st">`</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-29" data-line-number="29"><span class="st">      </span>ggrastr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/ggrastr/topics/geom_point_rast">geom_point_rast</a></span>(<span class="dt">size =</span> <span class="dv">2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-30" data-line-number="30"><span class="st">      </span><span class="kw">scale_y_log10</span>(<span class="dt">limits =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>, <span class="fl">1e4</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-31" data-line-number="31"><span class="st">      </span><span class="kw">scale_x_log10</span>(<span class="dt">limits =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>, <span class="fl">1e4</span>))     </a>
<a class="sourceLine" id="cb5-32" data-line-number="32"></a>
<a class="sourceLine" id="cb5-33" data-line-number="33">p4 &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb5-34" data-line-number="34"><span class="st">  </span>ltr12c <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-35" data-line-number="35"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="st">`</span><span class="dt">0</span><span class="st">`</span>, <span class="dt">y =</span> <span class="st">`</span><span class="dt">2</span><span class="st">`</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-36" data-line-number="36"><span class="st">      </span>ggrastr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/ggrastr/topics/geom_point_rast">geom_point_rast</a></span>(<span class="dt">size =</span> <span class="dv">2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-37" data-line-number="37"><span class="st">      </span><span class="kw">scale_y_log10</span>(<span class="dt">limits =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>, <span class="fl">1e4</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-38" data-line-number="38"><span class="st">      </span><span class="kw">scale_x_log10</span>(<span class="dt">limits =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>, <span class="fl">1e4</span>))  </a>
<a class="sourceLine" id="cb5-39" data-line-number="39"></a>
<a class="sourceLine" id="cb5-40" data-line-number="40">p5 &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb5-41" data-line-number="41"><span class="st">  </span>ltr12c <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-42" data-line-number="42"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="st">`</span><span class="dt">0</span><span class="st">`</span>, <span class="dt">y =</span> <span class="st">`</span><span class="dt">4</span><span class="st">`</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-43" data-line-number="43"><span class="st">      </span>ggrastr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/ggrastr/topics/geom_point_rast">geom_point_rast</a></span>(<span class="dt">size =</span> <span class="dv">2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-44" data-line-number="44"><span class="st">      </span><span class="kw">scale_y_log10</span>(<span class="dt">limits =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>, <span class="fl">1e4</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-45" data-line-number="45"><span class="st">      </span><span class="kw">scale_x_log10</span>(<span class="dt">limits =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>, <span class="fl">1e4</span>))      </a>
<a class="sourceLine" id="cb5-46" data-line-number="46"></a>
<a class="sourceLine" id="cb5-47" data-line-number="47">p6 &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb5-48" data-line-number="48"><span class="st">  </span>ltr12c <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb5-49" data-line-number="49"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="st">`</span><span class="dt">0</span><span class="st">`</span>, <span class="dt">y =</span> <span class="st">`</span><span class="dt">8</span><span class="st">`</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-50" data-line-number="50"><span class="st">      </span>ggrastr<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/ggrastr/topics/geom_point_rast">geom_point_rast</a></span>(<span class="dt">size =</span> <span class="dv">2</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-51" data-line-number="51"><span class="st">      </span><span class="kw">scale_y_log10</span>(<span class="dt">limits =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>, <span class="fl">1e4</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb5-52" data-line-number="52"><span class="st">      </span><span class="kw">scale_x_log10</span>(<span class="dt">limits =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>, <span class="fl">1e4</span>))       </a>
<a class="sourceLine" id="cb5-53" data-line-number="53">      </a>
<a class="sourceLine" id="cb5-54" data-line-number="54">p_comb &lt;-<span class="st"> </span>cowplot<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/cowplot/topics/plot_grid">plot_grid</a></span>(p1, p2, p3, p4, p5, p6)</a>
<a class="sourceLine" id="cb5-55" data-line-number="55"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/print">print</a></span>(p_comb)</a></code></pre></div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#benchmark-read-alignments-to-transposable-elements">Benchmark read alignments to transposable elements</a><ul class="nav nav-pills nav-stacked">
<li><a href="#getting-started">Getting started</a></li>
      <li><a href="#introduce-sequence-errors">Introduce sequence errors</a></li>
      <li><a href="#star-alignment">STAR alignment</a></li>
      <li><a href="#import-reads">Import reads</a></li>
      <li><a href="#benchmarking">Benchmarking</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by <a href="https://scholar.google.de/citations?user=ossdDj4AAAAJ&amp;hl=en&amp;oi=ao">David Brocks</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
