<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Repguide vignette • Reputils</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Repguide vignette">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Reputils</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="../articles/Reputils.html">
    <span class="fa fa-play-circle"></span>
     
    Get started
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-th-list"></span>
     
    Reference
  </a>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-news"></span>
     
    News
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://bitbucket.org/tanaylab/Reputils/src/default/">
    <span class="fa fa-bitbucket-square"></span>
     
    Bitbucket
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Repguide vignette</h1>
                        <h4 class="author">David Brocks</h4>
            
            <h4 class="date">2019-07-02</h4>
      
      
      <div class="hidden name"><code>Reputils.Rmd</code></div>

    </div>

    
    
<div id="prerequisites" class="section level1">
<h1 class="hasAnchor">
<a href="#prerequisites" class="anchor"></a>Prerequisites</h1>
<div id="genome-assembly-required" class="section level2">
<h2 class="hasAnchor">
<a href="#genome-assembly-required" class="anchor"></a>Genome assembly (required)</h2>
<p>Reputils can compute family-wise multiple sequence alignments of all repeat sequences in the genome to improve read mapping onto a consensus model. To do so, we require the genome assembly of our organism of choice stored as a <em>BSgenome</em> object. You can retrieve the full list of supported genomes by typing <code><a href="https://www.rdocumentation.org/packages/BSgenome/topics/available.genomes">BSgenome::available.genomes()</a></code> or create a custom <em>BSgenome</em> object following the <a href="https://www.bioconductor.org/packages//2.7/bioc/vignettes/BSgenome/inst/doc/BSgenomeForge.pdf">instructions</a>.</p>
<p>Since we are working with expression data from human cancer cell lines, we will first install the UCSC hg38 assemblies.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">BiocManager<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/BiocManager/topics/install">install</a></span>(<span class="st">"BSgenome.Hsapiens.UCSC.hg38"</span>)</a></code></pre></div>
<p><strong>Important</strong> <strong>note</strong>: Do not use <strong>repeat-masked</strong> <em>BSgenome</em> objects (contain ‘masked’ suffix, e.g. <em>BSgenome.Hsapiens.UCSC.hg38.masked</em>)!</p>
</div>
<div id="repeat-annotation-required" class="section level2">
<h2 class="hasAnchor">
<a href="#repeat-annotation-required" class="anchor"></a>Repeat annotation (required)</h2>
<p>A convient solution is to download transposon coordinates from the <a href="http://www.repeatmasker.org/genomicDatasets/RMGenomicDatasets.html">Repeatmasker homepage</a> or the <a href="http://www.dfam.org/releases/Dfam_3.0/annotations/">DFAM database</a> for your genome and assembly of choice. You can import the Repeatmasker <em>fa.out.gz</em> or DFAM <em>dfam.hits.gz</em> files using the Reputils <code>importRMSK</code> and <code>importDFAM</code> functions, respectively. Another option is to provide custom annotation as long as it provides the basic information about chromosome, start, end, strand, repname (family identifier), and id_unique (unique locus identifier). In this tutorial, we will show you how to import such information using the provided example datasets.</p>
</div>
<div id="readumi-genomic-coordinates-required" class="section level2">
<h2 class="hasAnchor">
<a href="#readumi-genomic-coordinates-required" class="anchor"></a>Read/UMI genomic coordinates (required)</h2>
<p>Reputils requires read alignment coordinates stored in <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">BAM format</a> as input, which are routinely generated during most common scRNA-seq workflows, including <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger">10x’ Cellranger pipeline</a>. BAM inputs should be duplicate removed, position sorted, and indexed. Chunking BAM inputs (e.g. by chromosome) can accelerate the import into your R environment using the <code>importBAM</code> function. <strong>Important:</strong> Reputils assumes the cell barcode is either stored as CB tag or BAM input files are seperated per cell. Other formats are currently not supported (e.g. cell barcode in read name).</p>
</div>
<div id="getting-started" class="section level2">
<h2 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h2>
<p>We start the workflow by loading Repsc and the human hg38 <em>BSgenome</em> object into our R environment:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/load_all">load_all</a></span>(<span class="st">'/net/mraid14/export/tgdata/users/davidbr/src/Reputils/'</span>)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/load_all">load_all</a></span>(<span class="st">'/net/mraid14/export/tgdata/users/davidbr/src/Repsc/'</span>)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="co"># adjust to your genome of interest (e.g. BSgenome.Mmusculus.UCSC.mm10)</span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(BSgenome.Hsapiens.UCSC.hg38)</a></code></pre></div>
</div>
</div>
<div id="introduce-sequence-errors" class="section level1">
<h1 class="hasAnchor">
<a href="#introduce-sequence-errors" class="anchor"></a>Introduce sequence errors</h1>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">bams    &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">grep</a></span>(<span class="st">'deduplicated'</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list.files">dir</a></span>(<span class="st">'~/davidbr/proj/epitherapy/data/h1299/10x/dacsb/aligned/chunked_bams/'</span>, <span class="dt">pattern =</span> <span class="st">'.bam$'</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>), <span class="dt">value =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">tes     &lt;-<span class="st"> </span><span class="kw">importRMSK</span>(<span class="st">'~/tgdata/src/Reputils/inst/extdata/hg38.fa.out.gz'</span>, <span class="dt">curate =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="kw">mutateReads</span>(<span class="dt">BSgenome   =</span> Hsapiens,</a>
<a class="sourceLine" id="cb3-5" data-line-number="5">            <span class="dt">reads      =</span> bams,</a>
<a class="sourceLine" id="cb3-6" data-line-number="6">            <span class="dt">paired     =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb3-7" data-line-number="7">            <span class="dt">tes        =</span> tes,</a>
<a class="sourceLine" id="cb3-8" data-line-number="8">            <span class="dt">n_reads    =</span> <span class="fl">1e6</span>,</a>
<a class="sourceLine" id="cb3-9" data-line-number="9">            <span class="dt">outdir     =</span> <span class="st">'~/davidbr/proj/epitherapy/data/h1299/10x/dacsb/fastqs/'</span>,</a>
<a class="sourceLine" id="cb3-10" data-line-number="10">            <span class="dt">error_rate =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>, <span class="fl">0.25</span>, <span class="fl">0.5</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">4</span>, <span class="dv">8</span>))</a></code></pre></div>
</div>
<div id="star-alignment" class="section level1">
<h1 class="hasAnchor">
<a href="#star-alignment" class="anchor"></a>STAR alignment</h1>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">r1 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list.files">dir</a></span>(<span class="st">'~/davidbr/proj/epitherapy/data/h1299/10x/dacsb/fastqs/'</span>, <span class="dt">pattern =</span> <span class="st">'perc_error_R1'</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">r2 &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list.files">dir</a></span>(<span class="st">'~/davidbr/proj/epitherapy/data/h1299/10x/dacsb/fastqs/'</span>, <span class="dt">pattern =</span> <span class="st">'perc_error_R2'</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb4-3" data-line-number="3"></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/getwd">setwd</a></span>(<span class="st">'~/davidbr/proj/epitherapy/data/h1299/10x/dacsb/aligned/'</span>)</a>
<a class="sourceLine" id="cb4-5" data-line-number="5">cmds &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>()</a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(r1))</a>
<a class="sourceLine" id="cb4-7" data-line-number="7">{</a>
<a class="sourceLine" id="cb4-8" data-line-number="8">  cmds[[i]] &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="st">    </span><span class="kw">glue</span>(<span class="st">"system('/net/mraid14/export/data/tools/star/STAR-2.5.1b/source/STAR --runThreadN 10 --genomeDir /net/mraid14/export/data/users/davidbr/tools/cellranger/references/refdata-cellranger-GRCh38-1.2.0/star/ --outSAMtype BAM SortedByCoordinate --outFilterMultimapScoreRange 2 --outFileNamePrefix {gsub('.fastq', '.', basename(r1[i]))} --outSAMunmapped Within --readFilesIn {r1[i]} {r2[i]}')"</span>)</a>
<a class="sourceLine" id="cb4-10" data-line-number="10">}</a>
<a class="sourceLine" id="cb4-11" data-line-number="11"></a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="co"># create new empty env and fill with relevant</span></a>
<a class="sourceLine" id="cb4-13" data-line-number="13">empty &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/environment">new.env</a></span>(<span class="dt">parent =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/environment">emptyenv</a></span>())</a>
<a class="sourceLine" id="cb4-14" data-line-number="14"></a>
<a class="sourceLine" id="cb4-15" data-line-number="15"><span class="co"># distribute, compute and combine</span></a>
<a class="sourceLine" id="cb4-16" data-line-number="16">res &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb4-17" data-line-number="17"><span class="st">  </span><span class="kw">gcluster.run3</span>(<span class="dt">command_list =</span> cmds,  </a>
<a class="sourceLine" id="cb4-18" data-line-number="18">                  <span class="dt">max.jobs =</span> <span class="dv">50</span>, </a>
<a class="sourceLine" id="cb4-19" data-line-number="19">                  <span class="dt">envir =</span> empty, </a>
<a class="sourceLine" id="cb4-20" data-line-number="20">                  <span class="dt">io_saturation =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
</div>
<div id="create-scset" class="section level1">
<h1 class="hasAnchor">
<a href="#create-scset" class="anchor"></a>Create scSet</h1>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="co"># path to Gencode gtf file (provided)</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2">gene_path &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="dt">package =</span> <span class="st">'Reputils'</span>, </a>
<a class="sourceLine" id="cb5-3" data-line-number="3">                        <span class="st">'extdata'</span>, </a>
<a class="sourceLine" id="cb5-4" data-line-number="4">                        <span class="st">'gencode.v29.annotation.gtf.gz'</span>)</a>
<a class="sourceLine" id="cb5-5" data-line-number="5"></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="co"># path to RepeatMasker hg38 repeat annotation (provided)</span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7">rmsk_path &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="dt">package =</span> <span class="st">'Reputils'</span>, </a>
<a class="sourceLine" id="cb5-8" data-line-number="8">                         <span class="st">'extdata'</span>, </a>
<a class="sourceLine" id="cb5-9" data-line-number="9">                         <span class="st">'hg38.fa.out.gz'</span>)</a>
<a class="sourceLine" id="cb5-10" data-line-number="10">                         </a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="co"># creating the scSet</span></a>
<a class="sourceLine" id="cb5-12" data-line-number="12">sc &lt;-<span class="st"> </span><span class="kw">createScSet</span>(<span class="dt">genome   =</span> Hsapiens,</a>
<a class="sourceLine" id="cb5-13" data-line-number="13">                  <span class="dt">protocol =</span> <span class="st">'fiveprime'</span>,</a>
<a class="sourceLine" id="cb5-14" data-line-number="14">                  <span class="dt">tes      =</span> rmsk_path,</a>
<a class="sourceLine" id="cb5-15" data-line-number="15">                  <span class="dt">genes    =</span> gene_path)</a></code></pre></div>
</div>
<div id="count-reads" class="section level1">
<h1 class="hasAnchor">
<a href="#count-reads" class="anchor"></a>Count reads</h1>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="co"># path to bam files containing mapped reads                         </span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2">bam_paths &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb6-3" data-line-number="3"><span class="st">  </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">grep</a></span>(<span class="st">'error'</span>, </a>
<a class="sourceLine" id="cb6-4" data-line-number="4">       <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list.files">dir</a></span>(<span class="st">'~/davidbr/proj/epitherapy/data/h1299/10x/dacsb/aligned/'</span>, </a>
<a class="sourceLine" id="cb6-5" data-line-number="5">           <span class="dt">pattern =</span> <span class="st">'bam$'</span>, </a>
<a class="sourceLine" id="cb6-6" data-line-number="6">           <span class="dt">full.names =</span> <span class="ot">TRUE</span>), </a>
<a class="sourceLine" id="cb6-7" data-line-number="7">       <span class="dt">value =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb6-8" data-line-number="8">                 </a>
<a class="sourceLine" id="cb6-9" data-line-number="9"><span class="co"># create a data.frame specifying import parameters                 </span></a>
<a class="sourceLine" id="cb6-10" data-line-number="10">bam_df    &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="dt">paths   =</span> bam_paths,</a>
<a class="sourceLine" id="cb6-11" data-line-number="11">                        <span class="dt">paired  =</span> <span class="ot">TRUE</span>,       <span class="co"># use FALSE for single-end libraries</span></a>
<a class="sourceLine" id="cb6-12" data-line-number="12">                        <span class="dt">mate    =</span> <span class="st">'first'</span>,    <span class="co"># only imports the first mate of properly aligned read pairs, set to NA when using single-end libraries</span></a>
<a class="sourceLine" id="cb6-13" data-line-number="13">                        <span class="dt">barcode =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'0.5'</span>, <span class="st">'0'</span>, <span class="st">'1'</span>, <span class="st">'2'</span>, <span class="st">'4'</span>, <span class="st">'8'</span>),       <span class="co"># 10x barcode included in BAM flag</span></a>
<a class="sourceLine" id="cb6-14" data-line-number="14">                        <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb6-15" data-line-number="15">sc &lt;-<span class="st"> </span><span class="kw">addCounts</span>(sc,</a>
<a class="sourceLine" id="cb6-16" data-line-number="16">                <span class="dt">reads    =</span> bam_df,</a>
<a class="sourceLine" id="cb6-17" data-line-number="17">                <span class="dt">bin_size =</span> <span class="dv">20</span>,</a>
<a class="sourceLine" id="cb6-18" data-line-number="18">                <span class="dt">msa_dir  =</span> <span class="ot">NULL</span>)</a></code></pre></div>
</div>
<div id="mapping-qc" class="section level1">
<h1 class="hasAnchor">
<a href="#mapping-qc" class="anchor"></a>Mapping QC</h1>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">ltr12c &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="st">  </span><span class="kw">counts</span>(sc) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="st">    </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/filter">filter</a></span>(name <span class="op">==</span><span class="st"> 'LTR12C'</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="st">    </span><span class="kw">group_by</span>(id_unique, barcode) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="st">    </span><span class="kw">summarize</span>(<span class="dt">counts =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sum">sum</a></span>(n)) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="st">    </span>ungroup</a>
<a class="sourceLine" id="cb7-7" data-line-number="7">                </a>
<a class="sourceLine" id="cb7-8" data-line-number="8"><span class="co"># plot</span></a>
<a class="sourceLine" id="cb7-9" data-line-number="9">p1 &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb7-10" data-line-number="10"><span class="st">  </span>ltr12c <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-11" data-line-number="11"><span class="st">    </span><span class="kw">spread</span>(barcode, counts, <span class="dt">fill =</span> <span class="dv">0</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-12" data-line-number="12"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="st">`</span><span class="dt">0</span><span class="st">`</span>, <span class="dt">y =</span> <span class="st">`</span><span class="dt">0.5</span><span class="st">`</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb7-13" data-line-number="13"><span class="st">      </span><span class="kw">geom_point</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb7-14" data-line-number="14"><span class="st">      </span><span class="kw">scale_y_log10</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb7-15" data-line-number="15"><span class="st">      </span><span class="kw">scale_x_log10</span>()</a>
<a class="sourceLine" id="cb7-16" data-line-number="16"></a>
<a class="sourceLine" id="cb7-17" data-line-number="17"><span class="co"># plot</span></a>
<a class="sourceLine" id="cb7-18" data-line-number="18">p2 &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb7-19" data-line-number="19"><span class="st">  </span>ltr12c <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-20" data-line-number="20"><span class="st">    </span><span class="kw">spread</span>(barcode, counts, <span class="dt">fill =</span> <span class="dv">0</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb7-21" data-line-number="21"><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="st">`</span><span class="dt">0</span><span class="st">`</span>, <span class="dt">y =</span> <span class="st">`</span><span class="dt">8</span><span class="st">`</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb7-22" data-line-number="22"><span class="st">      </span><span class="kw">geom_point</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb7-23" data-line-number="23"><span class="st">      </span><span class="kw">scale_y_log10</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb7-24" data-line-number="24"><span class="st">      </span><span class="kw">scale_x_log10</span>()</a>
<a class="sourceLine" id="cb7-25" data-line-number="25">      </a>
<a class="sourceLine" id="cb7-26" data-line-number="26">cowplot<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/cowplot/topics/plot_grid">plot_grid</a></span>(p1, p2)</a></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">bams  &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/grep">grep</a></span>(<span class="st">'error'</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list.files">dir</a></span>(<span class="st">'~/davidbr/proj/epitherapy/data/h1299/10x/dacsb/aligned/'</span>, <span class="dt">pattern =</span> <span class="st">'bam$'</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>), <span class="dt">value =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">reads &lt;-<span class="st"> </span><span class="kw">importBAM</span>(bams,</a>
<a class="sourceLine" id="cb8-3" data-line-number="3">                  <span class="dt">paired  =</span> <span class="ot">TRUE</span>, </a>
<a class="sourceLine" id="cb8-4" data-line-number="4">                  <span class="dt">mate    =</span> <span class="st">'first'</span>,</a>
<a class="sourceLine" id="cb8-5" data-line-number="5">                  <span class="dt">anchor  =</span> <span class="st">'fiveprime'</span>,</a>
<a class="sourceLine" id="cb8-6" data-line-number="6">                  <span class="dt">multi   =</span> <span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb8-7" data-line-number="7">                  <span class="dt">barcode =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'0.25'</span>, <span class="st">'0.5'</span>, <span class="st">'0'</span>, <span class="st">'1'</span>, <span class="st">'2'</span>, <span class="st">'4'</span>, <span class="st">'8'</span>))</a>
<a class="sourceLine" id="cb8-8" data-line-number="8"></a>
<a class="sourceLine" id="cb8-9" data-line-number="9"><span class="co"># add overlapping TE locus</span></a>
<a class="sourceLine" id="cb8-10" data-line-number="10">reads_anno &lt;-<span class="st"> </span><span class="kw">join_overlap_left_directed</span>(reads, tes)   </a>
<a class="sourceLine" id="cb8-11" data-line-number="11"></a>
<a class="sourceLine" id="cb8-12" data-line-number="12">reads_df &lt;-<span class="st"> </span>reads_anno <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">read_id =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/names">names</a></span>(reads_anno)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>()        </a>
<a class="sourceLine" id="cb8-13" data-line-number="13"></a>
<a class="sourceLine" id="cb8-14" data-line-number="14">n_reads    &lt;-<span class="st"> </span>reads_dt <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">count</span>(barcode)</a>
<a class="sourceLine" id="cb8-15" data-line-number="15">fam_counts &lt;-<span class="st"> </span>reads_dt[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/which">which</a></span>(barcode <span class="op">==</span><span class="st"> </span><span class="dv">0</span>), ] <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">count</span>(name)</a>
<a class="sourceLine" id="cb8-16" data-line-number="16"></a>
<a class="sourceLine" id="cb8-17" data-line-number="17">bla =<span class="st"> </span></a>
<a class="sourceLine" id="cb8-18" data-line-number="18"><span class="st">  </span>reads_dt[, same <span class="op">:</span><span class="er">=</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/MathFun">abs</a></span>(start <span class="op">-</span><span class="st"> </span>start[barcode <span class="op">==</span><span class="st"> </span><span class="dv">0</span>]) <span class="op">&lt;</span><span class="st"> </span><span class="dv">10</span>, by =<span class="st"> 'read_id'</span></a>
<a class="sourceLine" id="cb8-19" data-line-number="19">    ][, .(<span class="dt">correctly_mapped =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sum">sum</a></span>(same, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>) <span class="op">/</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(same) <span class="op">*</span><span class="st"> </span><span class="dv">100</span>, <span class="dt">repclass =</span> repclass[<span class="dv">1</span>]), by =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'barcode'</span>, <span class="st">'name'</span>)]</a>
<a class="sourceLine" id="cb8-20" data-line-number="20">bla <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb8-21" data-line-number="21"><span class="st">  </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/filter">filter</a></span>(name <span class="op">%in%</span><span class="st"> </span>(fam_counts <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/filter">filter</a></span>(n <span class="op">&gt;</span><span class="st"> </span><span class="dv">100</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pull</span>(name))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb8-22" data-line-number="22"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/numeric">as.numeric</a></span>(barcode), <span class="dt">y =</span> correctly_mapped, <span class="dt">group =</span> name, <span class="dt">col =</span> repclass, <span class="dt">alpha =</span> <span class="fl">0.25</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb8-23" data-line-number="23"><span class="st">    </span><span class="kw">geom_point</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb8-24" data-line-number="24"><span class="st">    </span><span class="kw">geom_line</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb8-25" data-line-number="25"><span class="st">    </span><span class="kw">facet_wrap</span>(<span class="op">~</span>repclass) <span class="op">+</span></a>
<a class="sourceLine" id="cb8-26" data-line-number="26"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">'none'</span>)</a></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#prerequisites">Prerequisites</a><ul class="nav nav-pills nav-stacked">
<li><a href="#genome-assembly-required">Genome assembly (required)</a></li>
      <li><a href="#repeat-annotation-required">Repeat annotation (required)</a></li>
      <li><a href="#readumi-genomic-coordinates-required">Read/UMI genomic coordinates (required)</a></li>
      <li><a href="#getting-started">Getting started</a></li>
      </ul>
</li>
      <li><a href="#introduce-sequence-errors">Introduce sequence errors</a></li>
      <li><a href="#star-alignment">STAR alignment</a></li>
      <li><a href="#create-scset">Create scSet</a></li>
      <li><a href="#count-reads">Count reads</a></li>
      <li><a href="#mapping-qc">Mapping QC</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by <a href="https://scholar.google.de/citations?user=ossdDj4AAAAJ&amp;hl=en&amp;oi=ao">David Brocks</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
