<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Benchmarking â€¢ Repguide</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Benchmarking">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Repguide</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="../articles/Repguide.html">
    <span class="fa fa-play-circle"></span>
     
    Get started
  </a>
</li>
<li>
  <a href="../articles/benchmark.html">
    <span class="fa fa-bar-chart"></span>
     
    Benchmark
  </a>
</li>
<li>
  <a href="../articles/guide_scoring.html">
    <span class="fa fa-star"></span>
     
    Guide scoring
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-th-list"></span>
     
    Reference
  </a>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-news"></span>
     
    News
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://bitbucket.org/tanaylab/repguide/src/default/">
    <span class="fa fa-bitbucket-square"></span>
     
    Bitbucket
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Benchmarking</h1>
                        <h4 class="author">David Brocks</h4>
            
            <h4 class="date">2019-07-31</h4>
      
      
      <div class="hidden name"><code>benchmark.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>In this benchmark, we will use Repguide to predict genomic binding sites for a publically available set of 12 guideRNAs targeting LTR5_Hs long terminal repeats and compare the results to experimentally determined Cas9 ChIP-seq peak intensities [1].</p>
</div>
<div id="running-repguide" class="section level2">
<h2 class="hasAnchor">
<a href="#running-repguide" class="anchor"></a>Running Repguide</h2>
<p>We first create a hg38 guideSet and provide the guide sequences to map.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(Repguide)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(BSgenome.Hsapiens.UCSC.hg38)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(tidyverse)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"></a>
<a class="sourceLine" id="cb1-5" data-line-number="5"><span class="co"># Path to RepeatMasker hg38 repeat annotation (provided, LTRs only)</span></a>
<a class="sourceLine" id="cb1-6" data-line-number="6">repeats_path &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="dt">package =</span> <span class="st">'Repguide'</span>, <span class="st">'extdata'</span>, <span class="st">'hg38_ucsc_rmsk_ltr.txt.gz'</span>)</a>
<a class="sourceLine" id="cb1-7" data-line-number="7"></a>
<a class="sourceLine" id="cb1-8" data-line-number="8">gs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/createGuideSet.html">createGuideSet</a></span>(<span class="dt">genome =</span> Hsapiens, <span class="co"># Hsapiens is the short version of BSgenome.Hsapiens.UCSC.hg38</span></a>
<a class="sourceLine" id="cb1-9" data-line-number="9">                     <span class="dt">tes =</span> repeats_path) <span class="co"># can be either path to file or GRanges object)</span></a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="co">#&gt; Created new guideSet at Wed Jul 31 15:26:26 2019</span></a>
<a class="sourceLine" id="cb1-11" data-line-number="11">                     </a>
<a class="sourceLine" id="cb1-12" data-line-number="12">gs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/addTargets.html">addTargets</a></span>(gs, <span class="st">'LTR5_Hs'</span>)</a>
<a class="sourceLine" id="cb1-13" data-line-number="13"></a>
<a class="sourceLine" id="cb1-14" data-line-number="14"><span class="co"># Import guideRNA sequences used in Fuentes et al. [1] against LTR5_Hs</span></a>
<a class="sourceLine" id="cb1-15" data-line-number="15">guides_path &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="dt">package =</span> <span class="st">'Repguide'</span>, <span class="st">'extdata'</span>, <span class="st">'fuentes_elife_2018_guides.fasta'</span>)</a>
<a class="sourceLine" id="cb1-16" data-line-number="16">guides_fuentes &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/character">as.character</a></span>(Biostrings<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/Biostrings/topics/XStringSet-io">readDNAStringSet</a></span>(guides_path))</a>
<a class="sourceLine" id="cb1-17" data-line-number="17"></a>
<a class="sourceLine" id="cb1-18" data-line-number="18">gs_fuentes &lt;-<span class="st"> </span><span class="kw"><a href="../reference/addGuides.html">addGuides</a></span>(gs, </a>
<a class="sourceLine" id="cb1-19" data-line-number="19">                        <span class="dt">guides =</span> guides_fuentes, </a>
<a class="sourceLine" id="cb1-20" data-line-number="20">                        <span class="dt">n_mismatches =</span> <span class="dv">3</span>, </a>
<a class="sourceLine" id="cb1-21" data-line-number="21">                        <span class="dt">min_Son =</span> <span class="dv">0</span>, </a>
<a class="sourceLine" id="cb1-22" data-line-number="22">                        <span class="dt">max_Soff =</span> <span class="ot">Inf</span>, </a>
<a class="sourceLine" id="cb1-23" data-line-number="23">                        <span class="dt">gc_content =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>,<span class="dv">1</span>),</a>
<a class="sourceLine" id="cb1-24" data-line-number="24">                        <span class="dt">PAM =</span> <span class="st">'NGG'</span>)                     </a>
<a class="sourceLine" id="cb1-25" data-line-number="25"><span class="co">#&gt; [1] "Aligning kmers against: BSgenome.Hsapiens.UCSC.hg38"</span></a>
<a class="sourceLine" id="cb1-26" data-line-number="26"><span class="co">#&gt; [1] "Annotating guides"</span></a>
<a class="sourceLine" id="cb1-27" data-line-number="27"><span class="co">#&gt; [1] "Computing guide scores"</span></a>
<a class="sourceLine" id="cb1-28" data-line-number="28"><span class="co">#&gt; [1] "Blacklisting kmers"</span></a></code></pre></div>
</div>
<div id="compare-predicted-and-experimental-binding-sites" class="section level2">
<h2 class="hasAnchor">
<a href="#compare-predicted-and-experimental-binding-sites" class="anchor"></a>Compare predicted and experimental binding sites</h2>
<p>We can extract predicted binding sites using the <code>mappings</code> accessor and import provided dCAS9 ChIP peak coordinates from Fuentes et al. [1]. To facilitate comparison, we restrict our analysis to peaks on the main chromosome assemblies.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># Extract predicted binding sites</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2">pred_sites &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mappings.html">mappings</a></span>(gs_fuentes)     </a>
<a class="sourceLine" id="cb2-3" data-line-number="3"></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="co"># Import experimental binding sites            </span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5">peaks_path &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="dt">package =</span> <span class="st">'Repguide'</span>, <span class="st">'extdata'</span>, <span class="st">'fuentes_elife_2018_dCas9_peaks.bed'</span>)</a>
<a class="sourceLine" id="cb2-6" data-line-number="6">exp_sites  &lt;-<span class="st"> </span>rtracklayer<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/rtracklayer/topics/BEDFile-class">import.bed</a></span>(peaks_path)</a>
<a class="sourceLine" id="cb2-7" data-line-number="7"></a>
<a class="sourceLine" id="cb2-8" data-line-number="8"><span class="co"># Throw chromosomes not present in guideSet</span></a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="kw">seqlevels</span>(exp_sites, <span class="dt">pruning.mode =</span> <span class="st">'coarse'</span>) &lt;-<span class="st"> </span><span class="kw">seqlevels</span>(pred_sites) </a></code></pre></div>
<p>We then compare the absolute number and binding intensities of correctly predicted binding sites.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">hits &lt;-<span class="st"> </span>IRanges<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/IRanges/topics/findOverlaps-methods">findOverlaps</a></span>(exp_sites, pred_sites)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co"># Add logical whether or not peak was predicted by Repguide</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4">exp_sites<span class="op">$</span>predicted &lt;-<span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(exp_sites) <span class="op">%in%</span><span class="st"> </span>hits<span class="op">@</span>from</a>
<a class="sourceLine" id="cb3-5" data-line-number="5"></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="co"># Add Repguide predicted binding score to ChIP peaks</span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7">exp_sites<span class="op">$</span>score_pred =<span class="st"> </span><span class="dv">0</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8">exp_sites[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(hits<span class="op">@</span>from)]<span class="op">$</span>score_pred &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="st">  </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/as.data.frame">as.data.frame</a></span>(hits) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">score_pred =</span> pred_sites[hits<span class="op">@</span>to]<span class="op">$</span>Sbind) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="st">  </span><span class="kw">group_by</span>(queryHits) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="st">    </span><span class="kw">summarise</span>(<span class="dt">score_pred =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sum">sum</a></span>(score_pred)) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="st">  </span><span class="kw">pull</span>(score_pred)</a></code></pre></div>
</div>
<div id="plotting-the-results" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-the-results" class="anchor"></a>Plotting the results</h2>
<p>We finally percent rank transform the predicted and experimental scores and visualize the results using <code>ggplot2</code>.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">ggdata &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="st">  </span>exp_sites <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="st">  </span>as.data.frame <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">score_pred =</span> <span class="kw">percent_rank</span>(score_pred), </a>
<a class="sourceLine" id="cb4-5" data-line-number="5">         <span class="dt">score      =</span> <span class="kw">percent_rank</span>(score)) </a>
<a class="sourceLine" id="cb4-6" data-line-number="6"></a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co"># Create plots         </span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8">p_peaks_pie  &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="st">  </span>ggdata <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="st">''</span>, <span class="dt">fill =</span> predicted)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="st">    </span><span class="kw">geom_bar</span>() <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="st">    </span><span class="kw">coord_polar</span>(<span class="st">"y"</span>, <span class="dt">start =</span> <span class="dv">0</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb4-13" data-line-number="13"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">'none'</span>,</a>
<a class="sourceLine" id="cb4-14" data-line-number="14">          <span class="dt">panel.border =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb4-15" data-line-number="15">          <span class="dt">axis.title.x =</span> <span class="kw">element_blank</span>(),</a>
<a class="sourceLine" id="cb4-16" data-line-number="16">          <span class="dt">axis.title.y =</span> <span class="kw">element_blank</span>())</a>
<a class="sourceLine" id="cb4-17" data-line-number="17">p_score_boxplot &lt;-<span class="st">              </span></a>
<a class="sourceLine" id="cb4-18" data-line-number="18"><span class="st">  </span>ggdata <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-19" data-line-number="19"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> predicted, <span class="dt">y =</span> score, <span class="dt">group =</span> predicted, <span class="dt">fill =</span> predicted)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-20" data-line-number="20"><span class="st">    </span><span class="kw">geom_boxplot</span>(<span class="dt">outlier.shape =</span> <span class="ot">NA</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-21" data-line-number="21"><span class="st">    </span><span class="kw">geom_jitter</span>(<span class="dt">size =</span> <span class="dv">1</span>, <span class="dt">alpha =</span> <span class="fl">0.1</span>) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-22" data-line-number="22"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">90</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb4-23" data-line-number="23"><span class="st">    </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">'none'</span>)</a>
<a class="sourceLine" id="cb4-24" data-line-number="24">p_score_scatter &lt;-<span class="st"> </span></a>
<a class="sourceLine" id="cb4-25" data-line-number="25"><span class="st">  </span>ggdata <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-26" data-line-number="26"><span class="st">  </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/filter">filter</a></span>(predicted) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-27" data-line-number="27"><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(score, score_pred)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb4-28" data-line-number="28"><span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb4-29" data-line-number="29"></a>
<a class="sourceLine" id="cb4-30" data-line-number="30"><span class="co"># Combine plots </span></a>
<a class="sourceLine" id="cb4-31" data-line-number="31">cowplot<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/cowplot/topics/plot_grid">plot_grid</a></span>(p_peaks_pie, </a>
<a class="sourceLine" id="cb4-32" data-line-number="32">                   p_score_boxplot,</a>
<a class="sourceLine" id="cb4-33" data-line-number="33">                   p_score_scatter, <span class="dt">rel_widths =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="fl">1.5</span>, <span class="dv">1</span>, <span class="dv">2</span>), </a>
<a class="sourceLine" id="cb4-34" data-line-number="34">                   <span class="dt">labels =</span> <span class="st">'auto'</span>, </a>
<a class="sourceLine" id="cb4-35" data-line-number="35">                   <span class="dt">nrow =</span> <span class="dv">1</span>)</a></code></pre></div>
<p><img src="benchmark_files/figure-html/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;"></p>
<p>We can see that Repguide correctly predicted more than 3/4 of experimentally determined Cas9 peaks (<strong>a</strong>) and failed predictions generally showed weak ChIP enrichment (<strong>b</strong>). Moreover, computed binding scores were largely consistent with ChIP signal intensities (<strong>c</strong>), suggesting Repguide on and off-targetd scores can be interpreted quantitatively.</p>
</div>
<div id="references" class="section level2">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<p>[1] Daniel R Fuentes, Tomek Swigut, Joanna Wysocka. Systematic perturbation of retroviral LTRs reveals widespread long-range effects on human gene regulation. eLife (2018).</p>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/sessionInfo">sessionInfo</a></span>()</a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="co">#&gt; R version 3.5.3 (2019-03-11)</span></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="co">#&gt; Running under: CentOS Linux 7 (Core)</span></a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-6" data-line-number="6"><span class="co">#&gt; Matrix products: default</span></a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="co">#&gt; BLAS/LAPACK: /net/mraid14/export/data/users/eladch/tools/CO7/mkl/2018.3/compilers_and_libraries_2018.3.222/linux/mkl/lib/intel64_lin/libmkl_gf_lp64.so</span></a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="co">#&gt; locale:</span></a>
<a class="sourceLine" id="cb5-10" data-line-number="10"><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8      </span></a>
<a class="sourceLine" id="cb5-11" data-line-number="11"><span class="co">#&gt;  [8] LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></a>
<a class="sourceLine" id="cb5-12" data-line-number="12"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-13" data-line-number="13"><span class="co">#&gt; attached base packages:</span></a>
<a class="sourceLine" id="cb5-14" data-line-number="14"><span class="co">#&gt; [1] stats4    parallel  stats     graphics  grDevices utils     datasets  methods   base     </span></a>
<a class="sourceLine" id="cb5-15" data-line-number="15"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-16" data-line-number="16"><span class="co">#&gt; other attached packages:</span></a>
<a class="sourceLine" id="cb5-17" data-line-number="17"><span class="co">#&gt;  [1] forcats_0.4.0                     stringr_1.4.0                     dplyr_0.8.0.1                     purrr_0.3.2                       readr_1.3.1                      </span></a>
<a class="sourceLine" id="cb5-18" data-line-number="18"><span class="co">#&gt;  [6] tidyr_0.8.3                       tibble_2.1.1                      ggplot2_3.1.0                     tidyverse_1.2.1                   BSgenome.Hsapiens.UCSC.hg38_1.4.1</span></a>
<a class="sourceLine" id="cb5-19" data-line-number="19"><span class="co">#&gt; [11] BSgenome_1.50.0                   rtracklayer_1.42.2                Biostrings_2.50.2                 XVector_0.22.0                    GenomicRanges_1.34.0             </span></a>
<a class="sourceLine" id="cb5-20" data-line-number="20"><span class="co">#&gt; [16] GenomeInfoDb_1.18.2               IRanges_2.16.0                    S4Vectors_0.20.1                  BiocGenerics_0.28.0               Repguide_0.0.1.2                 </span></a>
<a class="sourceLine" id="cb5-21" data-line-number="21"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb5-22" data-line-number="22"><span class="co">#&gt; loaded via a namespace (and not attached):</span></a>
<a class="sourceLine" id="cb5-23" data-line-number="23"><span class="co">#&gt;  [1] nlme_3.1-137                bitops_1.0-6                matrixStats_0.54.0          fs_1.2.7                    doMC_1.3.5                  lubridate_1.7.4             httr_1.4.0                 </span></a>
<a class="sourceLine" id="cb5-24" data-line-number="24"><span class="co">#&gt;  [8] rprojroot_1.3-2             tools_3.5.3                 backports_1.1.4             R6_2.4.0                    lazyeval_0.2.2              colorspace_1.4-1            withr_2.1.2                </span></a>
<a class="sourceLine" id="cb5-25" data-line-number="25"><span class="co">#&gt; [15] tidyselect_0.2.5            compiler_3.5.3              cli_1.1.0                   rvest_0.3.2                 Biobase_2.42.0              xml2_1.2.0                  desc_1.2.0                 </span></a>
<a class="sourceLine" id="cb5-26" data-line-number="26"><span class="co">#&gt; [22] DelayedArray_0.8.0          labeling_0.3                scales_1.0.0                pkgdown_1.3.0               commonmark_1.7              digest_0.6.19               Rsamtools_1.34.1           </span></a>
<a class="sourceLine" id="cb5-27" data-line-number="27"><span class="co">#&gt; [29] rmarkdown_1.12              R.utils_2.8.0               pkgconfig_2.0.2             htmltools_0.3.6             rlang_0.3.2                 readxl_1.3.1                rstudioapi_0.9.0           </span></a>
<a class="sourceLine" id="cb5-28" data-line-number="28"><span class="co">#&gt; [36] generics_0.0.2              jsonlite_1.6                BiocParallel_1.16.6         R.oo_1.22.0                 RCurl_1.95-4.12             magrittr_1.5                GenomeInfoDbData_1.2.0     </span></a>
<a class="sourceLine" id="cb5-29" data-line-number="29"><span class="co">#&gt; [43] Matrix_1.2-16               Rcpp_1.0.1                  munsell_0.5.0               Rbowtie_1.22.0              R.methodsS3_1.7.1           stringi_1.4.3               yaml_2.2.0                 </span></a>
<a class="sourceLine" id="cb5-30" data-line-number="30"><span class="co">#&gt; [50] MASS_7.3-51.1               SummarizedExperiment_1.12.0 zlibbioc_1.28.0             plyr_1.8.4                  grid_3.5.3                  crayon_1.3.4                lattice_0.20-38            </span></a>
<a class="sourceLine" id="cb5-31" data-line-number="31"><span class="co">#&gt; [57] cowplot_0.9.4               haven_2.1.0                 hms_0.4.2                   knitr_1.22                  pillar_1.3.1                codetools_0.2-16            XML_3.98-1.19              </span></a>
<a class="sourceLine" id="cb5-32" data-line-number="32"><span class="co">#&gt; [64] glue_1.3.1                  evaluate_0.13               data.table_1.12.2           modelr_0.1.4                foreach_1.4.4               cellranger_1.1.0            gtable_0.3.0               </span></a>
<a class="sourceLine" id="cb5-33" data-line-number="33"><span class="co">#&gt; [71] assertthat_0.2.1            xfun_0.5                    broom_0.5.1                 roxygen2_6.1.1              iterators_1.0.10            GenomicAlignments_1.18.1    memoise_1.1.0</span></a></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li><a href="#running-repguide">Running Repguide</a></li>
      <li><a href="#compare-predicted-and-experimental-binding-sites">Compare predicted and experimental binding sites</a></li>
      <li><a href="#plotting-the-results">Plotting the results</a></li>
      <li><a href="#references">References</a></li>
      <li><a href="#session-information">Session information</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by <a href="https://scholar.google.de/citations?user=ossdDj4AAAAJ&amp;hl=en&amp;oi=ao">David Brocks</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
