<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Repguide vignette • Repguide</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Repguide vignette">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Repguide</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="../articles/Repguide.html">
    <span class="fa fa-play-circle"></span>
     
    Get started
  </a>
</li>
<li>
  <a href="../articles/benchmark.html">
    <span class="fa fa-bar-chart"></span>
     
    Benchmark
  </a>
</li>
<li>
  <a href="../articles/guide_scoring.html">
    <span class="fa fa-star"></span>
     
    Guide scoring
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-th-list"></span>
     
    Reference
  </a>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-news"></span>
     
    News
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://bitbucket.org/tanaylab/repguide/src/default/">
    <span class="fa fa-bitbucket-square"></span>
     
    Bitbucket
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Repguide vignette</h1>
                        <h4 class="author">David Brocks</h4>
            
            <h4 class="date">2019-07-31</h4>
      
      
      <div class="hidden name"><code>Repguide.Rmd</code></div>

    </div>

    
    
<div id="workflow" class="section level1">
<h1 class="hasAnchor">
<a href="#workflow" class="anchor"></a>Workflow</h1>
<p>In this tutorial, we are going to design guideRNAs against long terminal repeats (LTRs) originating from the human endogenous retrovirus 9. By following this workflow, you will understand the basic steps of the Repguide package and learn how to adapt them to your custom requirements.</p>
<div id="getting-started" class="section level2">
<h2 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h2>
<div id="genome-assembly" class="section level3">
<h3 class="hasAnchor">
<a href="#genome-assembly" class="anchor"></a>Genome assembly</h3>
<p>Repguide requires the genome assembly of your organism of choice stored as a <em>BSgenome</em> object. You can retrieve the full list of supported genomes by typing <code><a href="https://www.rdocumentation.org/packages/BSgenome/topics/available.genomes">BSgenome::available.genomes()</a></code> or create a custom <em>BSgenome</em> object following the <a href="https://www.bioconductor.org/packages//2.7/bioc/vignettes/BSgenome/inst/doc/BSgenomeForge.pdf">instructions</a>.</p>
<p>Since we are designing guideRNAs against LTRs of the human genome, we will first install the UCSC hg38 assembly.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">BiocManager<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/BiocManager/topics/install">install</a></span>(<span class="st">"BSgenome.Hsapiens.UCSC.hg38"</span>)</a></code></pre></div>
<p><strong>Important</strong> <strong>note</strong>: Do not use <strong>repeat-masked</strong> <em>BSgenome</em> objects (contain ‘masked’ suffix, e.g. <em>BSgenome.Hsapiens.UCSC.hg38.masked</em>)!</p>
</div>
<div id="genome-annotation" class="section level3">
<h3 class="hasAnchor">
<a href="#genome-annotation" class="anchor"></a>Genome annotation</h3>
<p>In addition to the assembled genome sequence, Repguide imports genomic coordinates of transposable elements (mandatory) and cis regulatory features (optional). A convient solution is to download transposon coordinates using the <a href="https://genome.ucsc.edu/cgi-bin/hgTables">UCSC table browser</a> by selecting your genome of choice, group = Repeats, track = RepeatMasker, and output format = all fields from selected table. You can provide custom annotation as long as it provides the basic information about chromosome, start, end, strand, repname (family identifier), and te_id (unique locus identifier). In this tutorial, we will show you how to import such information using the provided example datasets.</p>
</div>
<div id="bowtie-aligner" class="section level3">
<h3 class="hasAnchor">
<a href="#bowtie-aligner" class="anchor"></a>Bowtie aligner</h3>
<p>Repguide uses a R implementation of the short read aligner <a href="http://bowtie-bio.sourceforge.net/index.shtml">Bowtie</a> to map potential guideRNAs against the genome. Bowtie requires a FM index based on the Burrows-Wheeler transform to do so. In case Repguide does not find the corresponding bowtie index files to your chosen genome, it will automatically create them. Since this takes about an hour (depends on workstation and genome), you can also <a href="ftp://ftp.ccb.jhu.edu/pub/data/bowtie_indexes/">download</a> the appropriate index files (if available), rename them following the format (BSgenome + suffix, see below) and provide the directory in the <code>refdir</code> argument of the <code><a href="../reference/createGuideSet.html">createGuideSet()</a></code> function.</p>
<p>Example naming of the human hg38 bowtie indeces:</p>
<ul>
<li>BSgenome.Hsapiens.UCSC.hg38.1.ebwt<br>
</li>
<li>BSgenome.Hsapiens.UCSC.hg38.2.ebwt<br>
</li>
<li>BSgenome.Hsapiens.UCSC.hg38.3.ebwt (not required)<br>
</li>
<li>BSgenome.Hsapiens.UCSC.hg38.4.ebwt (not required)<br>
</li>
<li>BSgenome.Hsapiens.UCSC.hg38.rev.1.ebwt<br>
</li>
<li>BSgenome.Hsapiens.UCSC.hg38.rev.2.ebwt</li>
</ul>
</div>
</div>
<div id="creating-a-guideset" class="section level2">
<h2 class="hasAnchor">
<a href="#creating-a-guideset" class="anchor"></a>Creating a guideSet</h2>
<p>We start the workflow by loading Repguide and the human hg38 <em>BSgenome</em> object into our R environment:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(Repguide)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(BSgenome.Hsapiens.UCSC.hg38)</a></code></pre></div>
<p>We then create a <em>guideSet</em> object, which is a S4 class to store genomic information and results of later functions. To create such object, we have to define the genome assembly and provide annotation about transposable elements and cis regulatory features (e.g. promoters/enhancers). While the latter is optional, we highly recommend it for optimal guideRNA design. In this tutorial, we will use the hg38 <a href="https://genome.ucsc.edu/cgi-bin/hgTables">UCSC</a> repeatmasker output and <a href="http://fantom.gsc.riken.jp/5/">FANTOM5 TSS annotation</a>. We additionally blacklist promoter regions of essential human genes (see <code><a href="../reference/essential_genes.html">?essential_genes</a></code> for further information). Those three datasets are included in the Repguide package. We can also specify the path to the corresponding bowtie index directory using the <code>refdir</code> parameter. If <code>refdir = NULL</code> (the default), Repguide will search for bowtie indeces in the package installation directory and create the indeces there if not found.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co"># Path to FANTOM5 hg38 TSS bed file (provided)</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2">tss_path &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="dt">package =</span> <span class="st">'Repguide'</span>, </a>
<a class="sourceLine" id="cb3-3" data-line-number="3">                        <span class="st">'extdata'</span>, </a>
<a class="sourceLine" id="cb3-4" data-line-number="4">                        <span class="st">'hg38_fair+new_CAGE_peaks_phase1and2.bed.gz'</span>)</a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co"># Path to RepeatMasker hg38 repeat annotation (provided, LTRs only)</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6">repeats_path &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="dt">package =</span> <span class="st">'Repguide'</span>, </a>
<a class="sourceLine" id="cb3-7" data-line-number="7">                            <span class="st">'extdata'</span>, </a>
<a class="sourceLine" id="cb3-8" data-line-number="8">                            <span class="st">'hg38_ucsc_rmsk_ltr.txt.gz'</span>)</a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="co"># Path to essential genes (provided)</span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10">essentials_path &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="dt">package =</span> <span class="st">'Repguide'</span>, </a>
<a class="sourceLine" id="cb3-11" data-line-number="11">                               <span class="st">'extdata'</span>, </a>
<a class="sourceLine" id="cb3-12" data-line-number="12">                               <span class="st">'hg38_essential_gene_tss_coords.bed'</span>)</a>
<a class="sourceLine" id="cb3-13" data-line-number="13"></a>
<a class="sourceLine" id="cb3-14" data-line-number="14">gs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/createGuideSet.html">createGuideSet</a></span>(<span class="dt">genome =</span> Hsapiens,           <span class="co"># Hsapiens is the short version of BSgenome.Hsapiens.UCSC.hg38</span></a>
<a class="sourceLine" id="cb3-15" data-line-number="15">                     <span class="dt">tes =</span> repeats_path,          <span class="co"># TE annotation</span></a>
<a class="sourceLine" id="cb3-16" data-line-number="16">                     <span class="dt">cis =</span> tss_path,              <span class="co"># TSS and/or enhancer annotation </span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17">                     <span class="dt">temp =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/tempfile">tempdir</a></span>(),            <span class="co"># Directory for temporary files</span></a>
<a class="sourceLine" id="cb3-18" data-line-number="18">                     <span class="dt">blacklist =</span> essentials_path) <span class="co"># Regions to blacklist</span></a>
<a class="sourceLine" id="cb3-19" data-line-number="19"><span class="co">#&gt; Created new guideSet at Wed Jul 31 13:29:40 2019</span></a>
<a class="sourceLine" id="cb3-20" data-line-number="20">gs                     </a>
<a class="sourceLine" id="cb3-21" data-line-number="21"><span class="co">#&gt; guideSet object of BSgenome.Hsapiens.UCSC.hg38</span></a>
<a class="sourceLine" id="cb3-22" data-line-number="22"><span class="co">#&gt; with 13944 blacklisted regions</span></a>
<a class="sourceLine" id="cb3-23" data-line-number="23"><span class="co">#&gt; with 687600 loci from 578 families</span></a>
<a class="sourceLine" id="cb3-24" data-line-number="24"><span class="co">#&gt; with 0 targets</span></a>
<a class="sourceLine" id="cb3-25" data-line-number="25"><span class="co">#&gt; with 0 valid guides</span></a>
<a class="sourceLine" id="cb3-26" data-line-number="26"><span class="co">#&gt; with 0 combinations</span></a>
<a class="sourceLine" id="cb3-27" data-line-number="27"><span class="co">#&gt; with 0 QC plots</span></a>
<a class="sourceLine" id="cb3-28" data-line-number="28"><span class="co">#&gt; registered 67 cores</span></a>
<a class="sourceLine" id="cb3-29" data-line-number="29"><span class="co">#&gt; matching bowtie index found: FALSE</span></a></code></pre></div>
</div>
<div id="adding-target-loci" class="section level2">
<h2 class="hasAnchor">
<a href="#adding-target-loci" class="anchor"></a>Adding target loci</h2>
<p>The generated <em>guideSet</em> contains all annotation required to design guideRNAs against LTRs of the human genome. To add target sites, we can either provide names of repeat families we wish to target or a GRanges object with coordinates of the loci (<strong>Important</strong> <strong>note</strong>: Such a GRanges object requires <strong>repname</strong> and <strong>te_id</strong> metadata columns compatible with the <em>guideSet’s</em> TE annotation, see examples of <code>?addTargets()</code> function).</p>
<p>We can retrieve the full list of available repeat families using the <code><a href="../reference/repnames.html">repnames()</a></code> accessor function or search for specific families that match a specific pattern. For example, LTRs originating from the human endogenous retrovirus 9 all conveniently share a common ‘LTR12’ prefix (The <a href="http://dfam.org/entry/DF0000399/relationships"><em>Dfam</em></a> database allows open access to transposon genetic relationships), allowing us to easily retrieve all its members by using regular expression:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">families &lt;-<span class="st"> </span><span class="kw"><a href="../reference/repnames.html">repnames</a></span>(gs, <span class="dt">pattern =</span> <span class="st">'LTR12'</span>)</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">families</a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="co">#&gt; [1] "LTR12F" "LTR12C" "LTR12E" "LTR12"  "LTR12D" "LTR12B" "LTR12_"</span></a></code></pre></div>
<p>We now add these families as targets to our previously generated <em>guideSet</em>:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">gs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/addTargets.html">addTargets</a></span>(gs, </a>
<a class="sourceLine" id="cb5-2" data-line-number="2">                 <span class="dt">targets =</span> families) <span class="co"># Either character vector of repname identifier or GRanges object</span></a></code></pre></div>
<p>To inspect targeted families more closely or to design guideRNAs against specific parts of the families’ consensus model, we can compute or import multiple sequence alignments. Repguide computes a rough multiple sequence alignment by weighted sampling (~2% from a families’ total genomic sequences). While this approach is fast and usually sufficient to generate a representative consensus model, it might fail for highly polymorphic families. Therefore, Repguide also supports reading pre-computed multiple sequence alignments stored in fasta format. To import several such alignments, we first need to create a data.frame containing the family identifier of the alignment and the corresponding file’s location:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="co"># Path to data directory containg multiple sequence alignment files</span></a>
<a class="sourceLine" id="cb6-2" data-line-number="2">data_dir &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="dt">package =</span> <span class="st">'Repguide'</span>, <span class="st">'extdata'</span>)</a>
<a class="sourceLine" id="cb6-3" data-line-number="3"></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="co"># Return paths to files ending with *.fasta.gz in that data directory</span></a>
<a class="sourceLine" id="cb6-5" data-line-number="5">msa_files &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list.files">dir</a></span>(data_dir, <span class="dt">full =</span> <span class="ot">TRUE</span>, <span class="dt">pattern =</span> <span class="st">'fasta.gz$'</span>)</a>
<a class="sourceLine" id="cb6-6" data-line-number="6"></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="co"># Create data.frame with repname and corresponding file path</span></a>
<a class="sourceLine" id="cb6-8" data-line-number="8">msa_df &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="st">'repname'</span> =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'LTR12B'</span>, <span class="st">'LTR12D'</span>), <span class="st">'path'</span> =<span class="st"> </span>msa_files)</a>
<a class="sourceLine" id="cb6-9" data-line-number="9">msa_df</a>
<a class="sourceLine" id="cb6-10" data-line-number="10"><span class="co">#&gt;   repname                                                                                           path</span></a>
<a class="sourceLine" id="cb6-11" data-line-number="11"><span class="co">#&gt; 1  LTR12B /home/davidbr/R/x86_64-pc-linux-gnu-library/3.5/Repguide/extdata/LTR12B_mafft_aligned.fasta.gz</span></a>
<a class="sourceLine" id="cb6-12" data-line-number="12"><span class="co">#&gt; 2  LTR12D /home/davidbr/R/x86_64-pc-linux-gnu-library/3.5/Repguide/extdata/LTR12D_mafft_aligned.fasta.gz</span></a>
<a class="sourceLine" id="cb6-13" data-line-number="13"></a>
<a class="sourceLine" id="cb6-14" data-line-number="14">gs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/addAlignments.html">addAlignments</a></span>(gs,                 <span class="co"># our guideSet</span></a>
<a class="sourceLine" id="cb6-15" data-line-number="15">                    <span class="dt">files =</span> msa_df,     <span class="co"># our data.frame</span></a>
<a class="sourceLine" id="cb6-16" data-line-number="16">                    <span class="dt">max_gap_freq =</span> <span class="fl">0.8</span>) <span class="co"># Removes columns with more than 80% gaps in the alignment</span></a>
<a class="sourceLine" id="cb6-17" data-line-number="17"><span class="co">#&gt; Computing alignments for LTR12, LTR12_, LTR12C, LTR12E, LTR12F</span></a>
<a class="sourceLine" id="cb6-18" data-line-number="18"><span class="co">#&gt; Aligning 19 sequences</span></a>
<a class="sourceLine" id="cb6-19" data-line-number="19"><span class="co">#&gt; Aligning 18 sequences</span></a>
<a class="sourceLine" id="cb6-20" data-line-number="20"><span class="co">#&gt; Aligning 57 sequences</span></a>
<a class="sourceLine" id="cb6-21" data-line-number="21"><span class="co">#&gt; Aligning 10 sequences</span></a>
<a class="sourceLine" id="cb6-22" data-line-number="22"><span class="co">#&gt; Aligning 18 sequences</span></a>
<a class="sourceLine" id="cb6-23" data-line-number="23"><span class="co">#&gt; Importing alignments for LTR12B, LTR12D</span></a></code></pre></div>
<p>Having added the targets and alignments to our <em>guideSet</em>, we can now generate several QC reports and inspect the targeted families more closely:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">gs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/plotTargets.html">plotTargets</a></span>(gs) <span class="co"># stores QC plots in guideSet and prints them to screen</span></a></code></pre></div>
<p><img src="Repguide_files/figure-html/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;"></p>
<p>The output generates general statistics on selected families and optional information that requires a multiple sequence alignment (see <code><a href="../reference/plotTargets.html">?plotTargets</a></code> for full annotation of the panels). As we can see from the figures, members of the LTR12C family are by far most prevalent and largest in size (panels <strong>a</strong> to <strong>c</strong>). Since we also calculated and added alignments, the output further visualizes the multiple sequence alignment (<strong>d</strong>), the density of CpG dinucleotides across it (<strong>e</strong>) and includes information on putative consensus transcription start sites (TSSs) (<strong>f</strong>). We can use this information for example, to optimize the design of our guideRNAs by targeting CpG dense regions close to the putative TSS.</p>
</div>
<div id="adding-guidernas" class="section level2">
<h2 class="hasAnchor">
<a href="#adding-guidernas" class="anchor"></a>Adding guideRNAs</h2>
<p>After we are satisfied with the loci/families we wish to target, we continue to compute all possible guideRNAs fulfilling any user-defined criteria (e.g. GC content, target/off-target score, or binding position on consensus models). In this tutorial, we will restrict ourselves to the design of 19 bp guideRNAs with a 3’ NGG PAM. We further tell Repguide to score guides based on their binding to CpG dense parts of the LTR12B and LTR12E families.</p>
<p>For that purpose, we have to first create a simple data.frame providing coordinates on the consensus we wish to target. The <code><a href="../reference/plotTargets.html">plotTargets()</a></code> report can assist you in deciding what start and end coordinates to supply.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">consensus_df &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="st">'repname'</span> =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'LTR12B'</span>, <span class="st">'LTR12E'</span>), </a>
<a class="sourceLine" id="cb8-2" data-line-number="2">                           <span class="st">'start'</span> =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>, <span class="dv">1</span>), </a>
<a class="sourceLine" id="cb8-3" data-line-number="3">                           <span class="st">'end'</span> =<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">200</span>, <span class="dv">750</span>))</a>
<a class="sourceLine" id="cb8-4" data-line-number="4">consensus_df                           </a>
<a class="sourceLine" id="cb8-5" data-line-number="5"><span class="co">#&gt;   repname start end</span></a>
<a class="sourceLine" id="cb8-6" data-line-number="6"><span class="co">#&gt; 1  LTR12B     1 200</span></a>
<a class="sourceLine" id="cb8-7" data-line-number="7"><span class="co">#&gt; 2  LTR12E     1 750</span></a></code></pre></div>
<p><em>Note</em> <em>that</em> <em>guide</em> <em>target</em> <em>binding</em> <em>sites</em> <em>outside</em> <em>the</em> <em>consensus</em> <em>range</em> <em>will</em> <em>be</em> <em>scored</em> <em>neutrally</em>.</p>
<p>We then call the <code><a href="../reference/addGuides.html">addGuides()</a></code> function to compute our guide universe. More specifically, this function first calls <a href="http://www.genome.umd.edu/jellyfish.html"><em>jellyfish</em></a> to calculate the 19mer (in our scenario) universe of specified targets, selects kmers with proper PAM sequence, and finds and annotates their putative genomic binding sites. Since most kmers are genetically similar and hence show redundanct binding profiles, Repguide finally clusters kmers into groups with homogenous binding profiles and marks the best kmer per cluster (the number of groups and selection procedure can be specified with the <code>n_clust</code> and <code>alpha</code> arguments of the <code><a href="../reference/addGuides.html">addGuides()</a></code> function). Depending on the target sequence complexity, guideRNA length, and allowed mismatches, the computation may take a bit.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">gs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/addGuides.html">addGuides</a></span>(gs,                             <span class="co"># our guideSet</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2">                <span class="dt">n_mismatches =</span> <span class="dv">2</span>,               <span class="co"># max allowed mismatch of reported binding sites</span></a>
<a class="sourceLine" id="cb9-3" data-line-number="3">                <span class="dt">min_Son =</span> <span class="dv">100</span>,                  <span class="co"># minimal score requirement for valid guides</span></a>
<a class="sourceLine" id="cb9-4" data-line-number="4">                <span class="dt">guide_length =</span> <span class="dv">19</span>,              <span class="co"># length of the guideRNAs to design</span></a>
<a class="sourceLine" id="cb9-5" data-line-number="5">                <span class="dt">consensus_range =</span> consensus_df, <span class="co"># restrict guideRNA design to parts on consensus</span></a>
<a class="sourceLine" id="cb9-6" data-line-number="6">                <span class="dt">gc_content =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="fl">0.4</span>, <span class="fl">0.8</span>))       <span class="co"># allowed guide GC content (between 40 and 80 percent)</span></a>
<a class="sourceLine" id="cb9-7" data-line-number="7"><span class="co">#&gt; [1] "Computing kmer universe"</span></a>
<a class="sourceLine" id="cb9-8" data-line-number="8"><span class="co">#&gt; [1] "Select kmers with proper PAM"</span></a>
<a class="sourceLine" id="cb9-9" data-line-number="9"><span class="co">#&gt; [1] "Creating bowtie index, this may take a while"</span></a>
<a class="sourceLine" id="cb9-10" data-line-number="10"><span class="co">#&gt; writing chr1 sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-11" data-line-number="11"><span class="co">#&gt; writing chr2 sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-12" data-line-number="12"><span class="co">#&gt; writing chr3 sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-13" data-line-number="13"><span class="co">#&gt; writing chr4 sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-14" data-line-number="14"><span class="co">#&gt; writing chr5 sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-15" data-line-number="15"><span class="co">#&gt; writing chr6 sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-16" data-line-number="16"><span class="co">#&gt; writing chr7 sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-17" data-line-number="17"><span class="co">#&gt; writing chr8 sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-18" data-line-number="18"><span class="co">#&gt; writing chr9 sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-19" data-line-number="19"><span class="co">#&gt; writing chr10 sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-20" data-line-number="20"><span class="co">#&gt; writing chr11 sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-21" data-line-number="21"><span class="co">#&gt; writing chr12 sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-22" data-line-number="22"><span class="co">#&gt; writing chr13 sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-23" data-line-number="23"><span class="co">#&gt; writing chr14 sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-24" data-line-number="24"><span class="co">#&gt; writing chr15 sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-25" data-line-number="25"><span class="co">#&gt; writing chr16 sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-26" data-line-number="26"><span class="co">#&gt; writing chr17 sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-27" data-line-number="27"><span class="co">#&gt; writing chr18 sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-28" data-line-number="28"><span class="co">#&gt; writing chr19 sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-29" data-line-number="29"><span class="co">#&gt; writing chr20 sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-30" data-line-number="30"><span class="co">#&gt; writing chr21 sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-31" data-line-number="31"><span class="co">#&gt; writing chr22 sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-32" data-line-number="32"><span class="co">#&gt; writing chrX sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-33" data-line-number="33"><span class="co">#&gt; writing chrY sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-34" data-line-number="34"><span class="co">#&gt; writing chrM sequence to file ... OK</span></a>
<a class="sourceLine" id="cb9-35" data-line-number="35"><span class="co">#&gt; [1] "Aligning kmers against: BSgenome.Hsapiens.UCSC.hg38"</span></a>
<a class="sourceLine" id="cb9-36" data-line-number="36"><span class="co">#&gt; [1] "Annotating guides"</span></a>
<a class="sourceLine" id="cb9-37" data-line-number="37"><span class="co">#&gt; [1] "Adding position on consensus"</span></a>
<a class="sourceLine" id="cb9-38" data-line-number="38"><span class="co">#&gt; [1] "Computing guide scores"</span></a>
<a class="sourceLine" id="cb9-39" data-line-number="39"><span class="co">#&gt; [1] "Blacklisting kmers"</span></a>
<a class="sourceLine" id="cb9-40" data-line-number="40"><span class="co">#&gt; Computing score blacklisting threshold</span></a>
<a class="sourceLine" id="cb9-41" data-line-number="41"><span class="co">#&gt; Set max_Soff to 15.77</span></a>
<a class="sourceLine" id="cb9-42" data-line-number="42"><span class="co">#&gt; Clustering kmers</span></a>
<a class="sourceLine" id="cb9-43" data-line-number="43"><span class="co">#&gt; [1] "Clustering 3752 kmers into 11 groups"</span></a>
<a class="sourceLine" id="cb9-44" data-line-number="44"><span class="co">#&gt; [1] "Clustering 5101 loci into 11 groups"</span></a></code></pre></div>
<p>We can now generate a QC report and sanity check the results by entering:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">gs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/plotGuides.html">plotGuides</a></span>(gs)</a></code></pre></div>
<p><img src="Repguide_files/figure-html/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;"></p>
<p>We can see from the report that while most of the potential guides have the desired GC content, about half of them do not pass our minimal score requirements (<strong>a</strong> - <strong>c</strong>). Panel <strong>d</strong> shows the Sbind score distribution by the total number of mismatches of a guide’s genomic complement and <strong>e</strong> summarizes the frequency of off-targets by their basepair distance to the nearest cis regulatory element. The dotted red line indicates the conversion of basepair distances to the corresponding cis score (Scis) of an off-target site. Panels <strong>f</strong> and <strong>g</strong> show the clustering of guideRNAs by their target binding profile and selection of the best guide per cluster, respectively. For full annotation of the panels, we refer to the documentation page of the <code><a href="../reference/plotGuides.html">plotGuides()</a></code> function.</p>
<p>To adjust the guide selection or clustering steps, you can use the <code><a href="../reference/selGuides.html">selGuides()</a></code> or <code><a href="../reference/clustGuides.html">clustGuides()</a></code> functions, respectively, without re-running the entire <code><a href="../reference/addGuides.html">addGuides()</a></code> pipeline.</p>
</div>
<div id="calculating-guide-combinations" class="section level2">
<h2 class="hasAnchor">
<a href="#calculating-guide-combinations" class="anchor"></a>Calculating guide combinations</h2>
<p>Multiplexing of several distinct guideRNAs for combined delivery may maximize recruitment of dCAS9 to polymorphic multi-copy families, as recently shown for human LTR5HS elements [1]. However, selecting the optimal set of guideRNAs for maximal on- and minimal off-targeting is not trivial. Repguide first computes the binding profiles of all possible combinations of previously selected guide representatives, i.e. the best guides per cluster. The best combination then serves as seed to initialize a greedy algorithm (can be disabled by setting <code>greedy = FALSE</code>) to further optimize the set. The <code>alpha</code> parameter controls the weight of off-targets, i.e. <code>alpha = 0</code> maximizes target coverage regardless of off-binding score while a larger <code>alpha</code> penalizes unspecific sets of guideRNAs (see <a href="https://tanaylab.bitbucket.io/Repguide/articles/guide_scoring.html">guideRNA scoring</a> for further information).</p>
<p>We calculate binding profiles of all possible combinations for up to 7 distinct guides by typing:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">gs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/addCombinations.html">addCombinations</a></span>(gs,              <span class="co"># our guideSet</span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2">                      <span class="dt">iterations =</span> <span class="dv">10</span>, <span class="co"># number of greedy search iterations</span></a>
<a class="sourceLine" id="cb11-3" data-line-number="3">                      <span class="dt">greedy =</span> <span class="ot">TRUE</span>,   <span class="co"># run greedy algorithm</span></a>
<a class="sourceLine" id="cb11-4" data-line-number="4">                      <span class="dt">alpha =</span> <span class="dv">100</span>,     <span class="co"># off-target score penalty coefficient</span></a>
<a class="sourceLine" id="cb11-5" data-line-number="5">                      <span class="dt">max_guides =</span> <span class="dv">7</span>)  <span class="co"># maximal number of guides to consider</span></a>
<a class="sourceLine" id="cb11-6" data-line-number="6"><span class="co">#&gt; Running greedy optimization on 8826 x 3752 dimensional matrix</span></a></code></pre></div>
<p>Having added the score statistics for computed combinations to our <em>guideSet</em>, we can inspect the results by entering:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1">gs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/plotCombinations.html">plotCombinations</a></span>(gs)</a></code></pre></div>
<p><img src="Repguide_files/figure-html/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;"></p>
<p>The table in panel <strong>a</strong> summarizes the on- and off-targets per guide combination. We can see that the on-target binding score plateaus with increasing number of distinct guides (<strong>a</strong> and <strong>b</strong>), meaning the addition of further guideRNAs to the delivery construct would provide little extra benefit. Panel <strong>c</strong> visualizes the performance of the greedy algorithm and panel <strong>d</strong> shows on- vs off-target scores for all computed guide combinations. Closer inspection of predicted off-targets is shown in panels <strong>e</strong> and <strong>f</strong> which summarize off-target binding coverage by genomic feature and distance to nearest cis regulatory element, respectively. The target loci coverage per number of guides is shown in panel (<strong>g</strong>). Note how several loci are covered by more than one binding site. Panel <strong>h</strong> shows the predicted binding position along the consensus sequences (in 250 bp bins) by the number of guides. Full annotation of the figure is available in the documentation of the <code><a href="../reference/plotCombinations.html">plotCombinations()</a></code> function.</p>
</div>
<div id="exporting-the-results" class="section level2">
<h2 class="hasAnchor">
<a href="#exporting-the-results" class="anchor"></a>Exporting the results</h2>
<p>We can finally save the relevant results and plots of our guideSet to a specified directory:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw"><a href="../reference/export.html">export</a></span>(gs,</a>
<a class="sourceLine" id="cb13-2" data-line-number="2">       <span class="dt">outdir =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/tempfile">tempdir</a></span>(), </a>
<a class="sourceLine" id="cb13-3" data-line-number="3">       <span class="dt">dpi =</span> <span class="dv">320</span>,         <span class="co"># image resolution,</span></a>
<a class="sourceLine" id="cb13-4" data-line-number="4">       <span class="dt">full =</span> <span class="ot">TRUE</span>,       <span class="co"># exports stats for all guides, not only selected ones</span></a>
<a class="sourceLine" id="cb13-5" data-line-number="5">       <span class="dt">workspace =</span> <span class="ot">FALSE</span>) <span class="co"># don't save guideSet as workspace</span></a></code></pre></div>
<p>This command will create a subdirectory in the specified outdir (or current working directory if <code>force = TRUE</code>) and export the results.</p>
</div>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<p>[1] Daniel R Fuentes, Tomek Swigut, Joanna Wysocka. Systematic perturbation of retroviral LTRs reveals widespread long-range effects on human gene regulation. eLife (2018).</p>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h1>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/sessionInfo">sessionInfo</a></span>()</a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="co">#&gt; R version 3.5.3 (2019-03-11)</span></a>
<a class="sourceLine" id="cb14-3" data-line-number="3"><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></a>
<a class="sourceLine" id="cb14-4" data-line-number="4"><span class="co">#&gt; Running under: CentOS Linux 7 (Core)</span></a>
<a class="sourceLine" id="cb14-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb14-6" data-line-number="6"><span class="co">#&gt; Matrix products: default</span></a>
<a class="sourceLine" id="cb14-7" data-line-number="7"><span class="co">#&gt; BLAS/LAPACK: /net/mraid14/export/data/users/eladch/tools/CO7/mkl/2018.3/compilers_and_libraries_2018.3.222/linux/mkl/lib/intel64_lin/libmkl_gf_lp64.so</span></a>
<a class="sourceLine" id="cb14-8" data-line-number="8"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb14-9" data-line-number="9"><span class="co">#&gt; locale:</span></a>
<a class="sourceLine" id="cb14-10" data-line-number="10"><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8      </span></a>
<a class="sourceLine" id="cb14-11" data-line-number="11"><span class="co">#&gt;  [8] LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></a>
<a class="sourceLine" id="cb14-12" data-line-number="12"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb14-13" data-line-number="13"><span class="co">#&gt; attached base packages:</span></a>
<a class="sourceLine" id="cb14-14" data-line-number="14"><span class="co">#&gt; [1] stats4    parallel  stats     graphics  grDevices utils     datasets  methods   base     </span></a>
<a class="sourceLine" id="cb14-15" data-line-number="15"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb14-16" data-line-number="16"><span class="co">#&gt; other attached packages:</span></a>
<a class="sourceLine" id="cb14-17" data-line-number="17"><span class="co">#&gt;  [1] BSgenome.Hsapiens.UCSC.hg38_1.4.1 BSgenome_1.50.0                   rtracklayer_1.42.2                Biostrings_2.50.2                 XVector_0.22.0                   </span></a>
<a class="sourceLine" id="cb14-18" data-line-number="18"><span class="co">#&gt;  [6] GenomicRanges_1.34.0              GenomeInfoDb_1.18.2               IRanges_2.16.0                    S4Vectors_0.20.1                  BiocGenerics_0.28.0              </span></a>
<a class="sourceLine" id="cb14-19" data-line-number="19"><span class="co">#&gt; [11] Repguide_0.0.1.2                 </span></a>
<a class="sourceLine" id="cb14-20" data-line-number="20"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb14-21" data-line-number="21"><span class="co">#&gt; loaded via a namespace (and not attached):</span></a>
<a class="sourceLine" id="cb14-22" data-line-number="22"><span class="co">#&gt;  [1] nlme_3.1-137                bitops_1.0-6                matrixStats_0.54.0          fs_1.2.7                    doMC_1.3.5                  bit64_0.9-7                 rprojroot_1.3-2            </span></a>
<a class="sourceLine" id="cb14-23" data-line-number="23"><span class="co">#&gt;  [8] tools_3.5.3                 backports_1.1.4             R6_2.4.0                    DBI_1.0.0                   lazyeval_0.2.2              colorspace_1.4-1            gridExtra_2.3              </span></a>
<a class="sourceLine" id="cb14-24" data-line-number="24"><span class="co">#&gt; [15] tidyselect_0.2.5            bit_1.1-14                  compiler_3.5.3              Biobase_2.42.0              xml2_1.2.0                  desc_1.2.0                  DelayedArray_0.8.0         </span></a>
<a class="sourceLine" id="cb14-25" data-line-number="25"><span class="co">#&gt; [22] labeling_0.3                slam_0.1-45                 phylogram_2.1.0             scales_1.0.0                pkgdown_1.3.0               commonmark_1.7              stringr_1.4.0              </span></a>
<a class="sourceLine" id="cb14-26" data-line-number="26"><span class="co">#&gt; [29] digest_0.6.19               Rsamtools_1.34.1            rmarkdown_1.12              R.utils_2.8.0               pkgconfig_2.0.2             htmltools_0.3.6             rlang_0.3.2                </span></a>
<a class="sourceLine" id="cb14-27" data-line-number="27"><span class="co">#&gt; [36] RSQLite_2.1.1               BiocParallel_1.16.6         dplyr_0.8.0.1               R.oo_1.22.0                 RCurl_1.95-4.12             magrittr_1.5                GenomeInfoDbData_1.2.0     </span></a>
<a class="sourceLine" id="cb14-28" data-line-number="28"><span class="co">#&gt; [43] Matrix_1.2-16               Rcpp_1.0.1                  munsell_0.5.0               DECIPHER_2.10.2             ape_5.3                     Rbowtie_1.22.0              R.methodsS3_1.7.1          </span></a>
<a class="sourceLine" id="cb14-29" data-line-number="29"><span class="co">#&gt; [50] stringi_1.4.3               yaml_2.2.0                  MASS_7.3-51.1               SummarizedExperiment_1.12.0 zlibbioc_1.28.0             plyr_1.8.4                  grid_3.5.3                 </span></a>
<a class="sourceLine" id="cb14-30" data-line-number="30"><span class="co">#&gt; [57] blob_1.1.1                  forcats_0.4.0               crayon_1.3.4                lattice_0.20-38             cowplot_0.9.4               knitr_1.22                  pillar_1.3.1               </span></a>
<a class="sourceLine" id="cb14-31" data-line-number="31"><span class="co">#&gt; [64] codetools_0.2-16            XML_3.98-1.19               glue_1.3.1                  evaluate_0.13               data.table_1.12.2           kmer_1.1.1                  foreach_1.4.4              </span></a>
<a class="sourceLine" id="cb14-32" data-line-number="32"><span class="co">#&gt; [71] gtable_0.3.0                purrr_0.3.2                 tidyr_0.8.3                 clue_0.3-57                 assertthat_0.2.1            ggplot2_3.1.0               xfun_0.5                   </span></a>
<a class="sourceLine" id="cb14-33" data-line-number="33"><span class="co">#&gt; [78] skmeans_0.2-11              roxygen2_6.1.1              tibble_2.1.1                iterators_1.0.10            GenomicAlignments_1.18.1    memoise_1.1.0               cluster_2.0.7-1</span></a></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#workflow">Workflow</a><ul class="nav nav-pills nav-stacked">
<li><a href="#getting-started">Getting started</a></li>
      <li><a href="#creating-a-guideset">Creating a guideSet</a></li>
      <li><a href="#adding-target-loci">Adding target loci</a></li>
      <li><a href="#adding-guidernas">Adding guideRNAs</a></li>
      <li><a href="#calculating-guide-combinations">Calculating guide combinations</a></li>
      <li><a href="#exporting-the-results">Exporting the results</a></li>
      </ul>
</li>
      <li><a href="#references">References</a></li>
      <li><a href="#session-information">Session information</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by <a href="https://scholar.google.de/citations?user=ossdDj4AAAAJ&amp;hl=en&amp;oi=ao">David Brocks</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
