<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ gpatterns</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">

<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">


<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>

<!-- mathjax -->
<script src='https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
  </head>

  <body>
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">gpatterns</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">Home</a>
</li>
<li>
  <a href="../articles/import.html">
    <span class="fa fa-truck"></span>
     
    Import
  </a>
</li>
<li>
  <a href="../articles/articles/analysis.html">
    <span class="fa fa-bar-chart"></span>
     
    Analyse
  </a>
</li>
<li>
  <a href="../articles/articles/advanced.html">
    <span class="fa fa-stethoscope"></span>
     
    Advanced
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-th-list"></span>
     
    Reference
  </a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://bitbucket.org/tanaylab/gpatterns">
    <span class="fa fa-bitbucket-square"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

      <div class="row">
  <div class="col-md-9">
    <div class="page-header">
      <h1>Function reference</h1>
      <h4>version&nbsp;0.0.0.9000</h4>
    </div>

    <div class="contents">
      <h2 id="section-importing-datasets" class="hasAnchor"><a href="#section-importing-datasets" class="anchor"></a>Importing datasets</h2>
      <p class="section-desc"><p>Functions for importing datasets to gpatterns</p></p>

      
      <p><a href="gpatterns.adjust_read_pos.html">Change tidy_cpgs coordinates to fragments coordinates</a></p>
      <ul>
        <li><code><a href="gpatterns.adjust_read_pos.html">gpatterns.adjust_read_pos</a></code></li>
      </ul>
      
      <p><a href="gpatterns.bissli2_build.html">Create an bowtie2 index for a bisulfite converted genome</a></p>
      <ul>
        <li><code><a href="gpatterns.bissli2_build.html">gpatterns.bissli2_build</a></code></li>
      </ul>
      
      <p><a href="gpatterns.bissli2.html">Align bisulfite converted reads to a reference genome using bowtie2</a></p>
      <ul>
        <li><code><a href="gpatterns.bissli2.html">gpatterns.bissli2</a></code></li>
      </ul>
      
      <p><a href="gpatterns.create_patterns_track.html">Creates patterns attributes of track</a></p>
      <ul>
        <li><code><a href="gpatterns.create_patterns_track.html">gpatterns.create_patterns_track</a></code></li>
      </ul>
      
      <p><a href="gpatterns.get_pipeline_stats.html">Calculate QC statistics for track</a></p>
      <ul>
        <li><code><a href="gpatterns.get_pipeline_stats.html">gpatterns.get_pipeline_stats</a></code></li>
      </ul>
      
      <p><a href="gpatterns.import_from_bam.html">Create a track from bam files.</a></p>
      <ul>
        <li><code><a href="gpatterns.import_from_bam.html">gpatterns.import_from_bam</a></code></li>
      </ul>
      
      <p><a href="gpatterns.import_from_tidy_cpgs.html">Create a track from tidy_cpgs files</a></p>
      <ul>
        <li><code><a href="gpatterns.import_from_tidy_cpgs.html">gpatterns.import_from_tidy_cpgs</a></code></li>
      </ul>
      
      <p><a href="gpatterns.merge_tracks.html">Merge tracks</a></p>
      <ul>
        <li><code><a href="gpatterns.merge_tracks.html">gpatterns.merge_tracks</a></code></li>
      </ul>
      
      <p><a href="gpatterns.separate_strands.html">Separate a track by strands (for QC purposes)</a></p>
      <ul>
        <li><code><a href="gpatterns.separate_strands.html">gpatterns.separate_strands</a></code></li>
      </ul>
          <h2 id="section-average-methylation" class="hasAnchor"><a href="#section-average-methylation" class="anchor"></a>Average methylation</h2>
      <p class="section-desc"><p>Functions for analyzing and plotting average methylation</p></p>

      
      <p><a href="gpatterns.cluster_avg_meth.html">Cluster average methylation of multiple tracks</a></p>
      <ul>
        <li><code><a href="gpatterns.cluster_avg_meth.html">gpatterns.cluster_avg_meth</a></code></li>
      </ul>
      
      <p><a href="gpatterns.count_cpgs.html">count CpGs in intervals</a></p>
      <ul>
        <li><code><a href="gpatterns.count_cpgs.html">gpatterns.count_cpgs</a></code></li>
      </ul>
      
      <p><a href="gpatterns.filter_by_dist.html">Filter intervals set by distance from another intervals set</a></p>
      <ul>
        <li><code><a href="gpatterns.filter_by_dist.html">gpatterns.filter_by_dist</a></code></li>
      </ul>
      
      <p><a href="gpatterns.filter_cpgs.html">Filter intervals by number of CpGs</a></p>
      <ul>
        <li><code><a href="gpatterns.filter_cpgs.html">gpatterns.filter_cpgs</a></code></li>
      </ul>
      
      <p><a href="gpatterns.get_avg_meth.html">Extract average methylation</a></p>
      <ul>
        <li><code><a href="gpatterns.get_avg_meth.html">gpatterns.get_avg_meth</a></code></li>
      </ul>
      
      <p><a href="gpatterns.get_fragment_cov.html">Get fragment coverage</a></p>
      <ul>
        <li><code><a href="gpatterns.get_fragment_cov.html">gpatterns.get_fragment_cov</a></code></li>
      </ul>
      
      <p><a href="gpatterns.get_promoters.html">Define promoter regions</a></p>
      <ul>
        <li><code><a href="gpatterns.get_promoters.html">gpatterns.get_promoters</a></code></li>
      </ul>
      
      <p><a href="gpatterns.global_meth_trend.html">Plot global methylation stratified on other tracks</a></p>
      <ul>
        <li><code><a href="gpatterns.global_meth_trend.html">gpatterns.global_meth_trend</a></code></li>
      </ul>
      
      <p><a href="gpatterns.intervals_coverage.html">Get coverage of intervals in multiple tracks</a></p>
      <ul>
        <li><code><a href="gpatterns.intervals_coverage.html">gpatterns.intervals_coverage</a></code></li>
      </ul>
      
      <p><a href="gpatterns.plot_cg_cor.html">Plot CpG correlation (pearson) as a function of distance between CpGs</a></p>
      <ul>
        <li><code><a href="gpatterns.plot_cg_cor.html">gpatterns.plot_cg_cor</a></code></li>
      </ul>
      
      <p><a href="gpatterns.plot_clustering.html">Plot a heatmap of average methylation clustering</a></p>
      <ul>
        <li><code><a href="gpatterns.plot_clustering.html">gpatterns.plot_clustering</a></code></li>
      </ul>
      
      <p><a href="gpatterns.screen_by_coverage.html">screen intervals by coverage in multiple tracks</a></p>
      <ul>
        <li><code><a href="gpatterns.screen_by_coverage.html">gpatterns.screen_by_coverage</a></code></li>
      </ul>
      
      <p><a href="gpatterns.smoothScatter.html">Plots smoothScatter of average methylation of two tracks</a></p>
      <ul>
        <li><code><a href="gpatterns.smoothScatter.html">gpatterns.smoothScatter</a></code></li>
      </ul>
      
      <p><a href="gpatterns.spatial_meth_trend.html">Plot spatial trend of methylation around intervals</a></p>
      <ul>
        <li><code><a href="gpatterns.spatial_meth_trend.html">gpatterns.spatial_meth_trend</a></code></li>
      </ul>
          <h2 id="section-tidy-cpgs-functions" class="hasAnchor"><a href="#section-tidy-cpgs-functions" class="anchor"></a>Tidy CpGs Functions</h2>
      <p class="section-desc"><p>Functions for retrieving and operating on tidy_cpgs and patterns</p></p>

      
      <p><a href="gpatterns.apply_tidy_cpgs.html">Apply a function on tidy cpgs (internally separated by coordinates)</a></p>
      <ul>
        <li><code><a href="gpatterns.apply_tidy_cpgs.html">gpatterns.apply_tidy_cpgs</a></code></li>
      </ul>
      
      <p><a href="gpatterns.downsample_patterns.html">Downsample patterns table</a></p>
      <ul>
        <li><code><a href="gpatterns.downsample_patterns.html">gpatterns.downsample_patterns</a></code></li>
      </ul>
      
      <p><a href="gpatterns.extract_all.html">Extract pattern data from tracks without specifying coordinates</a></p>
      <ul>
        <li><code><a href="gpatterns.extract_all.html">gpatterns.extract_all</a></code></li>
      </ul>
      
      <p><a href="gpatterns.extract_patterns.html">Extract patterns of a track</a></p>
      <ul>
        <li><code><a href="gpatterns.extract_patterns.html">gpatterns.extract_patterns</a></code></li>
      </ul>
      
      <p><a href="gpatterns.extract.html">Extract pattern data</a></p>
      <ul>
        <li><code><a href="gpatterns.extract.html">gpatterns.extract</a></code></li>
      </ul>
      
      <p><a href="gpatterns.frag_stats.html">transforms patterns to summary statistics: n,n0,n1,nx,nc,meth and epipoly</a></p>
      <ul>
        <li><code><a href="gpatterns.frag_stats.html">gpatterns.frag_stats</a></code></li>
      </ul>
      
      <p><a href="gpatterns.get_tidy_cpgs.html">Get tidy cpgs from a track</a></p>
      <ul>
        <li><code><a href="gpatterns.get_tidy_cpgs.html">gpatterns.get_tidy_cpgs</a></code></li>
      </ul>
          <h2 id="section-pattern-generation-functions" class="hasAnchor"><a href="#section-pattern-generation-functions" class="anchor"></a>Pattern generation functions</h2>
      <p class="section-desc"><p>Functions for generating methylation patterns</p></p>

      
      <p><a href="gpatterns.get_pat_cov.html">Calculate pattern coverage for each CpG</a></p>
      <ul>
        <li><code><a href="gpatterns.get_pat_cov.html">gpatterns.get_pat_cov</a></code></li>
      </ul>
      
      <p><a href="gpatterns.intervs_to_pat_space.html">Generate pattern space for specific genomic loci</a></p>
      <ul>
        <li><code><a href="gpatterns.intervs_to_pat_space.html">gpatterns.intervs_to_pat_space</a></code></li>
      </ul>
      
      <p><a href="gpatterns.tidy_cpgs_2_pat_freq.html">Generate CpG pattern frequency from tidy_cpgs</a></p>
      <ul>
        <li><code><a href="gpatterns.tidy_cpgs_2_pat_freq.html">gpatterns.tidy_cpgs_2_pat_freq</a></code></li>
      </ul>
      
      <p><a href="gpatterns.tidy_cpgs_2_pileup.html">Generate pileup form tidy_cpgs</a></p>
      <ul>
        <li><code><a href="gpatterns.tidy_cpgs_2_pileup.html">gpatterns.tidy_cpgs_2_pileup</a></code></li>
      </ul>
      
      <p><a href="gpatterns.tidy_cpgs_to_pats.html">Extracts patterns given pattern space</a></p>
      <ul>
        <li><code><a href="gpatterns.tidy_cpgs_to_pats.html">gpatterns.tidy_cpgs_to_pats</a></code></li>
      </ul>
      
      <p><a href="gpatterns.track_pat_space.html">Get pattern space of a track</a></p>
      <ul>
        <li><code><a href="gpatterns.track_pat_space.html">gpatterns.track_pat_space</a></code></li>
      </ul>
      
      <p><a href="gpatterns.tracks_to_pat_space.html">Generate pattern space</a></p>
      <ul>
        <li><code><a href="gpatterns.tracks_to_pat_space.html">gpatterns.tracks_to_pat_space</a></code></li>
      </ul>
          <h2 id="section-epipolymorphism-functions" class="hasAnchor"><a href="#section-epipolymorphism-functions" class="anchor"></a>Epipolymorphism functions</h2>
      <p class="section-desc"><p>Functions for analyzing and plotting epipolymorphism</p></p>

      
      <p><a href="gpatterns.border_pattern.html">Returns TRUE if pattern is a 'border' pattern</a></p>
      <ul>
        <li><code><a href="gpatterns.border_pattern.html">gpatterns.border_pattern</a></code></li>
      </ul>
      
      <p><a href="gpatterns.calc_bipolarity.html">Run bipolar model</a></p>
      <ul>
        <li><code><a href="gpatterns.calc_bipolarity.html">gpatterns.calc_bipolarity</a></code></li>
      </ul>
      
      <p><a href="gpatterns.calc_epipoly.html">Calculate epipolymorphism of samples</a></p>
      <ul>
        <li><code><a href="gpatterns.calc_epipoly.html">gpatterns.calc_epipoly</a></code></li>
      </ul>
      
      <p><a href="gpatterns.clean_pattern.html">Returns TRUE if pattern is fully methylated / unmethylated and FALSE if not</a></p>
      <ul>
        <li><code><a href="gpatterns.clean_pattern.html">gpatterns.clean_pattern</a></code></li>
      </ul>
      
      <p><a href="gpatterns.epipoly_plot.html">Plots epipolymorphism</a></p>
      <ul>
        <li><code><a href="gpatterns.epipoly_plot.html">gpatterns.epipoly_plot</a></code></li>
      </ul>
          <h2 id="section-chip-seq-functions" class="hasAnchor"><a href="#section-chip-seq-functions" class="anchor"></a>Chip-seq functions</h2>
      <p class="section-desc"><p>Functions for analyzing chipseq data</p></p>

      
      <p><a href="gpatterns.chip2peaks.html">Calculate peaks of chip-seq tracks</a></p>
      <ul>
        <li><code><a href="gpatterns.chip2peaks.html">gpatterns.chip2peaks</a></code></li>
      </ul>
      
      <p><a href="gpatterns.plot_chip_quantiles.html">Plot chip-seq singal with quantile lines</a></p>
      <ul>
        <li><code><a href="gpatterns.plot_chip_quantiles.html">gpatterns.plot_chip_quantiles</a></code></li>
      </ul>
      
      <p><a href="gpatterns.putative_enhancers.html">Define putative enhancer regions from chip-seq tracks</a></p>
      <ul>
        <li><code><a href="gpatterns.putative_enhancers.html">gpatterns.putative_enhancers</a></code></li>
      </ul>
          <h2 id="section-global-definitions" class="hasAnchor"><a href="#section-global-definitions" class="anchor"></a>Global definitions</h2>
      <p class="section-desc"><p>Set global definitions</p></p>

      
      <p><a href="gpatterns.set_parallel.html">Set parallel threads</a></p>
      <ul>
        <li><code><a href="gpatterns.set_parallel.html">gpatterns.set_parallel</a></code></li>
      </ul>
          <h2 id="section-utils" class="hasAnchor"><a href="#section-utils" class="anchor"></a>Utils</h2>
      <p class="section-desc"><p>Utility functions</p></p>

      
      <p><a href="build_pallette.html">Build a gradient pallete based on colors at specific values.</a></p>
      <ul>
        <li><code><a href="build_pallette.html">build_pallette</a></code></li>
      </ul>
      
      <p><a href="cume_dist_min.html">Return an alternate empirical cumulative distribution, counting only
</a></p>
      <ul>
        <li><code><a href="cume_dist_min.html">cume_dist_min</a></code></li>
      </ul>
      
      <p><a href="gbin_intervals.html">split genomic intervals to bins</a></p>
      <ul>
        <li><code><a href="gbin_intervals.html">gbin_intervals</a></code></li>
      </ul>
      
      <p><a href="gcluster.run2.html">Runs R commands on a cluster that supports SGE</a></p>
      <ul>
        <li><code><a href="gcluster.run2.html">gcluster.run2</a></code></li>
      </ul>
      
      <p><a href="gextract.left_join.html">Returns the result of track expressions evaluation for each of the
</a></p>
      <ul>
        <li><code><a href="gextract.left_join.html">gextract.left_join</a></code></li>
      </ul>
      
      <p><a href="gintervals.filter.html">Filter intervals set by distance to another.
</a></p>
      <ul>
        <li><code><a href="gintervals.filter.html">gintervals.filter</a></code></li>
      </ul>
      
      <p><a href="gintervals.neighbors1.html">Finds neighbors between two sets of intervals (and does not return conflicting column names)</a></p>
      <ul>
        <li><code><a href="gintervals.neighbors1.html">gintervals.neighbors1</a></code></li>
      </ul>
      
      <p><a href="glply.html">Apply a function on track expression in intervals</a></p>
      <ul>
        <li><code><a href="glply.html">glply</a></code></li><li><code><a href="glply.html">gdply</a></code></li>
      </ul>
      
      <p><a href="gpatterns.ls.html">Returns a list of track names</a></p>
      <ul>
        <li><code><a href="gpatterns.ls.html">gpatterns.ls</a></code></li>
      </ul>
      
      <p><a href="gpatterns.track_exists.html">check if tidy_cpgs exist for track</a></p>
      <ul>
        <li><code><a href="gpatterns.track_exists.html">gpatterns.track_exists</a></code></li>
      </ul>
      
      <p><a href="gtrack.array.import_from_df.html">Import array from data frame</a></p>
      <ul>
        <li><code><a href="gtrack.array.import_from_df.html">gtrack.array.import_from_df</a></code></li>
      </ul>
      
      <p><a href="gtrack.import_mappedseq_bam.html">Wrapper around gtrack.import_mappedseq for (multiple) bam files</a></p>
      <ul>
        <li><code><a href="gtrack.import_mappedseq_bam.html">gtrack.import_mappedseq_bam</a></code></li>
      </ul>
      
      <p><a href="gvextract.html">Creates a virtual track and runs gextract</a></p>
      <ul>
        <li><code><a href="gvextract.html">gvextract</a></code></li>
      </ul>
      
      <p><a href="p_smoothScatter.html">Smoothscatter that works with the pipe operator</a></p>
      <ul>
        <li><code><a href="p_smoothScatter.html">p_smoothScatter</a></code></li>
      </ul>
      
      <p><a href="pheatmap1.html">wrapper around pheatmap</a></p>
      <ul>
        <li><code><a href="pheatmap1.html">pheatmap1</a></code></li>
      </ul>
      
      <p><a href="TGL_kmeans.html">Effiecient implementation of kmeans++ algorithm</a></p>
      <ul>
        <li><code><a href="TGL_kmeans.html">TGL_kmeans</a></code></li>
      </ul>
      
      <p><a href="TGL_kmeans2.html">Alternative interface for TGL_kmeans
</a></p>
      <ul>
        <li><code><a href="TGL_kmeans2.html">TGL_kmeans2</a></code></li>
      </ul>
    
    </div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#section-importing-datasets">Importing datasets</a></li>
      <li><a href="#section-average-methylation">Average methylation</a></li>
      <li><a href="#section-tidy-cpgs-functions">Tidy CpGs Functions</a></li>
      <li><a href="#section-pattern-generation-functions">Pattern generation functions</a></li>
      <li><a href="#section-epipolymorphism-functions">Epipolymorphism functions</a></li>
      <li><a href="#section-chip-seq-functions">Chip-seq functions</a></li>
      <li><a href="#section-global-definitions">Global definitions</a></li>
      <li><a href="#section-utils">Utils</a></li>
    </ul>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Aviezer Lifshitz, Elad Chomsky, Amos Tanay.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  </body>
</html>
