<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>shaman - Sampling HiC contAct Matrices for Aparametric Normalization • shaman</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="jquery.sticky-kit.min.js"></script><script src="pkgdown.js"></script><!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">shaman</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="articles/shaman-package.html">
    <span class="fa fa-road"></span>
     
    Usage
  </a>
</li>
<li>
  <a href="articles/import.html">
    <span class="fa fa-filter"></span>
     
    Import
  </a>
</li>
<li>
  <a href="reference/index.html">
    <span class="fa fa-th-list"></span>
     
    Reference
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://bitbucket.org/tanaylab/shaman">
    <span class="fa fa-bitbucket-square"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    

    
    
<div class="contents">
<div id="the-shaman-package---sampling-hic-contact-matrices-for-a-parametric-normalization" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#the-shaman-package---sampling-hic-contact-matrices-for-a-parametric-normalization" class="anchor"></a>The shaman package - sampling HiC contact matrices for a-parametric normalization</h1></div>
<p>Shaman package impelements functions for resampling Hi-C matrices in order to generate expected contact distributions given constraints on marginal coverage and contact-distance probability distributions. The package also provides support for visualizing normalized matrices and statistical analysis of contact distributions around selected landmarks. It is an R package embedding algorithms implemened in C++, which is built over support from the Tanay’s lab <em>Misha</em> genomic database system that is provided with the package.</p>
<p>The normalization workflow consists of the following steps:</p>
<ol style="list-style-type: decimal">
<li>
<a href="https://tanaylab.bitbucket.io/shaman/articles/import.html">Import</a> observed HiC data to misha db. In order to utilize full functionality of the shaman package, you need to construct a Misha database for your genome of interest, and import your Hi-C data to Misha binary Quad-tree format.</li>
<li>Shuffle observed data to generate expected data. Reshuffling of an entire dataset will require 7 hours per 1 billion reads on a machine with one core per chromosome. Note that shaman can also run distributed using Sun Grid Engine.</li>
<li>Compute the normalized score of each observed data point. This computation can be conducted inline for a specific genomic region, or pre-processed on the entire dataset using distributed computation. Score computation on 1 billion reads on a distributed system may take 4-10 hours, depending on the number of cores available.</li>
<li>Visualize normalized contact maps. Precomputed normalized scores allow for flexible analysis of contact distributions.</li>
<li>Quantify contact enrichment acoording to genomic features.</li>
</ol>
<p>In the example misha database provided in this package we have created a low-footprint matrix to examplify the shaman workflow. We included 4.6 million contacts from ELA K562 dataset covering the hoxd locus (chr2:175e06-178e06) and convergent CTCF regions. Processing the complete matrix from this study requires downloading the full contact list and regenerating the reshuffled matrix.</p>
<div id="code" class="section level3">
<h3 class="hasAnchor">
<a href="#code" class="anchor"></a>Code</h3>
<p>Source code can be found at: <a href="https://bitbucket.org/tanaylab/shaman" class="uri">https://bitbucket.org/tanaylab/shaman</a></p>
</div>
<div id="requirements" class="section level3">
<h3 class="hasAnchor">
<a href="#requirements" class="anchor"></a>Requirements</h3>
<ul>
<li><em>Perl</em></li>
<li>
<em>devtools</em> R package (optional, for automatic installation of <em>bioconductor</em> dependencies)</li>
<li>
<em>System:</em> multi-core unix / linux based system or SGE (sun grid engine) cluster for distributed computing are required for large HiC datasets.</li>
</ul>
</div>
<div id="installation" class="section level3">
<h3 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h3>
<p>The quickest way to install <em>shaman</em> is to use the following command:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">devtools::<span class="kw">install_bitbucket</span>(<span class="st">'tanaylab/shaman@default'</span>, 
                            <span class="dt">vignette =</span> <span class="ot">TRUE</span>, 
                            <span class="dt">repos=</span><span class="kw">c</span>(<span class="kw">getOption</span>(<span class="st">'repos'</span>), <span class="st">'https://tanaylab.bitbucket.io/repo'</span>))</code></pre></div>
<p>If devtools fails to install all the bioconductor requirments please install <em>Gviz</em> and <em>GenomeInfoDb</em> manually from bioconductor:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(<span class="st">"https://bioconductor.org/biocLite.R"</span>)
<span class="kw">biocLite</span>(<span class="st">"Gviz"</span>)
<span class="kw">biocLite</span>(<span class="st">"GenomeInfoDb"</span>)</code></pre></div>
<div id="installing-without-the-devtools-package" class="section level4">
<h4 class="hasAnchor">
<a href="#installing-without-the-devtools-package" class="anchor"></a>Installing without the <em>devtools</em> package</h4>
<p>In order to install without the <em>devtools</em> package, please take the following steps:</p>
<ol style="list-style-type: decimal">
<li>Install <em>Gviz</em> and <em>GenomeInfoDb</em> from <em>bioconductor</em>:</li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(<span class="st">"https://bioconductor.org/biocLite.R"</span>)
<span class="kw">biocLite</span>(<span class="st">"Gviz"</span>)
<span class="kw">biocLite</span>(<span class="st">"GenomeInfoDb"</span>)</code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Install <em>shaman</em>:</li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">install.packages</span>(<span class="st">'shaman'</span>, <span class="dt">repos=</span><span class="kw">c</span>(<span class="kw">getOption</span>(<span class="st">'repos'</span>), <span class="st">'https://tanaylab.bitbucket.io/repo'</span>)) </code></pre></div>
</div>
<div id="when-all-else-fails" class="section level4">
<h4 class="hasAnchor">
<a href="#when-all-else-fails" class="anchor"></a>When all else fails:</h4>
<p>In order to install from source, please take the following steps:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">install.packages</span>(<span class="st">"http://www.wisdom.weizmann.ac.il/~nettam/shaman/misha_3.5.6.tar.gz"</span>, , <span class="dt">repos=</span><span class="ot">NULL</span>) <span class="co"># Download and install misha package) </span>
<span class="kw">source</span>(<span class="st">"https://bioconductor.org/biocLite.R"</span>) <span class="co">#installing Gviz</span>
<span class="kw">biocLite</span>(<span class="st">"Gviz"</span>)
<span class="kw">biocLite</span>(<span class="st">"GenomeInfoDb"</span>)
<span class="kw">install.packages</span>(<span class="st">"http://www.wisdom.weizmann.ac.il/~nettam/shaman/shaman_2.0.tar.gz"</span>, , <span class="dt">repos=</span><span class="ot">NULL</span>) <span class="co"># Download and install shaman package)</span></code></pre></div>
</div>
</div>
<div id="using-the-package" class="section level3">
<h3 class="hasAnchor">
<a href="#using-the-package" class="anchor"></a>Using the package</h3>
<p>Please refer to <a href="https://tanaylab.bitbucket.io/shaman/articles/shaman-package.html" class="uri">https://tanaylab.bitbucket.io/shaman/articles/shaman-package.html</a> for usage and workflow.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2 class="hasAnchor">
<a href="#sidebar" class="anchor"></a>Links</h2>
<ul class="list-unstyled">
<li>Report a bug at <br><a href="https://bitbucket.com/tanaylab/shaman/issues">https://​bitbucket.com/​tanaylab/​shaman/​issues</a>
</li>
</ul>
<h2>License</h2>
<p>GPL</p>
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Amos Tanay <br><small class="roles"> Author </small> </li>
<li>Netta Mendelson Cohen <br><small class="roles"> Author, maintainer </small> </li>
</ul>
</div>

</div>


      <footer><div class="copyright">
  <p>Developed by Amos Tanay, Netta Mendelson Cohen.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
