<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Advanced analysis • gpatterns</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../jquery.sticky-kit.min.js"></script><script src="../../pkgdown.js"></script><!-- mathjax --><script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../../index.html">gpatterns</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">Home</a>
</li>
<li>
  <a href="../../articles/import.html">
    <span class="fa fa-truck"></span>
     
    Import
  </a>
</li>
<li>
  <a href="../../articles/articles/analysis.html">
    <span class="fa fa-bar-chart"></span>
     
    Analyse
  </a>
</li>
<li>
  <a href="../../articles/articles/advanced.html">
    <span class="fa fa-stethoscope"></span>
     
    Advanced
  </a>
</li>
<li>
  <a href="../../reference/index.html">
    <span class="fa fa-th-list"></span>
     
    Reference
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://bitbucket.org/tanaylab/gpatterns">
    <span class="fa fa-bitbucket-square"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>Advanced analysis</h1>
                        <h4 class="author">Aviezer Lifshitz</h4>
            
            <h4 class="date">2017-06-18</h4>
          </div>

    
    
<div class="contents">
<p>IMPORTANT: THIS IS ONLY A SANDBOX. IT IS (NOT EVEN) IN BETA STAGE, READ WITH CAUTION AND CONSULT WITH US IF YOU HAVE ANY QUESTIONS. NO GUARANTEES!</p>
<p>This part is intended for more advanced users that want to do additional analyses using gpatterns and ‘misha’. For a standard analytical pipline please refer to <code>analysis</code> section.</p>
<div id="data" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#data" class="anchor"> </a></body></html>Data</h2>
<p>We we will use the same data from the analysis section: a dataset of UMI-RRBS of breast and lung tumors. The dataset contains 16 breast tumors samples from 4 patients, with multiple sections from each sample.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(gpatterns)
<span class="kw">gsetroot</span>(<span class="st">'/home/aviezerl/proj/sandbox_db'</span>)
<span class="kw">options</span>(<span class="dt">gmax.data.size=</span><span class="fl">1e09</span>)
tracks &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/gpatterns.ls.html">gpatterns.ls</a></span>(<span class="st">'gpatterns_nugget.Breast_.+_.+'</span>)
<span class="kw">print</span>(tracks)</code></pre></div>
<pre><code>##  [1] "gpatterns_nugget.Breast_2460_1_3"   
##  [2] "gpatterns_nugget.Breast_2460_1_6"   
##  [3] "gpatterns_nugget.Breast_2460_1_8"   
##  [4] "gpatterns_nugget.Breast_2460_Fresh" 
##  [5] "gpatterns_nugget.Breast_3174_1_2"   
##  [6] "gpatterns_nugget.Breast_3174_1_2p"  
##  [7] "gpatterns_nugget.Breast_3174_1_8"   
##  [8] "gpatterns_nugget.Breast_3174_1_9"   
##  [9] "gpatterns_nugget.Breast_3174_Fresh" 
## [10] "gpatterns_nugget.Breast_4310_1_2"   
## [11] "gpatterns_nugget.Breast_4310_Fresh" 
## [12] "gpatterns_nugget.Breast_4310_Normal"
## [13] "gpatterns_nugget.Breast_4550_3_1"   
## [14] "gpatterns_nugget.Breast_4550_3_2"   
## [15] "gpatterns_nugget.Breast_4550_3_3"   
## [16] "gpatterns_nugget.Breast_4550_Fresh"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">names &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">'gpatterns_nugget.'</span>, <span class="st">''</span>, tracks)
<span class="kw">print</span>(names)</code></pre></div>
<pre><code>##  [1] "Breast_2460_1_3"    "Breast_2460_1_6"    "Breast_2460_1_8"   
##  [4] "Breast_2460_Fresh"  "Breast_3174_1_2"    "Breast_3174_1_2p"  
##  [7] "Breast_3174_1_8"    "Breast_3174_1_9"    "Breast_3174_Fresh" 
## [10] "Breast_4310_1_2"    "Breast_4310_Fresh"  "Breast_4310_Normal"
## [13] "Breast_4550_3_1"    "Breast_4550_3_2"    "Breast_4550_3_3"   
## [16] "Breast_4550_Fresh"</code></pre>
</div>
<div id="data-internals" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#data-internals" class="anchor"> </a></body></html>Data internals</h2>
<p>Each track contains (at least) 5 components:</p>
<ul>
<li>4 ‘misha’ (sparse) tracks that contain data on CpGs in the genome: (Note that CpGs without data will be missing (NA) in the tracks)
<ul>
<li>
<em>‘track.meth’</em> with the pileup of methylated calls</li>
<li>
<em>‘track.unmeth’</em> with the pileup of unmethylated calls</li>
<li>
<em>‘track.cov’</em> with the total number of calls</li>
<li>
<em>‘track.avg’</em> with the average methylation (‘track.meth’ / ‘track.cov’)</li>
</ul>
</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">gtrack.ls</span>(tracks[<span class="dv">1</span>])</code></pre></div>
<pre><code>##  [1] "gpatterns_nugget.Breast_2460_1_3.avg"     
##  [2] "gpatterns_nugget.Breast_2460_1_3.cov"     
##  [3] "gpatterns_nugget.Breast_2460_1_3.epipoly" 
##  [4] "gpatterns_nugget.Breast_2460_1_3.fid"     
##  [5] "gpatterns_nugget.Breast_2460_1_3.meth"    
##  [6] "gpatterns_nugget.Breast_2460_1_3.n"       
##  [7] "gpatterns_nugget.Breast_2460_1_3.n0"      
##  [8] "gpatterns_nugget.Breast_2460_1_3.n1"      
##  [9] "gpatterns_nugget.Breast_2460_1_3.nc"      
## [10] "gpatterns_nugget.Breast_2460_1_3.ncpg"    
## [11] "gpatterns_nugget.Breast_2460_1_3.nx"      
## [12] "gpatterns_nugget.Breast_2460_1_3.pat_cov3"
## [13] "gpatterns_nugget.Breast_2460_1_3.pat_cov5"
## [14] "gpatterns_nugget.Breast_2460_1_3.pat_cov7"
## [15] "gpatterns_nugget.Breast_2460_1_3.pat_meth"
## [16] "gpatterns_nugget.Breast_2460_1_3.pat2"    
## [17] "gpatterns_nugget.Breast_2460_1_3.unmeth"</code></pre>
<ul>
<li>tidy CpGs - comma separated files that contain the raw calls for the sequencing data:</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../../reference/gpatterns.get_tidy_cpgs.html">gpatterns.get_tidy_cpgs</a></span>(tracks[<span class="dv">1</span>], <span class="dt">intervals=</span><span class="kw">gintervals</span>(<span class="dv">1</span>,<span class="fl">1e6</span>,<span class="fl">1e6</span> +<span class="st"> </span><span class="fl">1e4</span>))</code></pre></div>
<pre><code>## # A tibble: 1,949 × 12
##                                                read_id chrom   start
##                                                  &lt;chr&gt; &lt;chr&gt;   &lt;dbl&gt;
## 1    Breast_2406_1.3_2_A:2368391:AGAGGT:Xba_Xba_B_Dark  chr1 1001409
## 2  Breast_2406_1.3_2_A:3080811:ATATAA:Xba_EcoRI_T_Dark  chr1 1001409
## 3  Breast_2406_1.3_2_A:4179231:GAGGAG:Xba_EcoRI_T_Dark  chr1 1001409
## 4    Breast_2406_1.3_2_A:2462062:GGTGGT:Xba_Xba_B_Dark  chr1 1001409
## 5    Breast_2406_1.3_2_A:4163117:GTAGTA:Xba_Xba_B_Dark  chr1 1001409
## 6    Breast_2406_1.3_2_A:2677641:TGAGGG:Xba_Xba_T_Dark  chr1 1001409
## 7    Breast_2406_1.3_2_A:1844720:AGAGGT:Xba_Xba_T_Dark  chr1 1001510
## 8    Breast_2406_1.3_2_A:2527743:GGGTTT:Xba_Xba_T_Dark  chr1 1001510
## 9  Breast_2406_1.3_2_A:4552887:GTAGTA:Xba_EcoRI_T_Dark  chr1 1001510
## 10   Breast_2406_1.3_2_A:2384712:TAAGAG:Xba_Xba_B_Dark  chr1 1001510
##        end strand   umi1  umi2 insert_len   num  cg_pos
##      &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt;      &lt;dbl&gt; &lt;int&gt;   &lt;dbl&gt;
## 1  1001509      - AGAGGT     -         98     4 1001471
## 2  1001509      - ATATAA     -        -92     1 1001471
## 3  1001509      - GAGGAG     -         98     1 1001471
## 4  1001509      - GGTGGT     -         98     2 1001471
## 5  1001509      - GTAGTA     -         98     1 1001471
## 6  1001509      - TGAGGG     -        -98     5 1001471
## 7  1001664      + AGAGGT     -        -98     2 1001471
## 8  1001664      + GGGTTT     -        -98     2 1001471
## 9  1001664      + GTAGTA     -         98     1 1001471
## 10 1001664      + TAAGAG     -         98     4 1001471
## # ... with 1,939 more rows, and 2 more variables: meth &lt;dbl&gt;, qual &lt;dbl&gt;</code></pre>
<p>The tidy_cpgs table contains the following fields:</p>
<ul>
<li>
<em>read_id</em>: ID of the read from which the methylation call came from.</li>
<li>
<em>chrom, start, end, strand</em>: position of the <em>read</em>. single end reads would have ‘-’ in the ‘end’ field, and restriction based protocals such as RRBS would have the position of the restriction fragment.</li>
<li>
<em>umi1, umi2</em>: unique moclecular identifiers (UMIs) if exists.</li>
<li>
<em>insert_len</em>: insert length (only for paired end reads)</li>
<li>
<em>num (optional)</em>: number of times the molecule was observed. Protocols that do not have a duplicate filtering mechanism (such as vanilla RRBS) would not have this field.</li>
<li>
<em>cg_pos</em>: position of the CpG (on the ‘C’ bp) in chromosome .</li>
<li>
<em>meth</em>: 1 if the CpG was observed methylated and 0 if unmethyltated.</li>
<li>
<em>qual</em>: quality of the bp in which the methylation call was observed (C in C-&gt;T conversion protocols, G in G-&gt;A conversion protocols).</li>
</ul>
<p>Storing the raw reads allows us to analyze coupled methylation calls, and calculate features like epipolymorphism or other attributes of the full methylation pattern distribution, see ‘applying functions on tidy_CpGs’ section.</p>
</div>
<div id="extracting-methylation-data" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#extracting-methylation-data" class="anchor"> </a></body></html>Extracting methylation data</h2>
<p>In order to extract the methylation signal from a track we can use the basic misha functions, for example, to extract the methylation calls from chromosome1, 1,000,000 to 1,0010,000 we can do:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">gextract</span>(<span class="kw">paste0</span>(tracks[<span class="dv">1</span>], <span class="kw">c</span>(<span class="st">'.meth'</span>, <span class="st">'.unmeth'</span>, <span class="st">'.avg'</span>, <span class="st">'.cov'</span>)), <span class="dt">intervals=</span><span class="kw">gintervals</span>(<span class="dv">1</span>,<span class="fl">1e6</span>,<span class="fl">1e6+1e4</span>), <span class="dt">iterator=</span><span class="kw">paste0</span>(tracks[<span class="dv">1</span>], <span class="st">'.cov'</span>), <span class="dt">colnames=</span><span class="kw">c</span>(<span class="st">'meth'</span>, <span class="st">'unmeth'</span>, <span class="st">'avg'</span>, <span class="st">'cov'</span>)) %&gt;%<span class="st"> </span>tbl_df</code></pre></div>
<pre><code>## # A tibble: 170 × 8
##     chrom   start     end  meth unmeth       avg   cov intervalID
## *  &lt;fctr&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt;      &lt;int&gt;
## 1    chr1 1001471 1001472    12      9 0.5714286    21          1
## 2    chr1 1002663 1002664    15      1 0.9375000    16          1
## 3    chr1 1002666 1002667    16      0 1.0000000    16          1
## 4    chr1 1002687 1002688    13      2 0.8666667    15          1
## 5    chr1 1002689 1002690    11      3 0.7857143    14          1
## 6    chr1 1002697 1002698     8      0 1.0000000     8          1
## 7    chr1 1002702 1002703    12      1 0.9230769    13          1
## 8    chr1 1002705 1002706    14      1 0.9333333    15          1
## 9    chr1 1002737 1002738     4      1 0.8000000     5          1
## 10   chr1 1002739 1002740     4      2 0.6666667     6          1
## # ... with 160 more rows</code></pre>
<p>gpatterns provides wrapper to this function, , that has the following additional features: (Note that all this features can be achived only with ‘misha’, e.g. using virtual tracks)</p>
<ul>
<li>Extraction of multiple tracks.</li>
<li>Extraction of methyation of larger regions than a single CpG. Repeating the naive misha approach on larger intervals would give us average of averages. This is usually not what we want, so  computes meth / (meth + unmeth) for the entire interval.</li>
<li>Filtering by coverage, number of CpGs, minimal variance, range, etc.</li>
<li>Extraction of summation of tracks.</li>
</ul>
<p>For example, if we want to extract the average methylation in MSP1 fragments from all the tracks, limiting ourselves to fragments with covereage of 10 in at least 5 samples that have at least 2 CpGs we can write:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../../reference/gpatterns.get_avg_meth.html">gpatterns.get_avg_meth</a></span>(tracks, <span class="dt">names=</span>names, <span class="dt">intervals=</span><span class="kw">gintervals</span>(<span class="dv">1</span>, <span class="fl">1e6</span>, <span class="fl">1e6+1e4</span>), <span class="dt">iterator=</span><span class="st">'intervs.msp1.fid'</span>, <span class="dt">min_cov =</span> <span class="dv">10</span>, <span class="dt">min_samples=</span><span class="dv">5</span>, <span class="dt">min_cpgs =</span> <span class="dv">2</span>)</code></pre></div>
<pre><code>## Taking only intervals with at least 2 CpGs</code></pre>
<pre><code>## extracting...</code></pre>
<pre><code>## Taking only intervals with coverage &gt;= 10 in at least 5 samples</code></pre>
<pre><code>## number of intervals: 30</code></pre>
<pre><code>## # A tibble: 480 × 9
##     chrom   start     end intervalID            samp  meth unmeth
##    &lt;fctr&gt;   &lt;dbl&gt;   &lt;dbl&gt;      &lt;int&gt;           &lt;chr&gt; &lt;dbl&gt;  &lt;dbl&gt;
## 1    chr1 1001409 1001509          1 Breast_2460_1_3    12      9
## 2    chr1 1001510 1001664          1 Breast_2460_1_3   NaN    NaN
## 3    chr1 1002618 1002731          1 Breast_2460_1_3    89      8
## 4    chr1 1002732 1002764          1 Breast_2460_1_3    23      9
## 5    chr1 1002824 1003025          1 Breast_2460_1_3   NaN    NaN
## 6    chr1 1003026 1003262          1 Breast_2460_1_3   NaN    NaN
## 7    chr1 1003283 1003374          1 Breast_2460_1_3    46    153
## 8    chr1 1003887 1004001          1 Breast_2460_1_3     6    159
## 9    chr1 1004538 1004706          1 Breast_2460_1_3     0     26
## 10   chr1 1004707 1004788          1 Breast_2460_1_3     1     85
##           avg   cov
##         &lt;dbl&gt; &lt;dbl&gt;
## 1  0.57142857    21
## 2         NaN   NaN
## 3  0.91752577    97
## 4  0.71875000    32
## 5         NaN   NaN
## 6         NaN   NaN
## 7  0.23115578   199
## 8  0.03636364   165
## 9  0.00000000    26
## 10 0.01162791    86
## # ... with 470 more rows</code></pre>
<p>We can limit ourselves to ‘interesting’ fragments, e.g. fragments with minimal range (‘abs(min(m) - max(m) &gt;= min_range’):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../../reference/gpatterns.get_avg_meth.html">gpatterns.get_avg_meth</a></span>(tracks, <span class="dt">names=</span>names, <span class="dt">intervals=</span><span class="kw">gintervals</span>(<span class="dv">1</span>, <span class="fl">1e6</span>, <span class="fl">1e6+1e4</span>), <span class="dt">iterator=</span><span class="st">'intervs.msp1.fid'</span>, <span class="dt">min_cov =</span> <span class="dv">10</span>, <span class="dt">min_samples=</span><span class="dv">5</span>, <span class="dt">min_cpgs =</span> <span class="dv">2</span>, <span class="dt">min_range =</span> <span class="fl">0.3</span>)</code></pre></div>
<pre><code>## Taking only intervals with at least 2 CpGs</code></pre>
<pre><code>## extracting...</code></pre>
<pre><code>## Taking only intervals with coverage &gt;= 10 in at least 5 samples</code></pre>
<pre><code>## number of intervals: 19</code></pre>
<pre><code>## # A tibble: 304 × 9
##     chrom   start     end intervalID            samp  meth unmeth
##    &lt;fctr&gt;   &lt;dbl&gt;   &lt;dbl&gt;      &lt;int&gt;           &lt;chr&gt; &lt;dbl&gt;  &lt;dbl&gt;
## 1    chr1 1001409 1001509          1 Breast_2460_1_3    12      9
## 2    chr1 1001510 1001664          1 Breast_2460_1_3   NaN    NaN
## 3    chr1 1002618 1002731          1 Breast_2460_1_3    89      8
## 4    chr1 1002824 1003025          1 Breast_2460_1_3   NaN    NaN
## 5    chr1 1003026 1003262          1 Breast_2460_1_3   NaN    NaN
## 6    chr1 1003283 1003374          1 Breast_2460_1_3    46    153
## 7    chr1 1005099 1005188          1 Breast_2460_1_3    58     29
## 8    chr1 1005189 1005260          1 Breast_2460_1_3    12     56
## 9    chr1 1005316 1005500          1 Breast_2460_1_3     1      3
## 10   chr1 1006940 1007053          1 Breast_2460_1_3     0     15
##          avg   cov
##        &lt;dbl&gt; &lt;dbl&gt;
## 1  0.5714286    21
## 2        NaN   NaN
## 3  0.9175258    97
## 4        NaN   NaN
## 5        NaN   NaN
## 6  0.2311558   199
## 7  0.6666667    87
## 8  0.1764706    68
## 9  0.2500000     4
## 10 0.0000000    15
## # ... with 294 more rows</code></pre>
<p>If we want to extract data in single CpG level, we can explicitly set the iterator to CpG intervals or leave it NULL:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../../reference/gpatterns.get_avg_meth.html">gpatterns.get_avg_meth</a></span>(tracks, <span class="dt">names=</span>names, <span class="dt">intervals=</span><span class="kw">gintervals</span>(<span class="dv">1</span>, <span class="fl">1e6</span>, <span class="fl">1e6+1e4</span>), <span class="dt">min_cov =</span> <span class="dv">10</span>, <span class="dt">min_samples=</span><span class="dv">5</span>, <span class="dt">min_cpgs =</span> <span class="dv">2</span>, <span class="dt">min_range =</span> <span class="fl">0.3</span>)</code></pre></div>
<pre><code>## Using cpgs as iterator</code></pre>
<pre><code>## extracting...</code></pre>
<pre><code>## 0%...100%</code></pre>
<pre><code>## Taking only intervals with coverage &gt;= 10 in at least 5 samples</code></pre>
<pre><code>## number of intervals: 48</code></pre>
<pre><code>## # A tibble: 768 × 9
##     chrom   start     end intervalID            samp  meth unmeth
##    &lt;fctr&gt;   &lt;dbl&gt;   &lt;dbl&gt;      &lt;int&gt;           &lt;chr&gt; &lt;dbl&gt;  &lt;dbl&gt;
## 1    chr1 1001471 1001472          1 Breast_2460_1_3    12      9
## 2    chr1 1002663 1002664          1 Breast_2460_1_3    15      1
## 3    chr1 1002666 1002667          1 Breast_2460_1_3    16      0
## 4    chr1 1002687 1002688          1 Breast_2460_1_3    13      2
## 5    chr1 1002689 1002690          1 Breast_2460_1_3    11      3
## 6    chr1 1002697 1002698          1 Breast_2460_1_3     8      0
## 7    chr1 1002702 1002703          1 Breast_2460_1_3    12      1
## 8    chr1 1002705 1002706          1 Breast_2460_1_3    14      1
## 9    chr1 1003298 1003299          1 Breast_2460_1_3     6     20
## 10   chr1 1003302 1003303          1 Breast_2460_1_3     6     20
##          avg   cov
##        &lt;dbl&gt; &lt;dbl&gt;
## 1  0.5714286    21
## 2  0.9375000    16
## 3  1.0000000    16
## 4  0.8666667    15
## 5  0.7857143    14
## 6  1.0000000     8
## 7  0.9230769    13
## 8  0.9333333    15
## 9  0.2307692    26
## 10 0.2307692    26
## # ... with 758 more rows</code></pre>
<p>Sometimes it is useful to extract only the average methylation in a non ‘tidy’ format. This may also boost performance:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../../reference/gpatterns.get_avg_meth.html">gpatterns.get_avg_meth</a></span>(tracks, <span class="dt">names=</span>names, <span class="dt">intervals=</span><span class="kw">gintervals</span>(<span class="dv">1</span>, <span class="fl">1e6</span>, <span class="fl">1e6+1e4</span>), <span class="dt">iterator=</span><span class="st">'intervs.msp1.fid'</span>, <span class="dt">min_cov =</span> <span class="dv">10</span>, <span class="dt">min_cpgs =</span> <span class="dv">2</span>, <span class="dt">min_samples=</span><span class="dv">5</span>, <span class="dt">tidy=</span><span class="ot">FALSE</span>) </code></pre></div>
<pre><code>## Taking only intervals with at least 2 CpGs</code></pre>
<pre><code>## Taking only intervals with coverage &gt;= 10 in at least 5 samples</code></pre>
<pre><code>## number of intervals: 30</code></pre>
<pre><code>## # A tibble: 30 × 20
##     chrom   start     end Breast_2460_1_3 Breast_2460_1_6 Breast_2460_1_8
## *  &lt;fctr&gt;   &lt;dbl&gt;   &lt;dbl&gt;           &lt;dbl&gt;           &lt;dbl&gt;           &lt;dbl&gt;
## 1    chr1 1001409 1001509      0.57142857      0.31578947      0.72727273
## 2    chr1 1001510 1001664             NaN      0.57894737      1.00000000
## 3    chr1 1002618 1002731      0.91752577      0.97849462      0.93548387
## 4    chr1 1002732 1002764      0.71875000      0.96825397      0.96551724
## 5    chr1 1002824 1003025             NaN             NaN             NaN
## 6    chr1 1003026 1003262             NaN             NaN             NaN
## 7    chr1 1003283 1003374      0.23115578      0.13815789      0.56122449
## 8    chr1 1003887 1004001      0.03636364      0.02000000      0.02857143
## 9    chr1 1004538 1004706      0.00000000      0.01724138      0.03636364
## 10   chr1 1004707 1004788      0.01162791      0.02586207      0.00000000
##    Breast_2460_Fresh Breast_3174_1_2 Breast_3174_1_2p
## *              &lt;dbl&gt;           &lt;dbl&gt;            &lt;dbl&gt;
## 1                NaN      0.75000000        0.8333333
## 2                NaN      1.00000000        1.0000000
## 3          0.9285714      0.88888889        0.8275862
## 4          1.0000000             NaN              NaN
## 5                NaN      1.00000000        1.0000000
## 6                NaN             NaN              NaN
## 7          0.4146341      0.81818182        0.8360656
## 8          0.0000000      0.03174603        0.0781250
## 9                NaN      0.07843137        0.0000000
## 10         0.0000000      0.01694915        0.0000000
## # ... with 20 more rows, and 11 more variables: Breast_3174_1_8 &lt;dbl&gt;,
## #   Breast_3174_1_9 &lt;dbl&gt;, Breast_3174_Fresh &lt;dbl&gt;, Breast_4310_1_2 &lt;dbl&gt;,
## #   Breast_4310_Fresh &lt;dbl&gt;, Breast_4310_Normal &lt;dbl&gt;,
## #   Breast_4550_3_1 &lt;dbl&gt;, Breast_4550_3_2 &lt;dbl&gt;, Breast_4550_3_3 &lt;dbl&gt;,
## #   Breast_4550_Fresh &lt;dbl&gt;, intervalID &lt;int&gt;</code></pre>
<p>gpatterns provides a wrapper to extract the ‘sum’ of tracks, for example, if we want to combine the methylation of all the sections from tumor 2460 we can add ‘sum_tracks = TRUE’:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../../reference/gpatterns.get_avg_meth.html">gpatterns.get_avg_meth</a></span>(tracks[<span class="dv">1</span>:<span class="dv">4</span>], <span class="dt">intervals=</span><span class="kw">gintervals</span>(<span class="dv">1</span>, <span class="fl">1e6</span>, <span class="fl">1e6+1e4</span>), <span class="dt">iterator=</span><span class="st">'intervs.msp1.fid'</span>, <span class="dt">min_cov =</span> <span class="dv">10</span>, <span class="dt">min_cpgs =</span> <span class="dv">2</span>, <span class="dt">sum_tracks =</span> <span class="ot">TRUE</span>)</code></pre></div>
<pre><code>## Taking only intervals with at least 2 CpGs</code></pre>
<pre><code>## Taking only intervals with coverage &gt;= 10 in at least 1 samples</code></pre>
<pre><code>## # A tibble: 27 × 5
##     chrom   start     end        sum intervalID
## *  &lt;fctr&gt;   &lt;dbl&gt;   &lt;dbl&gt;      &lt;dbl&gt;      &lt;int&gt;
## 1    chr1 1001409 1001509 0.50980392          1
## 2    chr1 1001510 1001664 0.63636364          2
## 3    chr1 1002618 1002731 0.95011876          3
## 4    chr1 1002732 1002764 0.90697674          4
## 5    chr1 1003283 1003374 0.28367347          5
## 6    chr1 1003887 1004001 0.02721088          6
## 7    chr1 1004002 1004015 0.00000000          7
## 8    chr1 1004441 1004495 0.00000000          8
## 9    chr1 1004538 1004706 0.02158273          9
## 10   chr1 1004707 1004788 0.01351351         10
## # ... with 17 more rows</code></pre>
<p>Again, note that all these functions are just wrappers of ‘misha’ functions, and we encourage the reader to try and implement some of them using the basic misha verbs, as in many cases these wrappers would not be enough (e.g. complex track expressions, combination with other tracks).</p>
</div>
<div id="applying-functions-on-tidy_cpgs" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#applying-functions-on-tidy_cpgs" class="anchor"> </a></body></html>Applying functions on tidy_CpGs</h2>
</div>
<div id="analyzing-raw-pattern-data" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#analyzing-raw-pattern-data" class="anchor"> </a></body></html>Analyzing raw pattern data</h2>
</div>
<div id="defining-chipseq-peaks" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#defining-chipseq-peaks" class="anchor"> </a></body></html>Defining chipseq peaks</h2>
</div>
<div id="parallelization" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#parallelization" class="anchor"> </a></body></html>Parallelization</h2>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#data">Data</a></li>
      <li><a href="#data-internals">Data internals</a></li>
      <li><a href="#extracting-methylation-data">Extracting methylation data</a></li>
      <li><a href="#applying-functions-on-tidy_cpgs">Applying functions on tidy_CpGs</a></li>
      <li><a href="#analyzing-raw-pattern-data">Analyzing raw pattern data</a></li>
      <li><a href="#defining-chipseq-peaks">Defining chipseq peaks</a></li>
      <li><a href="#parallelization">Parallelization</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Aviezer Lifshitz, Elad Chomsky, Amos Tanay.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
