<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Basic analysis workflow • gpatterns</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../jquery.sticky-kit.min.js"></script><script src="../../pkgdown.js"></script><!-- mathjax --><script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../../index.html">gpatterns</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">Home</a>
</li>
<li>
  <a href="../../articles/import.html">
    <span class="fa fa-truck"></span>
     
    Import
  </a>
</li>
<li>
  <a href="../../articles/articles/analysis.html">
    <span class="fa fa-bar-chart"></span>
     
    Analyse
  </a>
</li>
<li>
  <a href="../../articles/articles/advanced.html">
    <span class="fa fa-stethoscope"></span>
     
    Advanced
  </a>
</li>
<li>
  <a href="../../reference/index.html">
    <span class="fa fa-th-list"></span>
     
    Reference
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://bitbucket.org/tanaylab/gpatterns">
    <span class="fa fa-bitbucket-square"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>Basic analysis workflow</h1>
                        <h4 class="author">Aviezer Lifshitz</h4>
            
            <h4 class="date">2017-06-18</h4>
          </div>

    
    
<div class="contents">
<p>IMPORTANT: THIS IS A SANDBOX OF THE ANALYTICAL PIPELINE. IT IS (NOT EVEN) IN BETA STAGE, READ WITH CAUTION AND CONSULT WITH US IF YOU HAVE ANY QUESTIONS. NO GUARANTEES!</p>
<p>The package provides functions to assist analyzing 5mC data. The basic analytic workflow consists of the following steps:</p>
<ul>
<li>Single sample analysis:
<ul>
<li>global methylation in different contexts (time of replication, spatial distributions in promoters,exons,enhancers)</li>
<li>CpG correlations</li>
<li>Epipolymorphism</li>
<li>Bipolar analysis</li>
</ul>
</li>
<li>Multiple sample analysis:
<ul>
<li>clustering of average methylation in different contexts (promoters, enhancers, polycomb domains)</li>
<li>annotation of clusters (genes, motifs, other genomic features)</li>
<li>comparison of pattern distribution in selected loci</li>
<li>comparison of ‘background’ methylation</li>
</ul>
</li>
</ul>
<div id="data-umi-rrbs-of-breast-and-lung-tumors" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#data-umi-rrbs-of-breast-and-lung-tumors" class="anchor"> </a></body></html>Data: UMI-RRBS of Breast and Lung tumors</h2>
<p>To explore gpatterns’ functions, we will start with a dataset of UMI-RRBS of breast and lung tumors. The dataset contains 16 breast tumors samples from 4 patients, with multiple sections from each sample.</p>
<ul>
<li>(samples table)</li>
</ul>
<p>After importing the data, we can check that we have all the necessary tracks:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(gpatterns)
<span class="kw">gsetroot</span>(<span class="st">'/home/aviezerl/proj/sandbox_db'</span>)
<span class="kw">options</span>(<span class="dt">gmax.data.size=</span><span class="fl">1e09</span>)
tracks &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/gpatterns.ls.html">gpatterns.ls</a></span>(<span class="st">'gpatterns_nugget.Breast_.+_.+'</span>)
tracks</code></pre></div>
<pre><code>##  [1] "gpatterns_nugget.Breast_2460_1_3"   
##  [2] "gpatterns_nugget.Breast_2460_1_6"   
##  [3] "gpatterns_nugget.Breast_2460_1_8"   
##  [4] "gpatterns_nugget.Breast_2460_Fresh" 
##  [5] "gpatterns_nugget.Breast_3174_1_2"   
##  [6] "gpatterns_nugget.Breast_3174_1_2p"  
##  [7] "gpatterns_nugget.Breast_3174_1_8"   
##  [8] "gpatterns_nugget.Breast_3174_1_9"   
##  [9] "gpatterns_nugget.Breast_3174_Fresh" 
## [10] "gpatterns_nugget.Breast_4310_1_2"   
## [11] "gpatterns_nugget.Breast_4310_Fresh" 
## [12] "gpatterns_nugget.Breast_4310_Normal"
## [13] "gpatterns_nugget.Breast_4550_3_1"   
## [14] "gpatterns_nugget.Breast_4550_3_2"   
## [15] "gpatterns_nugget.Breast_4550_3_3"   
## [16] "gpatterns_nugget.Breast_4550_Fresh"</code></pre>
<p>And remove the prefix for plotting track names:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">names &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">'gpatterns_nugget.'</span>, <span class="st">''</span>, tracks)
names</code></pre></div>
<pre><code>##  [1] "Breast_2460_1_3"    "Breast_2460_1_6"    "Breast_2460_1_8"   
##  [4] "Breast_2460_Fresh"  "Breast_3174_1_2"    "Breast_3174_1_2p"  
##  [7] "Breast_3174_1_8"    "Breast_3174_1_9"    "Breast_3174_Fresh" 
## [10] "Breast_4310_1_2"    "Breast_4310_Fresh"  "Breast_4310_Normal"
## [13] "Breast_4550_3_1"    "Breast_4550_3_2"    "Breast_4550_3_3"   
## [16] "Breast_4550_Fresh"</code></pre>
</div>
<div id="global-methylation" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#global-methylation" class="anchor"> </a></body></html>Global methylation</h2>
<p>The fisrt thing we will to do, is to try and observe the most common feature of methylation data in mammals - low methylation in CpG dense regions and high methylation in regions with low CpG content. We will do that by plotting the average methylation as a function of CpG content (smoothed in 500bp):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">trend &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/gpatterns.global_meth_trend.html">gpatterns.global_meth_trend</a></span>(tracks, <span class="dt">names=</span>names, <span class="dt">xlab=</span><span class="st">'CG content (500bp)'</span>)</code></pre></div>
<p>the resulting object has 2 fields. ‘trend’ with the data, and ‘p’ with the plot:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">trend$trend</code></pre></div>
<pre><code>## # A tibble: 640 × 7
##                               track            name            samp
##                               &lt;chr&gt;           &lt;chr&gt;           &lt;chr&gt;
## 1  gpatterns_nugget.Breast_2460_1_3 Breast_2460_1_3 Breast_2460_1_3
## 2  gpatterns_nugget.Breast_2460_1_3 Breast_2460_1_3 Breast_2460_1_3
## 3  gpatterns_nugget.Breast_2460_1_3 Breast_2460_1_3 Breast_2460_1_3
## 4  gpatterns_nugget.Breast_2460_1_3 Breast_2460_1_3 Breast_2460_1_3
## 5  gpatterns_nugget.Breast_2460_1_3 Breast_2460_1_3 Breast_2460_1_3
## 6  gpatterns_nugget.Breast_2460_1_3 Breast_2460_1_3 Breast_2460_1_3
## 7  gpatterns_nugget.Breast_2460_1_3 Breast_2460_1_3 Breast_2460_1_3
## 8  gpatterns_nugget.Breast_2460_1_3 Breast_2460_1_3 Breast_2460_1_3
## 9  gpatterns_nugget.Breast_2460_1_3 Breast_2460_1_3 Breast_2460_1_3
## 10 gpatterns_nugget.Breast_2460_1_3 Breast_2460_1_3 Breast_2460_1_3
##            breaks      meth breaks_numeric cg_num
##             &lt;chr&gt;     &lt;dbl&gt;          &lt;dbl&gt;  &lt;dbl&gt;
## 1      [0, 0.002] 0.7356322          0.001    330
## 2  (0.002, 0.004] 0.6792340          0.003   1490
## 3  (0.004, 0.006] 0.6930049          0.005   3491
## 4  (0.006, 0.008] 0.6791553          0.007   6188
## 5   (0.008, 0.01] 0.6431006          0.009   9658
## 6   (0.01, 0.012] 0.6587487          0.011  13855
## 7  (0.012, 0.014] 0.6547867          0.013  18358
## 8  (0.014, 0.016] 0.6359075          0.015  22968
## 9  (0.016, 0.018] 0.6462722          0.017  27780
## 10  (0.018, 0.02] 0.6270250          0.019  31500
## # ... with 630 more rows</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">trend$p</code></pre></div>
<p><img src="analysis_files/figure-html/unnamed-chunk-6-1.png" width="864"></p>
<p>Next, we will look at average methylation in different genomic contexts:</p>
<div id="tss" class="section level5">
<h5 class="hasAnchor">
<html><body><a href="#tss" class="anchor"> </a></body></html>tss</h5>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">trend &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/gpatterns.spatial_meth_trend.html">gpatterns.spatial_meth_trend</a></span>(tracks, <span class="dt">names=</span>names, <span class="dt">intervals=</span><span class="st">'tss'</span>)
trend$p</code></pre></div>
<p><img src="analysis_files/figure-html/unnamed-chunk-7-1.png" width="864"></p>
</div>
<div id="enhancers" class="section level5">
<h5 class="hasAnchor">
<html><body><a href="#enhancers" class="anchor"> </a></body></html>enhancers</h5>
</div>
</div>
<div id="correlations-between-cpgs" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#correlations-between-cpgs" class="anchor"> </a></body></html>Correlations between CpGs</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">trend &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/gpatterns.plot_cg_cor.html">gpatterns.plot_cg_cor</a></span>(tracks, <span class="dt">names=</span>names, <span class="dt">adjacent=</span><span class="ot">TRUE</span>, <span class="dt">dist_breaks =</span> <span class="kw">seq</span>(<span class="dv">0</span>,<span class="dv">200</span>,<span class="dv">20</span>), <span class="dt">min_cgs=</span><span class="dv">100</span>)
trend$p</code></pre></div>
<p><img src="analysis_files/figure-html/unnamed-chunk-8-1.png" width="864"></p>
<p>We can repeat that in different genomic contexts as well:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">trend &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/gpatterns.plot_cg_cor.html">gpatterns.plot_cg_cor</a></span>(tracks, <span class="dt">names=</span>names, <span class="dt">adjacent=</span><span class="ot">TRUE</span>, <span class="dt">dist_breaks =</span> <span class="kw">seq</span>(<span class="dv">0</span>,<span class="dv">200</span>,<span class="dv">20</span>), <span class="dt">min_cgs=</span><span class="dv">100</span>, <span class="dt">intervals=</span><span class="st">'cgi'</span>)
trend$p</code></pre></div>
<p><img src="analysis_files/figure-html/unnamed-chunk-9-1.png" width="864"></p>
</div>
<div id="epipolymorphism" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#epipolymorphism" class="anchor"> </a></body></html>Epipolymorphism</h2>
<p>In order to calculate epipolymorphism we first need to define the CpG scope, or ‘pattern space’ and create the patterns attributes of our tracks, see <code>import</code> vignette for details. Assuming we defined a pattern space and created generated patterns for our tracks, we can plot epipolymorphism for different methylation regimes and genomic scopes:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../../reference/gpatterns.epipoly_plot.html">gpatterns.epipoly_plot</a></span>(tracks[<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">5</span>,<span class="dv">13</span>)], <span class="kw">gintervals.all</span>(), <span class="dt">colnames=</span>names[<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">5</span>,<span class="dv">13</span>)])</code></pre></div>
<p><img src="analysis_files/figure-html/unnamed-chunk-10-1.png" width="864"></p>
<p>And again, in different genomic contexts:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../../reference/gpatterns.epipoly_plot.html">gpatterns.epipoly_plot</a></span>(tracks[<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">5</span>,<span class="dv">13</span>)], <span class="dt">intervals=</span><span class="st">'tss'</span>, <span class="dt">colnames=</span>names[<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">5</span>,<span class="dv">13</span>)])</code></pre></div>
<p><img src="analysis_files/figure-html/unnamed-chunk-11-1.png" width="864"></p>
</div>
<div id="clustering" class="section level2">
<h2 class="hasAnchor">
<html><body><a href="#clustering" class="anchor"> </a></body></html>Clustering</h2>
<p>First step for multiple sample analysis is to retrieve the methylation calls in a joined set of CpGs. gpatterns provides a function for that purpose. Given genomic regions <code>gpatterns.get_avg_meth</code> screens for the places that have enough coverage (<code>min_cov</code>) in enough samples (<code>min_samples</code>), and extracts methylation calls in those regions. In addition, there is an options to screen for the most variable regions (<code>var_quantile</code> and <code>min_var</code>), and for minimal number of CpGs per region (<code>min_cpgs</code>). If no iterator is given the function uses CpGs is the given intervals.</p>
<p>Let’s extract the 10% most variable CpGs that have at least 20 methylation calls in at least 15 samples:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">avgs &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/gpatterns.get_avg_meth.html">gpatterns.get_avg_meth</a></span>(<span class="dt">tracks=</span>tracks, <span class="dt">intervals=</span><span class="st">'intervs.msp1.fid'</span>, <span class="dt">var_quantile=</span><span class="fl">0.1</span>, <span class="dt">names=</span>names, <span class="dt">min_cov=</span><span class="dv">20</span>, <span class="dt">min_samples=</span><span class="dv">15</span>)
avgs</code></pre></div>
<p>Now we can cluster the average methylation using kmeans++:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">clust &lt;-<span class="st"> </span><span class="kw"><a href="../../reference/gpatterns.cluster_avg_meth.html">gpatterns.cluster_avg_meth</a></span>(avgs, <span class="dt">K=</span><span class="dv">10</span>)
clust</code></pre></div>
<p>And plot the clustering:</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#data-umi-rrbs-of-breast-and-lung-tumors">Data: UMI-RRBS of Breast and Lung tumors</a></li>
      <li><a href="#global-methylation">Global methylation</a></li>
      <li><a href="#correlations-between-cpgs">Correlations between CpGs</a></li>
      <li><a href="#epipolymorphism">Epipolymorphism</a></li>
      <li><a href="#clustering">Clustering</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Aviezer Lifshitz, Elad Chomsky, Amos Tanay.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
